{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import seaborn as sns\n",
    "from sklearn.impute import SimpleImputer\n",
    "import mgitools.os_helpers as os_helpers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install git+https://github.com/estorrs/mgitools\n",
    "# !pip install sklearn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "CANCER_TYPE_MAP = {\n",
    "    'CPTAC2_BRCA_prospective': 'BRCA_prospective',\n",
    "    'TCGA_BRCA_retrospective': 'BRCA_retrospective',\n",
    "    'BR': 'BRCA_prospective',\n",
    "    'BRCA': 'BRCA_prospective',\n",
    "    'CPTAC2_CRC_prospective': 'CRC_prospective',\n",
    "    'CO': 'CRC_prospective',\n",
    "    'COADREAD': 'CRC_prospective',\n",
    "    'CPTAC2_OV_prospective': 'OV_prospective',\n",
    "    'TCGA_OV_retrospective': 'OV_retrospective',\n",
    "    'OV': 'OV_prospective',\n",
    "    'CPTAC3_CCRCC_discovery': 'CCRCC',\n",
    "    'KIRC': 'CCRCC',\n",
    "    'CPTAC3_GBM_discovery': 'GBM',\n",
    "    'CPTAC3_HNSCC_discovery': 'HNSCC',\n",
    "    'HNSC': 'HNSCC',\n",
    "    'CPTAC3_LSCC_discovery': 'LSCC',\n",
    "    'LUSC': 'LSCC',\n",
    "    'CPTAC3_LUAD_discovery': 'LUAD',\n",
    "    'CPTAC3_UCEC_discovery': 'UCEC'\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## read in driver genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>PANCAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACVR1</td>\n",
       "      <td>UCEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACVR1B</td>\n",
       "      <td>PANCAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>COADREAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>LIHC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>734</th>\n",
       "      <td>ZMYM3</td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>ZNF133</td>\n",
       "      <td>OV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>736</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>PANCAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>ESCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>HNSC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>739 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene    Cancer\n",
       "0      ABL1    PANCAN\n",
       "1     ACVR1      UCEC\n",
       "2    ACVR1B    PANCAN\n",
       "3    ACVR2A  COADREAD\n",
       "4    ACVR2A      LIHC\n",
       "..      ...       ...\n",
       "734   ZMYM3      PRAD\n",
       "735  ZNF133        OV\n",
       "736  ZNF750    PANCAN\n",
       "737  ZNF750      ESCA\n",
       "738  ZNF750      HNSC\n",
       "\n",
       "[739 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = pd.read_csv('/data/driver_genes/bailey_smg_table.tsv', sep='\\t')\n",
    "genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ACC',\n",
       " 'BLCA',\n",
       " 'BRCA',\n",
       " 'CESC',\n",
       " 'CHOL',\n",
       " 'COADREAD',\n",
       " 'DLBC',\n",
       " 'ESCA',\n",
       " 'GBM',\n",
       " 'HNSC',\n",
       " 'KICH',\n",
       " 'KIRC',\n",
       " 'KIRP',\n",
       " 'LAML',\n",
       " 'LGG',\n",
       " 'LIHC',\n",
       " 'LUAD',\n",
       " 'LUSC',\n",
       " 'MESO',\n",
       " 'OV',\n",
       " 'PAAD',\n",
       " 'PANCAN',\n",
       " 'PCPG',\n",
       " 'PRAD',\n",
       " 'SARC',\n",
       " 'SKCM',\n",
       " 'STAD',\n",
       " 'TGCT',\n",
       " 'THCA',\n",
       " 'THYM',\n",
       " 'UCEC',\n",
       " 'UCS',\n",
       " 'UVM']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(set(genes['Cancer']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Tumor suppressor or oncogene prediction (by 20/20+)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PHF6</td>\n",
       "      <td>possible tsg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALK</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>KMT2A</td>\n",
       "      <td>tsg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>KMT2B</td>\n",
       "      <td>tsg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>MAX</td>\n",
       "      <td>oncogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>MED12</td>\n",
       "      <td>oncogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>MEN1</td>\n",
       "      <td>tsg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>188 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Tumor suppressor or oncogene prediction (by 20/20+)\n",
       "0     PHF6                                       possible tsg \n",
       "1     ABL1                                                NaN \n",
       "2      ALK                                                NaN \n",
       "3       AR                                                NaN \n",
       "4     ARAF                                                NaN \n",
       "..     ...                                                ... \n",
       "183  KMT2A                                                tsg \n",
       "184  KMT2B                                                tsg \n",
       "185    MAX                                           oncogene \n",
       "186  MED12                                           oncogene \n",
       "187   MEN1                                                tsg \n",
       "\n",
       "[188 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drivers = pd.read_csv('/data/driver_genes/199_driver_list.txt', sep='\\t')\n",
    "drivers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['ACC', 'BLCA', 'BRCA_prospective', 'CESC', 'CHOL', 'CRC_prospective', 'DLBC', 'ESCA', 'GBM', 'HNSCC', 'KICH', 'CCRCC', 'KIRP', 'LAML', 'LGG', 'LIHC', 'LUAD', 'LSCC', 'MESO', 'OV_prospective', 'PAAD', 'PANCAN', 'PCPG', 'PRAD', 'SARC', 'SKCM', 'STAD', 'TGCT', 'THCA', 'THYM', 'UCEC', 'UCS', 'UVM'])"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_driver_genes = {CANCER_TYPE_MAP.get(c, c):genes[genes['Cancer']==c].set_index('Gene')\n",
    "                          for c in sorted(set(genes['Cancer']))}\n",
    "cancer_to_driver_genes.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARHGAP35</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN2A</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUL3</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EP300</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAT1</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FBXW7</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-A</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HRAS</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM6A</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEAP1</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KLF5</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2D</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NF1</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE2L2</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH1</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RASA1</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RB1</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Cancer\n",
       "Gene           \n",
       "ARHGAP35   LUSC\n",
       "ARID1A     LUSC\n",
       "CDKN2A     LUSC\n",
       "CUL3       LUSC\n",
       "EP300      LUSC\n",
       "FAT1       LUSC\n",
       "FBXW7      LUSC\n",
       "FGFR2      LUSC\n",
       "HLA-A      LUSC\n",
       "HRAS       LUSC\n",
       "KDM6A      LUSC\n",
       "KEAP1      LUSC\n",
       "KLF5       LUSC\n",
       "KMT2D      LUSC\n",
       "NF1        LUSC\n",
       "NFE2L2     LUSC\n",
       "NOTCH1     LUSC\n",
       "PIK3CA     LUSC\n",
       "PTEN       LUSC\n",
       "RASA1      LUSC\n",
       "RB1        LUSC\n",
       "TP53       LUSC"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_driver_genes['LSCC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ucec_genes = genes[[True if c in ['UCEC'] else False for c in genes['Cancer']]]\n",
    "# ucec_genes = pd.merge(drivers, ucec_genes, left_on='Gene', right_on='Gene')\n",
    "# ucec_genes = ucec_genes.set_index('Gene')\n",
    "# ucec_genes.index.name = 'gene'\n",
    "# ucec_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ucec_genes.to_csv('/data/driver_genes/ucec_drivers.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drivers.to_csv('/data/driver_genes/expanded_drivers.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## read in somatic mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/somtatic_mutations/BR.maf',\n",
       " '/data/somtatic_mutations/CCRCC.maf',\n",
       " '/data/somtatic_mutations/CO.maf',\n",
       " '/data/somtatic_mutations/GBM.maf',\n",
       " '/data/somtatic_mutations/HNSCC.maf',\n",
       " '/data/somtatic_mutations/LSCC.maf',\n",
       " '/data/somtatic_mutations/LUAD.maf',\n",
       " '/data/somtatic_mutations/OV.maf',\n",
       " '/data/somtatic_mutations/UCEC.maf']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_dir = '/data/somtatic_mutations/'\n",
    "mutation_fps = sorted(os_helpers.listfiles(mutation_dir, regex=r'.maf$'))\n",
    "mutation_fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3146: DtypeWarning: Columns (88) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "dict_keys(['BRCA_prospective', 'CCRCC', 'CRC_prospective', 'GBM', 'HNSCC', 'LSCC', 'LUAD', 'OV_prospective', 'UCEC'])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_maf = {}\n",
    "for fp in mutation_fps:\n",
    "    cancer = fp.split('/')[-1].split('.')[0]\n",
    "    mutation_df = pd.read_csv(fp, sep='\\t')\n",
    "    mutation_df['sample_id'] = [x.split('_')[0] for x in mutation_df['Tumor_Sample_Barcode']]\n",
    "    mutation_df = mutation_df[mutation_df['Variant_Classification']!='Silent']\n",
    "    cancer_to_maf[CANCER_TYPE_MAP.get(cancer, cancer)] = mutation_df\n",
    "cancer_to_maf.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## read in expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/expression/AML_fpkm.tsv',\n",
       " '/data/expression/BR_fpkm.tsv',\n",
       " '/data/expression/CCRCC_fpkm.tsv',\n",
       " '/data/expression/CM_fpkm.tsv',\n",
       " '/data/expression/CO_fpkm.tsv',\n",
       " '/data/expression/EGAD-PDAC_fpkm.tsv',\n",
       " '/data/expression/GBM_fpkm.tsv',\n",
       " '/data/expression/HNSCC_fpkm.tsv',\n",
       " '/data/expression/LSCC_fpkm.tsv',\n",
       " '/data/expression/LUAD_fpkm.tsv',\n",
       " '/data/expression/OV_fpkm.tsv',\n",
       " '/data/expression/PDA_fpkm.tsv',\n",
       " '/data/expression/SAR_fpkm.tsv',\n",
       " '/data/expression/UCEC_fpkm.tsv',\n",
       " '/data/expression/eocg_fpkm.tsv',\n",
       " '/data/expression/hipsci_managed_fpkm.tsv',\n",
       " '/data/expression/hipsci_public_fpkm.tsv']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_dir = '/data/expression'\n",
    "expression_fps = sorted(os_helpers.listfiles(expression_dir, regex=r'fpkm.tsv$'))\n",
    "expression_fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['AML', 'BRCA_prospective', 'CCRCC', 'CM', 'CRC_prospective', 'EGAD-PDAC', 'GBM', 'HNSCC', 'LSCC', 'LUAD', 'OV_prospective', 'PDA', 'SAR', 'UCEC', 'eocg', 'hipsci_managed', 'hipsci_public'])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_type_to_expression = {}\n",
    "for fp in expression_fps:\n",
    "    cancer = fp.split('/')[-1].replace('_fpkm.tsv', '')\n",
    "    \n",
    "    cancer_type_to_expression[CANCER_TYPE_MAP.get(cancer, cancer)] = pd.read_csv(fp, sep='\\t', index_col='gene')\n",
    "cancer_type_to_expression.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## read in raw protein data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/v2.0/CPTAC2_BRCA_prospective/CPTAC2_BRCA_prospective_broad_acetylome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC2_BRCA_prospective/CPTAC2_BRCA_prospective_broad_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC2_BRCA_prospective/CPTAC2_BRCA_prospective_broad_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC2_CRC_prospective/CPTAC2_CRC_prospective_pnnl_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC2_CRC_prospective/CPTAC2_CRC_prospective_pnnl_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC2_OV_prospective/CPTAC2_OV_prospective_pnnl_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC2_OV_prospective/CPTAC2_OV_prospective_pnnl_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_CCRCC_discovery/CPTAC3_CCRCC_discovery_umich_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_CCRCC_discovery/CPTAC3_CCRCC_discovery_umich_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_GBM_discovery/CPTAC3_GBM_discovery_pnnl_acetylome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_GBM_discovery/CPTAC3_GBM_discovery_pnnl_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_GBM_discovery/CPTAC3_GBM_discovery_pnnl_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_HNSCC_discovery/CPTAC3_HNSCC_discovery_umich_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_HNSCC_discovery/CPTAC3_HNSCC_discovery_umich_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LSCC_discovery/CPTAC3_LSCC_discovery_broad_acetylome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LSCC_discovery/CPTAC3_LSCC_discovery_broad_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LSCC_discovery/CPTAC3_LSCC_discovery_broad_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LSCC_discovery/CPTAC3_LSCC_discovery_broad_ubiquitylome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LUAD_discovery/CPTAC3_LUAD_discovery_broad_acetylome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LUAD_discovery/CPTAC3_LUAD_discovery_broad_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_LUAD_discovery/CPTAC3_LUAD_discovery_broad_proteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_UCEC_discovery/CPTAC3_UCEC_discovery_pnnl_acetylome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_UCEC_discovery/CPTAC3_UCEC_discovery_pnnl_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/CPTAC3_UCEC_discovery/CPTAC3_UCEC_discovery_pnnl_proteome.v2.0.gct',\n",
       " '/data/v2.0/TCGA_BRCA_retrospective/TCGA_BRCA_retrospective_broad_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/TCGA_BRCA_retrospective/TCGA_BRCA_retrospective_broad_proteome.v2.0.gct',\n",
       " '/data/v2.0/TCGA_OV_retrospective/TCGA_OV_retrospective_pnnl_phosphoproteome.v2.0.gct',\n",
       " '/data/v2.0/TCGA_OV_retrospective/TCGA_OV_retrospective_pnnl_proteome.v2.0.gct']"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(os_helpers.listfiles('/data/v2.0/', regex=r'.gct'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# f = open('/data/v2.0/CPTAC3_UCEC_discovery/CPTAC3_UCEC_discovery_pnnl_phosphoproteome.v2.0.gct')\n",
    "# for i, line in enumerate(f):\n",
    "#     print(line)\n",
    "#     if i % 20 == 19:\n",
    "#         break\n",
    "# f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert_gct_to_df(fp):\n",
    "    f = open(fp)\n",
    "    f.readline()\n",
    "    _, n_samples, n_feature_metadata, n_sample_metadata = [int(x) for x in f.readline().strip().split('\\t')]\n",
    "    df = pd.read_csv(f, sep='\\t')\n",
    "    df.columns = [c[1:] if c[0]=='X' else c for c in df.columns]\n",
    "    \n",
    "    ## do some renaming and split metadata into different table\n",
    "    ids = df['id'].to_list()[:n_sample_metadata - 1]\n",
    "    metadata = df.iloc[:n_sample_metadata - 1, -n_samples:].copy()\n",
    "    metadata.index = ids\n",
    "    metadata = metadata.transpose()\n",
    "    \n",
    "    \n",
    "    df = df.iloc[n_sample_metadata:, :]\n",
    "#     print(df.shape)\n",
    "    df = df.set_index('id')\n",
    "    df.index.name = 'site_id'\n",
    "    \n",
    "    return df, metadata\n",
    "\n",
    "\n",
    "def change_sample_ids(df, metadata):\n",
    "    site_cols = list(df.columns[:-metadata.shape[0]])\n",
    "    ids = []\n",
    "    for s_id in df.columns[-metadata.shape[0]:]:\n",
    "        case = metadata.loc[s_id, 'case_id']\n",
    "        case = case.replace('.', '-')\n",
    "        identifier = '.T' if 'Tumor' in metadata.loc[s_id, 'sample_type'] else '.N'\n",
    "        new = case + identifier\n",
    "        ids.append(new)\n",
    "    site_cols += ids\n",
    "    df.columns = site_cols\n",
    "\n",
    "    ids = []\n",
    "    for s_id in metadata.index:\n",
    "        case = metadata.loc[s_id, 'case_id']\n",
    "        case = case.replace('.', '-')\n",
    "        identifier = '.T' if 'Tumor' in metadata.loc[s_id, 'sample_type'] else '.N'\n",
    "        new = case + identifier\n",
    "        ids.append(new)\n",
    "    metadata.index = ids\n",
    "    \n",
    "    return df, metadata\n",
    "\n",
    "def get_cancer_to_dfs(fps):\n",
    "    d = {}\n",
    "    for fp in fps:\n",
    "        cancer = fp.split('/')[3]\n",
    "        cancer = CANCER_TYPE_MAP.get(cancer, cancer)\n",
    "        filetype = fp.split('.v2.0')[-2].split('_')[-1]\n",
    "        print(cancer, filetype)\n",
    "        \n",
    "        if cancer not in d:\n",
    "            d[cancer] = {\n",
    "                k:{} for k in ['acetylome', 'phosphoproteome', 'acetylome', 'ubiquitylome']\n",
    "            }\n",
    "        \n",
    "        df, metadata = convert_gct_to_df(fp)\n",
    "        \n",
    "        ## make sample_type if not already in dataframe\n",
    "        if cancer in ['LSCC', 'LUAD']:\n",
    "            metadata['sample_type'] = ['Normal' if s[-2:]=='.N' else 'Tumor' for s in metadata.index]\n",
    "            \n",
    "        ## fix identifiers\n",
    "        if cancer in ['LSCC', 'LUAD', 'CCRCC']:\n",
    "            df, metadata = change_sample_ids(df, metadata)\n",
    "        \n",
    "        \n",
    "        d[cancer][filetype] = {\n",
    "            'metadata': metadata,\n",
    "            'data': df\n",
    "        }\n",
    "    ## filter out empty datatypes\n",
    "#     d = {for cancer, dicts in d.items() for dtypes, m in dicts.items() if len(m)}\n",
    "    for cancer in list(d.keys()):\n",
    "        for dtypes in list(d[cancer].keys()):\n",
    "            if not len(d[cancer][dtypes]):\n",
    "                d[cancer].pop(dtypes)\n",
    "            \n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['AML', 'BRCA_prospective', 'CCRCC', 'CM', 'CRC_prospective', 'EGAD-PDAC', 'GBM', 'HNSCC', 'LSCC', 'LUAD', 'OV_prospective', 'PDA', 'SAR', 'UCEC', 'eocg', 'hipsci_managed', 'hipsci_public'])"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_type_to_expression.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_prospective acetylome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_prospective phosphoproteome\n",
      "BRCA_prospective proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (2,3,4,5,6,7,8,10,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CRC_prospective phosphoproteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (5,6,7,8,13,15,16,17,18,19,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CRC_prospective proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (2,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OV_prospective phosphoproteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (5,6,13,15,16,17,18,19,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OV_prospective proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (3,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CCRCC phosphoproteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CCRCC proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (1,4,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GBM acetylome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GBM phosphoproteome\n",
      "GBM proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (3,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HNSCC phosphoproteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HNSCC proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (2,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LSCC acetylome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LSCC phosphoproteome\n",
      "LSCC proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (2,3,4,5,6,7,8,10,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LSCC ubiquitylome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD acetylome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD phosphoproteome\n",
      "LUAD proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (2,3,4,5,6,7,8,10,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UCEC acetylome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (6,7,11,13,14,15,16,17,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UCEC phosphoproteome\n",
      "UCEC proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (2,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_retrospective phosphoproteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (8,9,12,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_retrospective proteome\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3417: DtypeWarning: Columns (3,4,5,6,7,9,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OV_retrospective phosphoproteome\n",
      "OV_retrospective proteome\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "dict_keys(['BRCA_prospective', 'CRC_prospective', 'OV_prospective', 'CCRCC', 'GBM', 'HNSCC', 'LSCC', 'LUAD', 'UCEC', 'BRCA_retrospective', 'OV_retrospective'])"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein = get_cancer_to_dfs(fps)\n",
    "cancer_to_protein.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_prospective ['acetylome', 'phosphoproteome', 'proteome']\n",
      "CRC_prospective ['phosphoproteome', 'proteome']\n",
      "OV_prospective ['phosphoproteome', 'proteome']\n",
      "CCRCC ['phosphoproteome', 'proteome']\n",
      "GBM ['acetylome', 'phosphoproteome', 'proteome']\n",
      "HNSCC ['phosphoproteome', 'proteome']\n",
      "LSCC ['acetylome', 'phosphoproteome', 'ubiquitylome', 'proteome']\n",
      "LUAD ['acetylome', 'phosphoproteome', 'proteome']\n",
      "UCEC ['acetylome', 'phosphoproteome', 'proteome']\n",
      "BRCA_retrospective ['phosphoproteome', 'proteome']\n",
      "OV_retrospective ['phosphoproteome', 'proteome']\n"
     ]
    }
   ],
   "source": [
    "for k, d in cancer_to_protein.items():\n",
    "    print(k, list(d.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>original_id</th>\n",
       "      <th>refseq_prot_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>phosphosites</th>\n",
       "      <th>peptide</th>\n",
       "      <th>peptide_start</th>\n",
       "      <th>peptide_end</th>\n",
       "      <th>refseq_tx_id</th>\n",
       "      <th>uniparc_id</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>...</th>\n",
       "      <th>C3N-01646.T</th>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <th>C3N-01648.T</th>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <th>C3N-01649.T</th>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <th>C3N-01651.T</th>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <th>C3N-01808.T</th>\n",
       "      <th>C3N-01808.N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>site_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG:NP_570602.2:S302</th>\n",
       "      <td>NP_570602.2_291_317_1_1_S302</td>\n",
       "      <td>NP_570602.2</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>S302</td>\n",
       "      <td>LHDNQNGWSGDSAPVELILsDETLPAPEFSPEPESGR</td>\n",
       "      <td>283</td>\n",
       "      <td>319</td>\n",
       "      <td>NM_130786</td>\n",
       "      <td>UPI0000167B10</td>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG:NP_570602.2:T171</th>\n",
       "      <td>NP_570602.2_171_184_1_1_T171</td>\n",
       "      <td>NP_570602.2</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>T171</td>\n",
       "      <td>EGDHEFLEVPEAQEDVEAtFPVHQPGNYSCSYR</td>\n",
       "      <td>153</td>\n",
       "      <td>185</td>\n",
       "      <td>NM_130786</td>\n",
       "      <td>UPI0000167B10</td>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG:NP_570602.2:T171;Y180;Y184</th>\n",
       "      <td>NP_570602.2_171_184_3_3_T171Y180Y184</td>\n",
       "      <td>NP_570602.2</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>T171;Y180;Y184</td>\n",
       "      <td>REGDHEFLEVPEAQEDVEAtFPVHQPGNySCSyR</td>\n",
       "      <td>152</td>\n",
       "      <td>185</td>\n",
       "      <td>NM_130786</td>\n",
       "      <td>UPI0000167B10</td>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF:NP_001185747.1:T491</th>\n",
       "      <td>NP_001185747.1_476_491_1_1_T491</td>\n",
       "      <td>NP_001185747.1</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>T491</td>\n",
       "      <td>ITIPALASQNPAIHPFtPPK</td>\n",
       "      <td>475</td>\n",
       "      <td>494</td>\n",
       "      <td>NM_001198818</td>\n",
       "      <td>UPI000006EC73</td>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.69413</td>\n",
       "      <td>0.117396</td>\n",
       "      <td>-0.633383</td>\n",
       "      <td>0.076533</td>\n",
       "      <td>0.467653</td>\n",
       "      <td>0.480063</td>\n",
       "      <td>0.464454</td>\n",
       "      <td>0.643927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M:NP_000005.2:S710</th>\n",
       "      <td>NP_000005.2_708_710_1_1_S710</td>\n",
       "      <td>NP_000005.2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>S710</td>\n",
       "      <td>VGFYEsDVMGR</td>\n",
       "      <td>705</td>\n",
       "      <td>715</td>\n",
       "      <td>NM_000014</td>\n",
       "      <td>UPI0000155718</td>\n",
       "      <td>HGNC:7</td>\n",
       "      <td>...</td>\n",
       "      <td>0.888674</td>\n",
       "      <td>-0.7553</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.407549</td>\n",
       "      <td>-0.73731</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3:NP_056349.1:S468</th>\n",
       "      <td>NP_056349.1_468_468_1_1_S468</td>\n",
       "      <td>NP_056349.1</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>S468</td>\n",
       "      <td>LNIGHLPsAK</td>\n",
       "      <td>461</td>\n",
       "      <td>470</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>UPI0000074256</td>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.721719</td>\n",
       "      <td>0.372557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.097481</td>\n",
       "      <td>0.241512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3:NP_056349.1:S777</th>\n",
       "      <td>NP_056349.1_763_786_1_1_S777</td>\n",
       "      <td>NP_056349.1</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>S777</td>\n",
       "      <td>SCFHSHMNTAVEDAsDDESIPIMYR</td>\n",
       "      <td>763</td>\n",
       "      <td>787</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>UPI0000074256</td>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3:NP_056349.1:S89</th>\n",
       "      <td>NP_056349.1_89_91_1_1_S89</td>\n",
       "      <td>NP_056349.1</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>S89</td>\n",
       "      <td>RGLsSSEK</td>\n",
       "      <td>86</td>\n",
       "      <td>93</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>UPI0000074256</td>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3:NP_056349.1:Y399</th>\n",
       "      <td>NP_056349.1_396_434_1_1_Y399</td>\n",
       "      <td>NP_056349.1</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>Y399</td>\n",
       "      <td>NSSPyRENGQFEENNLSPNETNATVSDNVSQSPTNPGEISQNEK</td>\n",
       "      <td>395</td>\n",
       "      <td>438</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>UPI0000074256</td>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3:NP_056349.1:Y399;T428</th>\n",
       "      <td>NP_056349.1_396_434_2_2_Y399T428</td>\n",
       "      <td>NP_056349.1</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>Y399;T428</td>\n",
       "      <td>NSSPyRENGQFEENNLSPNETNATVSDNVSQSPtNPGEISQNEK</td>\n",
       "      <td>395</td>\n",
       "      <td>438</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>UPI0000074256</td>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>81780 rows × 215 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                          original_id  \\\n",
       "site_id                                                                 \n",
       "A1BG:NP_570602.2:S302                    NP_570602.2_291_317_1_1_S302   \n",
       "A1BG:NP_570602.2:T171                    NP_570602.2_171_184_1_1_T171   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184  NP_570602.2_171_184_3_3_T171Y180Y184   \n",
       "A1CF:NP_001185747.1:T491              NP_001185747.1_476_491_1_1_T491   \n",
       "A2M:NP_000005.2:S710                     NP_000005.2_708_710_1_1_S710   \n",
       "...                                                               ...   \n",
       "ZZZ3:NP_056349.1:S468                    NP_056349.1_468_468_1_1_S468   \n",
       "ZZZ3:NP_056349.1:S777                    NP_056349.1_763_786_1_1_S777   \n",
       "ZZZ3:NP_056349.1:S89                        NP_056349.1_89_91_1_1_S89   \n",
       "ZZZ3:NP_056349.1:Y399                    NP_056349.1_396_434_1_1_Y399   \n",
       "ZZZ3:NP_056349.1:Y399;T428           NP_056349.1_396_434_2_2_Y399T428   \n",
       "\n",
       "                                 refseq_prot_id symbol    phosphosites  \\\n",
       "site_id                                                                  \n",
       "A1BG:NP_570602.2:S302               NP_570602.2   A1BG            S302   \n",
       "A1BG:NP_570602.2:T171               NP_570602.2   A1BG            T171   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184     NP_570602.2   A1BG  T171;Y180;Y184   \n",
       "A1CF:NP_001185747.1:T491         NP_001185747.1   A1CF            T491   \n",
       "A2M:NP_000005.2:S710                NP_000005.2    A2M            S710   \n",
       "...                                         ...    ...             ...   \n",
       "ZZZ3:NP_056349.1:S468               NP_056349.1   ZZZ3            S468   \n",
       "ZZZ3:NP_056349.1:S777               NP_056349.1   ZZZ3            S777   \n",
       "ZZZ3:NP_056349.1:S89                NP_056349.1   ZZZ3             S89   \n",
       "ZZZ3:NP_056349.1:Y399               NP_056349.1   ZZZ3            Y399   \n",
       "ZZZ3:NP_056349.1:Y399;T428          NP_056349.1   ZZZ3       Y399;T428   \n",
       "\n",
       "                                                                      peptide  \\\n",
       "site_id                                                                         \n",
       "A1BG:NP_570602.2:S302                   LHDNQNGWSGDSAPVELILsDETLPAPEFSPEPESGR   \n",
       "A1BG:NP_570602.2:T171                       EGDHEFLEVPEAQEDVEAtFPVHQPGNYSCSYR   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184            REGDHEFLEVPEAQEDVEAtFPVHQPGNySCSyR   \n",
       "A1CF:NP_001185747.1:T491                                 ITIPALASQNPAIHPFtPPK   \n",
       "A2M:NP_000005.2:S710                                              VGFYEsDVMGR   \n",
       "...                                                                       ...   \n",
       "ZZZ3:NP_056349.1:S468                                              LNIGHLPsAK   \n",
       "ZZZ3:NP_056349.1:S777                               SCFHSHMNTAVEDAsDDESIPIMYR   \n",
       "ZZZ3:NP_056349.1:S89                                                 RGLsSSEK   \n",
       "ZZZ3:NP_056349.1:Y399            NSSPyRENGQFEENNLSPNETNATVSDNVSQSPTNPGEISQNEK   \n",
       "ZZZ3:NP_056349.1:Y399;T428       NSSPyRENGQFEENNLSPNETNATVSDNVSQSPtNPGEISQNEK   \n",
       "\n",
       "                                peptide_start peptide_end  refseq_tx_id  \\\n",
       "site_id                                                                   \n",
       "A1BG:NP_570602.2:S302                     283         319     NM_130786   \n",
       "A1BG:NP_570602.2:T171                     153         185     NM_130786   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184           152         185     NM_130786   \n",
       "A1CF:NP_001185747.1:T491                  475         494  NM_001198818   \n",
       "A2M:NP_000005.2:S710                      705         715     NM_000014   \n",
       "...                                       ...         ...           ...   \n",
       "ZZZ3:NP_056349.1:S468                     461         470     NM_015534   \n",
       "ZZZ3:NP_056349.1:S777                     763         787     NM_015534   \n",
       "ZZZ3:NP_056349.1:S89                       86          93     NM_015534   \n",
       "ZZZ3:NP_056349.1:Y399                     395         438     NM_015534   \n",
       "ZZZ3:NP_056349.1:Y399;T428                395         438     NM_015534   \n",
       "\n",
       "                                    uniparc_id     hgnc_id  ... C3N-01646.T  \\\n",
       "site_id                                                     ...               \n",
       "A1BG:NP_570602.2:S302            UPI0000167B10      HGNC:5  ...         NaN   \n",
       "A1BG:NP_570602.2:T171            UPI0000167B10      HGNC:5  ...         NaN   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184  UPI0000167B10      HGNC:5  ...         NaN   \n",
       "A1CF:NP_001185747.1:T491         UPI000006EC73  HGNC:24086  ...         NaN   \n",
       "A2M:NP_000005.2:S710             UPI0000155718      HGNC:7  ...    0.888674   \n",
       "...                                        ...         ...  ...         ...   \n",
       "ZZZ3:NP_056349.1:S468            UPI0000074256  HGNC:24523  ...         NaN   \n",
       "ZZZ3:NP_056349.1:S777            UPI0000074256  HGNC:24523  ...         NaN   \n",
       "ZZZ3:NP_056349.1:S89             UPI0000074256  HGNC:24523  ...         NaN   \n",
       "ZZZ3:NP_056349.1:Y399            UPI0000074256  HGNC:24523  ...         NaN   \n",
       "ZZZ3:NP_056349.1:Y399;T428       UPI0000074256  HGNC:24523  ...         NaN   \n",
       "\n",
       "                                C3N-01646.N C3N-01648.T C3N-01648.N  \\\n",
       "site_id                                                               \n",
       "A1BG:NP_570602.2:S302                   NaN         NaN         NaN   \n",
       "A1BG:NP_570602.2:T171                   NaN         NaN         NaN   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184         NaN         NaN         NaN   \n",
       "A1CF:NP_001185747.1:T491                NaN    -0.69413    0.117396   \n",
       "A2M:NP_000005.2:S710                -0.7553         NaN         NaN   \n",
       "...                                     ...         ...         ...   \n",
       "ZZZ3:NP_056349.1:S468                   NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:S777                   NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:S89                    NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:Y399                   NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:Y399;T428              NaN         NaN         NaN   \n",
       "\n",
       "                                C3N-01649.T C3N-01649.N C3N-01651.T  \\\n",
       "site_id                                                               \n",
       "A1BG:NP_570602.2:S302                   NaN         NaN         NaN   \n",
       "A1BG:NP_570602.2:T171                   NaN         NaN         NaN   \n",
       "A1BG:NP_570602.2:T171;Y180;Y184         NaN         NaN         NaN   \n",
       "A1CF:NP_001185747.1:T491          -0.633383    0.076533    0.467653   \n",
       "A2M:NP_000005.2:S710                    NaN         NaN    0.407549   \n",
       "...                                     ...         ...         ...   \n",
       "ZZZ3:NP_056349.1:S468             -0.721719    0.372557         NaN   \n",
       "ZZZ3:NP_056349.1:S777                   NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:S89                    NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:Y399                   NaN         NaN         NaN   \n",
       "ZZZ3:NP_056349.1:Y399;T428              NaN         NaN         NaN   \n",
       "\n",
       "                                C3N-01651.N C3N-01808.T C3N-01808.N  \n",
       "site_id                                                              \n",
       "A1BG:NP_570602.2:S302                   NaN         NaN         NaN  \n",
       "A1BG:NP_570602.2:T171                   NaN         NaN         NaN  \n",
       "A1BG:NP_570602.2:T171;Y180;Y184         NaN         NaN         NaN  \n",
       "A1CF:NP_001185747.1:T491           0.480063    0.464454    0.643927  \n",
       "A2M:NP_000005.2:S710               -0.73731         NaN         NaN  \n",
       "...                                     ...         ...         ...  \n",
       "ZZZ3:NP_056349.1:S468                   NaN    0.097481    0.241512  \n",
       "ZZZ3:NP_056349.1:S777                   NaN         NaN         NaN  \n",
       "ZZZ3:NP_056349.1:S89                    NaN         NaN         NaN  \n",
       "ZZZ3:NP_056349.1:Y399                   NaN         NaN         NaN  \n",
       "ZZZ3:NP_056349.1:Y399;T428              NaN         NaN         NaN  \n",
       "\n",
       "[81780 rows x 215 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein['CCRCC']['phosphoproteome']['data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cancer_to_protein['LSCC']['phosphoproteome']['metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def filter_sites(df, metadata_df):\n",
    "    ## filter sites with excessive missing NA\n",
    "    sample_df = df[metadata_df.index]\n",
    "    usable_count = [np.count_nonzero(~pd.isnull(ls)) for ls in sample_df.values]\n",
    "#     sns.distplot(np.asarray(usable_count)/sample_df.shape[1])\n",
    "    mask = [True if c / sample_df.shape[1] >= .75 else False for c in usable_count]\n",
    "    df = df[mask]\n",
    "    sample_df = sample_df[mask]\n",
    "    \n",
    "    sample_df = sample_df.transpose()\n",
    "    \n",
    "    ## impute nans\n",
    "    sample_df = pd.DataFrame(data=SimpleImputer(strategy='median').fit_transform(sample_df.values),\n",
    "                       index=sample_df.index, columns=sample_df.columns)\n",
    "    \n",
    "    return pd.concat((df.iloc[:, :-metadata_df.shape[0]], sample_df.transpose()), axis=1)\n",
    "    \n",
    "\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_prospective acetylome\n",
      "(18328, 143) (3800, 143)\n",
      "BRCA_prospective phosphoproteome\n",
      "(63330, 143) (19502, 143)\n",
      "BRCA_prospective proteome\n",
      "(9764, 133) (8890, 133)\n",
      "CRC_prospective phosphoproteome\n",
      "(41891, 220) (5851, 220)\n",
      "CRC_prospective proteome\n",
      "(7402, 201) (5940, 201)\n",
      "OV_prospective phosphoproteome\n",
      "(48571, 126) (2422, 126)\n",
      "OV_prospective proteome\n",
      "(10095, 108) (8624, 108)\n",
      "CCRCC phosphoproteome\n",
      "(81780, 215) (15255, 215)\n",
      "CCRCC proteome\n",
      "(11355, 199) (8442, 199)\n",
      "GBM acetylome\n",
      "(18767, 130) (3388, 130)\n",
      "GBM phosphoproteome\n",
      "(56292, 130) (19501, 130)\n",
      "GBM proteome\n",
      "(11141, 113) (9799, 113)\n",
      "HNSCC phosphoproteome\n",
      "(66577, 207) (13008, 207)\n",
      "HNSCC proteome\n",
      "(11744, 189) (8845, 189)\n",
      "LSCC acetylome\n",
      "(15056, 228) (3047, 228)\n",
      "LSCC phosphoproteome\n",
      "(68544, 228) (23790, 228)\n",
      "LSCC ubiquitylome\n",
      "(25430, 168) (6827, 168)\n",
      "LSCC proteome\n",
      "(11117, 218) (9882, 218)\n",
      "LUAD acetylome\n",
      "(13368, 232) (2749, 232)\n",
      "LUAD phosphoproteome\n",
      "(64996, 232) (22344, 232)\n",
      "LUAD proteome\n",
      "(10305, 222) (8987, 222)\n",
      "UCEC acetylome\n",
      "(14184, 165) (2183, 165)\n",
      "UCEC phosphoproteome\n",
      "(105878, 165) (15982, 165)\n",
      "UCEC proteome\n",
      "(10999, 147) (9245, 147)\n",
      "BRCA_retrospective phosphoproteome\n",
      "(33239, 100) (15819, 100)\n",
      "BRCA_retrospective proteome\n",
      "(11365, 93) (9210, 93)\n",
      "OV_retrospective phosphoproteome\n",
      "(6598, 86) (667, 86)\n",
      "OV_retrospective proteome\n",
      "(3354, 175) (3354, 175)\n"
     ]
    }
   ],
   "source": [
    "for cancer, dicts in cancer_to_protein.items():\n",
    "    for dtype, d in dicts.items():\n",
    "        print(cancer, dtype)\n",
    "        d['processed_data'] = filter_sites(d['data'], d['metadata'])\n",
    "        print(d['data'].shape, d['processed_data'].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>original_id</th>\n",
       "      <th>refseq_prot_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>acetylsites</th>\n",
       "      <th>peptide</th>\n",
       "      <th>peptide_start</th>\n",
       "      <th>peptide_end</th>\n",
       "      <th>refseq_tx_id</th>\n",
       "      <th>uniparc_id</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>...</th>\n",
       "      <th>S144</th>\n",
       "      <th>S145</th>\n",
       "      <th>S146</th>\n",
       "      <th>S147</th>\n",
       "      <th>S148</th>\n",
       "      <th>S149</th>\n",
       "      <th>S150</th>\n",
       "      <th>S151</th>\n",
       "      <th>S152</th>\n",
       "      <th>S153</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>site_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAGAB:NP_078942.3:K290</th>\n",
       "      <td>VHAEK#VAK</td>\n",
       "      <td>NP_078942.3</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>K290</td>\n",
       "      <td>VHAEkVAK</td>\n",
       "      <td>286</td>\n",
       "      <td>293</td>\n",
       "      <td>NM_024666</td>\n",
       "      <td>UPI000013D219</td>\n",
       "      <td>HGNC:25662</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.428</td>\n",
       "      <td>-0.87300</td>\n",
       "      <td>-0.29500</td>\n",
       "      <td>0.15200</td>\n",
       "      <td>-0.53000</td>\n",
       "      <td>-0.75700</td>\n",
       "      <td>-0.72800</td>\n",
       "      <td>-1.16000</td>\n",
       "      <td>-0.59600</td>\n",
       "      <td>-0.22400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCE1:NP_001035809.1:K431</th>\n",
       "      <td>QLLHEK#IR</td>\n",
       "      <td>NP_001035809.1</td>\n",
       "      <td>ABCE1</td>\n",
       "      <td>K431</td>\n",
       "      <td>QLLHEkIR</td>\n",
       "      <td>426</td>\n",
       "      <td>433</td>\n",
       "      <td>NM_001040876</td>\n",
       "      <td>UPI0000001226</td>\n",
       "      <td>HGNC:69</td>\n",
       "      <td>...</td>\n",
       "      <td>0.149</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.04345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAA2:NP_006102.2:K13</th>\n",
       "      <td>GVFVVAAK#R</td>\n",
       "      <td>NP_006102.2</td>\n",
       "      <td>ACAA2</td>\n",
       "      <td>K13</td>\n",
       "      <td>GVFVVAAkR</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>NM_006111</td>\n",
       "      <td>UPI000006FECE</td>\n",
       "      <td>HGNC:83</td>\n",
       "      <td>...</td>\n",
       "      <td>0.779</td>\n",
       "      <td>0.23200</td>\n",
       "      <td>0.08980</td>\n",
       "      <td>0.14800</td>\n",
       "      <td>0.19800</td>\n",
       "      <td>0.06040</td>\n",
       "      <td>-0.04920</td>\n",
       "      <td>0.26900</td>\n",
       "      <td>0.02100</td>\n",
       "      <td>0.05590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAA2:NP_006102.2:K137</th>\n",
       "      <td>FGTK#LGSDIK</td>\n",
       "      <td>NP_006102.2</td>\n",
       "      <td>ACAA2</td>\n",
       "      <td>K137</td>\n",
       "      <td>FGTkLGSDIK</td>\n",
       "      <td>134</td>\n",
       "      <td>143</td>\n",
       "      <td>NM_006111</td>\n",
       "      <td>UPI000006FECE</td>\n",
       "      <td>HGNC:83</td>\n",
       "      <td>...</td>\n",
       "      <td>0.257</td>\n",
       "      <td>0.37200</td>\n",
       "      <td>0.01440</td>\n",
       "      <td>0.18900</td>\n",
       "      <td>0.17800</td>\n",
       "      <td>0.20600</td>\n",
       "      <td>-0.13800</td>\n",
       "      <td>0.41600</td>\n",
       "      <td>0.28200</td>\n",
       "      <td>-0.22200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAA2:NP_006102.2:K234</th>\n",
       "      <td>QTMQVDEHARPQTTLEQLQK#LPPVFK</td>\n",
       "      <td>NP_006102.2</td>\n",
       "      <td>ACAA2</td>\n",
       "      <td>K234</td>\n",
       "      <td>QTMQVDEHARPQTTLEQLQkLPPVFK</td>\n",
       "      <td>215</td>\n",
       "      <td>240</td>\n",
       "      <td>NM_006111</td>\n",
       "      <td>UPI000006FECE</td>\n",
       "      <td>HGNC:83</td>\n",
       "      <td>...</td>\n",
       "      <td>1.080</td>\n",
       "      <td>0.21800</td>\n",
       "      <td>0.00190</td>\n",
       "      <td>-0.15900</td>\n",
       "      <td>0.06660</td>\n",
       "      <td>-0.55800</td>\n",
       "      <td>-0.22000</td>\n",
       "      <td>-0.23800</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>-0.59600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF703:NP_079345.1:K141</th>\n",
       "      <td>SAPGAASAAAALK#QLGDSPAEDK</td>\n",
       "      <td>NP_079345.1</td>\n",
       "      <td>ZNF703</td>\n",
       "      <td>K141</td>\n",
       "      <td>SAPGAASAAAALkQLGDSPAEDK</td>\n",
       "      <td>129</td>\n",
       "      <td>151</td>\n",
       "      <td>NM_025069</td>\n",
       "      <td>UPI0000073D30</td>\n",
       "      <td>HGNC:25883</td>\n",
       "      <td>...</td>\n",
       "      <td>0.194</td>\n",
       "      <td>0.71000</td>\n",
       "      <td>0.55000</td>\n",
       "      <td>0.25200</td>\n",
       "      <td>0.67200</td>\n",
       "      <td>0.10800</td>\n",
       "      <td>0.02580</td>\n",
       "      <td>0.06670</td>\n",
       "      <td>0.29800</td>\n",
       "      <td>-0.00305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZRANB2:NP_976225.1:K54</th>\n",
       "      <td>AGGTEIGK#TLAEK</td>\n",
       "      <td>NP_976225.1</td>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>K54</td>\n",
       "      <td>AGGTEIGkTLAEK</td>\n",
       "      <td>47</td>\n",
       "      <td>59</td>\n",
       "      <td>NM_203350</td>\n",
       "      <td>UPI000013CE63</td>\n",
       "      <td>HGNC:13058</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.604</td>\n",
       "      <td>0.61200</td>\n",
       "      <td>0.44600</td>\n",
       "      <td>0.52400</td>\n",
       "      <td>0.64700</td>\n",
       "      <td>0.06380</td>\n",
       "      <td>0.11200</td>\n",
       "      <td>0.63000</td>\n",
       "      <td>0.69100</td>\n",
       "      <td>0.16700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN18:NP_001139014.1:K422</th>\n",
       "      <td>GTAK#LGTK</td>\n",
       "      <td>NP_001139014.1</td>\n",
       "      <td>ZSCAN18</td>\n",
       "      <td>K422</td>\n",
       "      <td>GTAkLGTK</td>\n",
       "      <td>419</td>\n",
       "      <td>426</td>\n",
       "      <td>NM_001145542</td>\n",
       "      <td>UPI00017A6DAF</td>\n",
       "      <td>HGNC:21037</td>\n",
       "      <td>...</td>\n",
       "      <td>0.290</td>\n",
       "      <td>-0.38100</td>\n",
       "      <td>0.35200</td>\n",
       "      <td>-0.17200</td>\n",
       "      <td>-0.22400</td>\n",
       "      <td>-0.15000</td>\n",
       "      <td>-0.31400</td>\n",
       "      <td>-0.67600</td>\n",
       "      <td>-0.37900</td>\n",
       "      <td>0.04730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX:NP_001010972.1:K25</th>\n",
       "      <td>PSPAISVSVSAPAFYAPQKK#</td>\n",
       "      <td>NP_001010972.1</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>K25</td>\n",
       "      <td>PSPAISVSVSAPAFYAPQKk</td>\n",
       "      <td>6</td>\n",
       "      <td>25</td>\n",
       "      <td>NM_001010972</td>\n",
       "      <td>UPI00000424F2</td>\n",
       "      <td>HGNC:13200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.445</td>\n",
       "      <td>0.82000</td>\n",
       "      <td>0.89600</td>\n",
       "      <td>0.46200</td>\n",
       "      <td>0.78300</td>\n",
       "      <td>0.00951</td>\n",
       "      <td>0.28700</td>\n",
       "      <td>1.06000</td>\n",
       "      <td>1.10000</td>\n",
       "      <td>0.56200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX:NP_001010972.1:K279</th>\n",
       "      <td>FTPVASK#FSPGAPGGSGSQPNQK</td>\n",
       "      <td>NP_001010972.1</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>K279</td>\n",
       "      <td>FTPVASkFSPGAPGGSGSQPNQK</td>\n",
       "      <td>273</td>\n",
       "      <td>295</td>\n",
       "      <td>NM_001010972</td>\n",
       "      <td>UPI00000424F2</td>\n",
       "      <td>HGNC:13200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.935</td>\n",
       "      <td>0.42800</td>\n",
       "      <td>0.16800</td>\n",
       "      <td>0.13900</td>\n",
       "      <td>0.16500</td>\n",
       "      <td>-0.35400</td>\n",
       "      <td>-0.02160</td>\n",
       "      <td>0.63300</td>\n",
       "      <td>0.35100</td>\n",
       "      <td>0.17200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2183 rows × 165 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             original_id  refseq_prot_id  \\\n",
       "site_id                                                                    \n",
       "AAGAB:NP_078942.3:K290                         VHAEK#VAK     NP_078942.3   \n",
       "ABCE1:NP_001035809.1:K431                      QLLHEK#IR  NP_001035809.1   \n",
       "ACAA2:NP_006102.2:K13                         GVFVVAAK#R     NP_006102.2   \n",
       "ACAA2:NP_006102.2:K137                       FGTK#LGSDIK     NP_006102.2   \n",
       "ACAA2:NP_006102.2:K234       QTMQVDEHARPQTTLEQLQK#LPPVFK     NP_006102.2   \n",
       "...                                                  ...             ...   \n",
       "ZNF703:NP_079345.1:K141         SAPGAASAAAALK#QLGDSPAEDK     NP_079345.1   \n",
       "ZRANB2:NP_976225.1:K54                    AGGTEIGK#TLAEK     NP_976225.1   \n",
       "ZSCAN18:NP_001139014.1:K422                    GTAK#LGTK  NP_001139014.1   \n",
       "ZYX:NP_001010972.1:K25             PSPAISVSVSAPAFYAPQKK#  NP_001010972.1   \n",
       "ZYX:NP_001010972.1:K279         FTPVASK#FSPGAPGGSGSQPNQK  NP_001010972.1   \n",
       "\n",
       "                              symbol acetylsites                     peptide  \\\n",
       "site_id                                                                        \n",
       "AAGAB:NP_078942.3:K290         AAGAB        K290                    VHAEkVAK   \n",
       "ABCE1:NP_001035809.1:K431      ABCE1        K431                    QLLHEkIR   \n",
       "ACAA2:NP_006102.2:K13          ACAA2         K13                   GVFVVAAkR   \n",
       "ACAA2:NP_006102.2:K137         ACAA2        K137                  FGTkLGSDIK   \n",
       "ACAA2:NP_006102.2:K234         ACAA2        K234  QTMQVDEHARPQTTLEQLQkLPPVFK   \n",
       "...                              ...         ...                         ...   \n",
       "ZNF703:NP_079345.1:K141       ZNF703        K141     SAPGAASAAAALkQLGDSPAEDK   \n",
       "ZRANB2:NP_976225.1:K54        ZRANB2         K54               AGGTEIGkTLAEK   \n",
       "ZSCAN18:NP_001139014.1:K422  ZSCAN18        K422                    GTAkLGTK   \n",
       "ZYX:NP_001010972.1:K25           ZYX         K25        PSPAISVSVSAPAFYAPQKk   \n",
       "ZYX:NP_001010972.1:K279          ZYX        K279     FTPVASkFSPGAPGGSGSQPNQK   \n",
       "\n",
       "                            peptide_start peptide_end  refseq_tx_id  \\\n",
       "site_id                                                               \n",
       "AAGAB:NP_078942.3:K290                286         293     NM_024666   \n",
       "ABCE1:NP_001035809.1:K431             426         433  NM_001040876   \n",
       "ACAA2:NP_006102.2:K13                   6          14     NM_006111   \n",
       "ACAA2:NP_006102.2:K137                134         143     NM_006111   \n",
       "ACAA2:NP_006102.2:K234                215         240     NM_006111   \n",
       "...                                   ...         ...           ...   \n",
       "ZNF703:NP_079345.1:K141               129         151     NM_025069   \n",
       "ZRANB2:NP_976225.1:K54                 47          59     NM_203350   \n",
       "ZSCAN18:NP_001139014.1:K422           419         426  NM_001145542   \n",
       "ZYX:NP_001010972.1:K25                  6          25  NM_001010972   \n",
       "ZYX:NP_001010972.1:K279               273         295  NM_001010972   \n",
       "\n",
       "                                uniparc_id     hgnc_id  ...   S144     S145  \\\n",
       "site_id                                                 ...                   \n",
       "AAGAB:NP_078942.3:K290       UPI000013D219  HGNC:25662  ... -0.428 -0.87300   \n",
       "ABCE1:NP_001035809.1:K431    UPI0000001226     HGNC:69  ...  0.149 -0.04345   \n",
       "ACAA2:NP_006102.2:K13        UPI000006FECE     HGNC:83  ...  0.779  0.23200   \n",
       "ACAA2:NP_006102.2:K137       UPI000006FECE     HGNC:83  ...  0.257  0.37200   \n",
       "ACAA2:NP_006102.2:K234       UPI000006FECE     HGNC:83  ...  1.080  0.21800   \n",
       "...                                    ...         ...  ...    ...      ...   \n",
       "ZNF703:NP_079345.1:K141      UPI0000073D30  HGNC:25883  ...  0.194  0.71000   \n",
       "ZRANB2:NP_976225.1:K54       UPI000013CE63  HGNC:13058  ... -0.604  0.61200   \n",
       "ZSCAN18:NP_001139014.1:K422  UPI00017A6DAF  HGNC:21037  ...  0.290 -0.38100   \n",
       "ZYX:NP_001010972.1:K25       UPI00000424F2  HGNC:13200  ...  0.445  0.82000   \n",
       "ZYX:NP_001010972.1:K279      UPI00000424F2  HGNC:13200  ...  0.935  0.42800   \n",
       "\n",
       "                                S146     S147     S148     S149     S150  \\\n",
       "site_id                                                                    \n",
       "AAGAB:NP_078942.3:K290      -0.29500  0.15200 -0.53000 -0.75700 -0.72800   \n",
       "ABCE1:NP_001035809.1:K431   -0.04345 -0.04345 -0.04345 -0.04345 -0.04345   \n",
       "ACAA2:NP_006102.2:K13        0.08980  0.14800  0.19800  0.06040 -0.04920   \n",
       "ACAA2:NP_006102.2:K137       0.01440  0.18900  0.17800  0.20600 -0.13800   \n",
       "ACAA2:NP_006102.2:K234       0.00190 -0.15900  0.06660 -0.55800 -0.22000   \n",
       "...                              ...      ...      ...      ...      ...   \n",
       "ZNF703:NP_079345.1:K141      0.55000  0.25200  0.67200  0.10800  0.02580   \n",
       "ZRANB2:NP_976225.1:K54       0.44600  0.52400  0.64700  0.06380  0.11200   \n",
       "ZSCAN18:NP_001139014.1:K422  0.35200 -0.17200 -0.22400 -0.15000 -0.31400   \n",
       "ZYX:NP_001010972.1:K25       0.89600  0.46200  0.78300  0.00951  0.28700   \n",
       "ZYX:NP_001010972.1:K279      0.16800  0.13900  0.16500 -0.35400 -0.02160   \n",
       "\n",
       "                                S151     S152     S153  \n",
       "site_id                                                 \n",
       "AAGAB:NP_078942.3:K290      -1.16000 -0.59600 -0.22400  \n",
       "ABCE1:NP_001035809.1:K431   -0.04345 -0.04345 -0.04345  \n",
       "ACAA2:NP_006102.2:K13        0.26900  0.02100  0.05590  \n",
       "ACAA2:NP_006102.2:K137       0.41600  0.28200 -0.22200  \n",
       "ACAA2:NP_006102.2:K234      -0.23800 -0.03560 -0.59600  \n",
       "...                              ...      ...      ...  \n",
       "ZNF703:NP_079345.1:K141      0.06670  0.29800 -0.00305  \n",
       "ZRANB2:NP_976225.1:K54       0.63000  0.69100  0.16700  \n",
       "ZSCAN18:NP_001139014.1:K422 -0.67600 -0.37900  0.04730  \n",
       "ZYX:NP_001010972.1:K25       1.06000  1.10000  0.56200  \n",
       "ZYX:NP_001010972.1:K279      0.63300  0.35100  0.17200  \n",
       "\n",
       "[2183 rows x 165 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein['UCEC']['acetylome']['processed_data']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "849"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "len(Counter([g.split(':')[0] for g in cancer_to_protein['UCEC']['acetylome']['processed_data'].index]\n",
    "       ).most_common())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4095"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Counter([g.split(':')[0] for g in cancer_to_protein['UCEC']['phosphoproteome']['processed_data'].index]\n",
    "       ).most_common())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>aliquot_id</th>\n",
       "      <th>tmt_experiment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>CPT0001540009</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.N</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>CPT0001550001</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010.T</th>\n",
       "      <td>C3L-00010</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>CPT0001220008</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010.N</th>\n",
       "      <td>C3L-00010</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>CPT0001230001</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011.T</th>\n",
       "      <td>C3L-00011</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>CPT0001340003</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>C3N-01649</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>CPT0088640003</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.T</th>\n",
       "      <td>C3N-01651</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>CPT0088690003</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>C3N-01651</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>CPT0088710001</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.T</th>\n",
       "      <td>C3N-01808</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>CPT0089460004</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>C3N-01808</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>CPT0089480003</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               case_id          sample_type     aliquot_id tmt_experiment\n",
       "C3L-00004.T  C3L-00004        Primary Tumor  CPT0001540009              6\n",
       "C3L-00004.N  C3L-00004  Solid Tissue Normal  CPT0001550001              6\n",
       "C3L-00010.T  C3L-00010        Primary Tumor  CPT0001220008              3\n",
       "C3L-00010.N  C3L-00010  Solid Tissue Normal  CPT0001230001              3\n",
       "C3L-00011.T  C3L-00011        Primary Tumor  CPT0001340003             11\n",
       "...                ...                  ...            ...            ...\n",
       "C3N-01649.N  C3N-01649  Solid Tissue Normal  CPT0088640003             12\n",
       "C3N-01651.T  C3N-01651        Primary Tumor  CPT0088690003             14\n",
       "C3N-01651.N  C3N-01651  Solid Tissue Normal  CPT0088710001             14\n",
       "C3N-01808.T  C3N-01808        Primary Tumor  CPT0089460004             12\n",
       "C3N-01808.N  C3N-01808  Solid Tissue Normal  CPT0089480003             12\n",
       "\n",
       "[194 rows x 4 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein['CCRCC']['phosphoproteome']['metadata'].sort_values('case_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>aliquot_ids</th>\n",
       "      <th>tmt_experiment</th>\n",
       "      <th>tmt_channel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>5</td>\n",
       "      <td>128N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S105</th>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>CPT0001470001</td>\n",
       "      <td>5</td>\n",
       "      <td>128C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>16</td>\n",
       "      <td>130N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>2</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>12</td>\n",
       "      <td>129C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S140</th>\n",
       "      <td>NX5</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>CPT0189520002;CPT0189520003;CPT0189520004;CPT0...</td>\n",
       "      <td>15</td>\n",
       "      <td>127C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S141</th>\n",
       "      <td>NX6</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>CPT0189170002;CPT0189170003;CPT0189180002;CPT0...</td>\n",
       "      <td>15</td>\n",
       "      <td>128N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S142</th>\n",
       "      <td>NX7</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>CPT0183020002;CPT0183050002;CPT0183050003;CPT0...</td>\n",
       "      <td>15</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S143</th>\n",
       "      <td>NX8</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>CPT0191190002;CPT0191190003;CPT0191200002;CPT0...</td>\n",
       "      <td>16</td>\n",
       "      <td>127C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S144</th>\n",
       "      <td>NX9</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>CPT0191210002;CPT0191210003;CPT0191210004</td>\n",
       "      <td>16</td>\n",
       "      <td>128N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        case_id      sample_type  \\\n",
       "S001  C3L-00006            Tumor   \n",
       "S105  C3L-00006  Adjacent_normal   \n",
       "S002  C3L-00008            Tumor   \n",
       "S003  C3L-00032            Tumor   \n",
       "S005  C3L-00090            Tumor   \n",
       "...         ...              ...   \n",
       "S140        NX5  Enriched_normal   \n",
       "S141        NX6  Enriched_normal   \n",
       "S142        NX7  Enriched_normal   \n",
       "S143        NX8  Enriched_normal   \n",
       "S144        NX9  Enriched_normal   \n",
       "\n",
       "                                            aliquot_ids tmt_experiment  \\\n",
       "S001                                      CPT0001460012              5   \n",
       "S105                                      CPT0001470001              5   \n",
       "S002                                      CPT0001300009             16   \n",
       "S003                                      CPT0001420009              2   \n",
       "S005                                      CPT0001140003             12   \n",
       "...                                                 ...            ...   \n",
       "S140  CPT0189520002;CPT0189520003;CPT0189520004;CPT0...             15   \n",
       "S141  CPT0189170002;CPT0189170003;CPT0189180002;CPT0...             15   \n",
       "S142  CPT0183020002;CPT0183050002;CPT0183050003;CPT0...             15   \n",
       "S143  CPT0191190002;CPT0191190003;CPT0191200002;CPT0...             16   \n",
       "S144          CPT0191210002;CPT0191210003;CPT0191210004             16   \n",
       "\n",
       "     tmt_channel  \n",
       "S001        128N  \n",
       "S105        128C  \n",
       "S002        130N  \n",
       "S003         131  \n",
       "S005        129C  \n",
       "...          ...  \n",
       "S140        127C  \n",
       "S141        128N  \n",
       "S142         131  \n",
       "S143        127C  \n",
       "S144        128N  \n",
       "\n",
       "[144 rows x 5 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein['UCEC']['phosphoproteome']['metadata'].sort_values('case_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>aliquot_id</th>\n",
       "      <th>tmt_experiment</th>\n",
       "      <th>tmt_channel</th>\n",
       "      <th>sample_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00081.T</th>\n",
       "      <td>C3L-00081</td>\n",
       "      <td>CPT0001020003</td>\n",
       "      <td>3</td>\n",
       "      <td>128C</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00081.N</th>\n",
       "      <td>C3L-00081</td>\n",
       "      <td>CPT0001030003</td>\n",
       "      <td>3</td>\n",
       "      <td>129N</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00415.T</th>\n",
       "      <td>C3L-00415</td>\n",
       "      <td>CPT0010050003</td>\n",
       "      <td>16</td>\n",
       "      <td>130C</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00415.N</th>\n",
       "      <td>C3L-00415</td>\n",
       "      <td>CPT0010060003</td>\n",
       "      <td>16</td>\n",
       "      <td>131N</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00445.N</th>\n",
       "      <td>C3L-00445</td>\n",
       "      <td>CPT0130440004</td>\n",
       "      <td>15</td>\n",
       "      <td>130N</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04127.T</th>\n",
       "      <td>C3N-04127</td>\n",
       "      <td>CPT0238110004</td>\n",
       "      <td>10</td>\n",
       "      <td>128C</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04155.N</th>\n",
       "      <td>C3N-04155</td>\n",
       "      <td>CPT0223820003</td>\n",
       "      <td>11</td>\n",
       "      <td>130N</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04155.T</th>\n",
       "      <td>C3N-04155</td>\n",
       "      <td>CPT0223790003</td>\n",
       "      <td>11</td>\n",
       "      <td>129C</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04162.N</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>CPT0222740003</td>\n",
       "      <td>2</td>\n",
       "      <td>128N</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04162.T</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>CPT0222730003</td>\n",
       "      <td>2</td>\n",
       "      <td>127C</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>207 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               case_id     aliquot_id tmt_experiment tmt_channel sample_type\n",
       "C3L-00081.T  C3L-00081  CPT0001020003              3        128C       Tumor\n",
       "C3L-00081.N  C3L-00081  CPT0001030003              3        129N      Normal\n",
       "C3L-00415.T  C3L-00415  CPT0010050003             16        130C       Tumor\n",
       "C3L-00415.N  C3L-00415  CPT0010060003             16        131N      Normal\n",
       "C3L-00445.N  C3L-00445  CPT0130440004             15        130N      Normal\n",
       "...                ...            ...            ...         ...         ...\n",
       "C3N-04127.T  C3N-04127  CPT0238110004             10        128C       Tumor\n",
       "C3N-04155.N  C3N-04155  CPT0223820003             11        130N      Normal\n",
       "C3N-04155.T  C3N-04155  CPT0223790003             11        129C       Tumor\n",
       "C3N-04162.N  C3N-04162  CPT0222740003              2        128N      Normal\n",
       "C3N-04162.T  C3N-04162  CPT0222730003              2        127C       Tumor\n",
       "\n",
       "[207 rows x 5 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein['LSCC']['phosphoproteome']['metadata'].sort_values('case_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_prospective\n",
      "BRCA_prospective {'Primary Tumor'}\n",
      "CRC_prospective\n",
      "CRC_prospective {'Primary Tumor', 'Solid Tissue Normal'}\n",
      "OV_prospective\n",
      "OV_prospective {'Primary Tumor', 'Solid Tissue Normal'}\n",
      "CCRCC\n",
      "CCRCC {'Primary Tumor', 'Solid Tissue Normal'}\n",
      "GBM\n",
      "HNSCC\n",
      "HNSCC {'Primary Tumor', 'Solid Tissue Normal'}\n",
      "LSCC\n",
      "LSCC {'Normal', 'Tumor'}\n",
      "LUAD\n",
      "LUAD {'Normal', 'Tumor'}\n",
      "UCEC\n",
      "UCEC {'Adjacent_normal', 'Enriched_normal', 'Myometrium_normal', 'Tumor'}\n",
      "BRCA_retrospective\n",
      "OV_retrospective\n"
     ]
    }
   ],
   "source": [
    "for cancer, d in cancer_to_protein.items():\n",
    "    print(cancer)\n",
    "#     print(d['phosphoproteome']['metadata'].columns)\n",
    "    if 'sample_type' in d['phosphoproteome']['metadata']:\n",
    "        print(cancer, set(d['phosphoproteome']['metadata']['sample_type']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('C3L-00006', 2),\n",
       " ('C3L-00361', 2),\n",
       " ('C3L-00563', 2),\n",
       " ('C3L-00586', 2),\n",
       " ('C3L-00601', 2),\n",
       " ('C3L-00605', 2),\n",
       " ('C3L-00769', 2),\n",
       " ('C3L-00770', 2),\n",
       " ('C3L-00771', 2),\n",
       " ('C3L-00932', 2),\n",
       " ('C3L-00947', 2),\n",
       " ('C3L-00963', 2),\n",
       " ('C3L-01246', 2),\n",
       " ('C3L-01249', 2),\n",
       " ('C3L-01252', 2),\n",
       " ('C3L-01256', 2),\n",
       " ('C3L-01257', 2),\n",
       " ('C3L-01282', 2),\n",
       " ('C3L-01304', 2),\n",
       " ('C3L-01307', 2),\n",
       " ('C3L-01311', 2),\n",
       " ('C3L-01744', 2),\n",
       " ('C3N-00200', 2),\n",
       " ('C3N-00333', 2),\n",
       " ('C3N-00383', 2),\n",
       " ('C3N-00729', 2),\n",
       " ('C3N-00858', 2),\n",
       " ('C3N-00866', 2),\n",
       " ('C3N-01211', 2),\n",
       " ('C3N-01346', 2),\n",
       " ('C3L-00008', 1),\n",
       " ('C3L-00032', 1),\n",
       " ('C3L-00090', 1),\n",
       " ('C3L-00098', 1),\n",
       " ('C3L-00136', 1),\n",
       " ('C3L-00137', 1),\n",
       " ('C3L-00139', 1),\n",
       " ('C3L-00143', 1),\n",
       " ('C3L-00145', 1),\n",
       " ('C3L-00156', 1),\n",
       " ('C3L-00161', 1),\n",
       " ('C3L-00358', 1),\n",
       " ('C3L-00362', 1),\n",
       " ('C3L-00413', 1),\n",
       " ('C3L-00449', 1),\n",
       " ('C3L-00767', 1),\n",
       " ('C3L-00780', 1),\n",
       " ('C3L-00781', 1),\n",
       " ('C3L-00905', 1),\n",
       " ('C3L-00918', 1),\n",
       " ('C3L-00921', 1),\n",
       " ('C3L-00942', 1),\n",
       " ('C3L-00946', 1),\n",
       " ('C3L-00949', 1),\n",
       " ('C3L-00961', 1),\n",
       " ('C3L-01248', 1),\n",
       " ('C3L-01275', 1),\n",
       " ('C3L-01312', 1),\n",
       " ('C3L-01925', 1),\n",
       " ('C3N-00151', 1),\n",
       " ('C3N-00321', 1),\n",
       " ('C3N-00322', 1),\n",
       " ('C3N-00323', 1),\n",
       " ('C3N-00324', 1),\n",
       " ('C3N-00326', 1),\n",
       " ('C3N-00328', 1),\n",
       " ('C3N-00334', 1),\n",
       " ('C3N-00335', 1),\n",
       " ('C3N-00337', 1),\n",
       " ('C3N-00339', 1),\n",
       " ('C3N-00340', 1),\n",
       " ('C3N-00377', 1),\n",
       " ('C3N-00379', 1),\n",
       " ('C3N-00386', 1),\n",
       " ('C3N-00388', 1),\n",
       " ('C3N-00389', 1),\n",
       " ('C3N-00734', 1),\n",
       " ('C3N-00743', 1),\n",
       " ('C3N-00836', 1),\n",
       " ('C3N-00847', 1),\n",
       " ('C3N-00848', 1),\n",
       " ('C3N-00850', 1),\n",
       " ('C3N-00880', 1),\n",
       " ('C3N-01003', 1),\n",
       " ('C3N-01212', 1),\n",
       " ('C3N-01217', 1),\n",
       " ('C3N-01219', 1),\n",
       " ('C3N-01267', 1),\n",
       " ('C3N-01349', 1),\n",
       " ('C3N-01510', 1),\n",
       " ('C3N-01520', 1),\n",
       " ('C3N-01521', 1),\n",
       " ('C3N-01537', 1),\n",
       " ('C3N-01802', 1),\n",
       " ('C3N-01825', 1),\n",
       " ('C3L-00930', 1),\n",
       " ('NX1', 1),\n",
       " ('NX2', 1),\n",
       " ('NX3', 1),\n",
       " ('NX4', 1),\n",
       " ('NX5', 1),\n",
       " ('NX6', 1),\n",
       " ('NX7', 1),\n",
       " ('NX8', 1),\n",
       " ('NX9', 1),\n",
       " ('NX10', 1),\n",
       " ('NX11', 1),\n",
       " ('NX12', 1),\n",
       " ('NX13', 1),\n",
       " ('NX14', 1),\n",
       " ('NX15', 1),\n",
       " ('NX16', 1),\n",
       " ('NX17', 1),\n",
       " ('NX18', 1)]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "Counter(cancer_to_protein['UCEC']['acetylome']['metadata']['case_id']).most_common()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "## process for final output\n",
    "def generate_model_input(data_dict):\n",
    "    output = None\n",
    "    \n",
    "    for dtype, df_dict in data_dict.items():\n",
    "        print(dtype)\n",
    "        filtered = df_dict['processed_data'].copy()\n",
    "#         filtered = df_dict['data'].copy()\n",
    "        metadata = df_dict['metadata'].copy()\n",
    "        filtered = filtered[metadata.index]\n",
    "        filtered = filtered.transpose()\n",
    "\n",
    "        ## add in metadata\n",
    "        if 'sample_type' in metadata.columns:\n",
    "            filtered['sample_type'] = ['normal' if 'tumor' not in x.lower() else 'tumor'\n",
    "                                       for x in metadata['sample_type']]\n",
    "        else:\n",
    "            filtered['sample_type'] = ['tumor'] * metadata.shape[0]\n",
    "            \n",
    "        ## rename_columns\n",
    "        filtered.columns = [f'{c}_{dtype}' for c in filtered.columns]\n",
    "#         filtered = filtered.transpose()\n",
    "        \n",
    "        \n",
    "        if output is None:\n",
    "            output = filtered\n",
    "        else:\n",
    "            print(output.shape, filtered.shape, len(set(filtered.index).intersection(set(output.index))))\n",
    "            output = pd.merge(output, filtered, right_index=True, left_index=True)\n",
    "            \n",
    "    ## normalize sample types\n",
    "    df = output[[c for c in output.columns if 'sample_type' in c]]\n",
    "    output = output[[c for c in output.columns if f'sample_type' not in c]]\n",
    "    output['sample_type'] = df.iloc[:, 0].to_list()\n",
    "    \n",
    "    ## split into gene groupings\n",
    "    ## groupby tissue\n",
    "#     output = output.groupby('sample_type').mean()\n",
    "#     output = output[[c for c in output.columns if '_proteome' in c]]\n",
    "#     output = output.transpose()\n",
    "\n",
    "    return output\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA_prospective\n",
      "acetylome\n",
      "phosphoproteome\n",
      "(122, 3801) (122, 19503) 122\n",
      "proteome\n",
      "(122, 23304) (122, 8891) 122\n",
      "CRC_prospective\n",
      "phosphoproteome\n",
      "proteome\n",
      "(197, 5852) (197, 5941) 197\n",
      "OV_prospective\n",
      "phosphoproteome\n",
      "proteome\n",
      "(103, 2423) (103, 8625) 103\n",
      "CCRCC\n",
      "phosphoproteome\n",
      "proteome\n",
      "(194, 15256) (194, 8443) 194\n",
      "GBM\n",
      "acetylome\n",
      "phosphoproteome\n",
      "(109, 3389) (109, 19502) 109\n",
      "proteome\n",
      "(109, 22891) (109, 9800) 109\n",
      "HNSCC\n",
      "phosphoproteome\n",
      "proteome\n",
      "(186, 13009) (186, 8846) 186\n",
      "LSCC\n",
      "acetylome\n",
      "phosphoproteome\n",
      "(207, 3048) (207, 23791) 207\n",
      "ubiquitylome\n",
      "(207, 26839) (147, 6828) 143\n",
      "proteome\n",
      "(143, 33667) (207, 9883) 143\n",
      "LUAD\n",
      "acetylome\n",
      "phosphoproteome\n",
      "(211, 2750) (211, 22345) 211\n",
      "proteome\n",
      "(211, 25095) (211, 8988) 211\n",
      "UCEC\n",
      "acetylome\n",
      "phosphoproteome\n",
      "(144, 2184) (144, 15983) 144\n",
      "proteome\n",
      "(144, 18167) (144, 9246) 144\n",
      "BRCA_retrospective\n",
      "phosphoproteome\n",
      "proteome\n",
      "(83, 15820) (83, 9211) 83\n",
      "OV_retrospective\n",
      "phosphoproteome\n",
      "proteome\n",
      "(69, 668) (169, 3355) 67\n"
     ]
    }
   ],
   "source": [
    "cancer_to_model_inputs = {}\n",
    "df = None\n",
    "for cancer, data_dict in cancer_to_protein.items():\n",
    "    print(cancer)\n",
    "    output = generate_model_input(data_dict)\n",
    "    cancer_to_model_inputs[cancer] = output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AAGAB:NP_078942.3:K290_acetylome</th>\n",
       "      <th>ABCE1:NP_001035809.1:K431_acetylome</th>\n",
       "      <th>ACAA2:NP_006102.2:K13_acetylome</th>\n",
       "      <th>ACAA2:NP_006102.2:K137_acetylome</th>\n",
       "      <th>ACAA2:NP_006102.2:K234_acetylome</th>\n",
       "      <th>ACADM:NP_001272972.1:K312_acetylome</th>\n",
       "      <th>ACADVL:NP_001257376.1:K262_acetylome</th>\n",
       "      <th>ACADVL:NP_001257376.1:K299_acetylome</th>\n",
       "      <th>ACADVL:NP_001257376.1:K301_acetylome</th>\n",
       "      <th>ACAT1:NP_000010.1:K124_acetylome</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteome</th>\n",
       "      <th>ZW10_proteome</th>\n",
       "      <th>ZWILCH_proteome</th>\n",
       "      <th>ZWINT_proteome</th>\n",
       "      <th>ZXDC_proteome</th>\n",
       "      <th>ZYG11B_proteome</th>\n",
       "      <th>ZYX_proteome</th>\n",
       "      <th>ZZEF1_proteome</th>\n",
       "      <th>ZZZ3_proteome</th>\n",
       "      <th>sample_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>0.46100</td>\n",
       "      <td>-0.22700</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-1.030</td>\n",
       "      <td>-1.0400</td>\n",
       "      <td>2.41000</td>\n",
       "      <td>1.070</td>\n",
       "      <td>1.930</td>\n",
       "      <td>1.670</td>\n",
       "      <td>0.5430</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>-0.000315</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.08590</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>1.77000</td>\n",
       "      <td>0.56400</td>\n",
       "      <td>-1.0300</td>\n",
       "      <td>-0.906</td>\n",
       "      <td>-0.6040</td>\n",
       "      <td>-0.55700</td>\n",
       "      <td>-0.329</td>\n",
       "      <td>0.360</td>\n",
       "      <td>-0.586</td>\n",
       "      <td>0.1670</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737000</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.47300</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>-0.81500</td>\n",
       "      <td>0.10500</td>\n",
       "      <td>0.0573</td>\n",
       "      <td>0.719</td>\n",
       "      <td>0.6040</td>\n",
       "      <td>0.65100</td>\n",
       "      <td>1.160</td>\n",
       "      <td>1.100</td>\n",
       "      <td>0.902</td>\n",
       "      <td>-0.6980</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>-0.000315</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.41900</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>-0.00334</td>\n",
       "      <td>-0.04860</td>\n",
       "      <td>-0.7900</td>\n",
       "      <td>-0.949</td>\n",
       "      <td>-0.3210</td>\n",
       "      <td>0.70400</td>\n",
       "      <td>0.934</td>\n",
       "      <td>1.250</td>\n",
       "      <td>1.190</td>\n",
       "      <td>-0.1600</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156000</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.46700</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>0.20500</td>\n",
       "      <td>0.48900</td>\n",
       "      <td>0.6510</td>\n",
       "      <td>0.434</td>\n",
       "      <td>0.0320</td>\n",
       "      <td>-0.45000</td>\n",
       "      <td>0.111</td>\n",
       "      <td>0.247</td>\n",
       "      <td>-0.441</td>\n",
       "      <td>-0.0157</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.937000</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.05130</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S149</th>\n",
       "      <td>-0.75700</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>0.0604</td>\n",
       "      <td>0.206</td>\n",
       "      <td>-0.5580</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>-0.505</td>\n",
       "      <td>-0.715</td>\n",
       "      <td>-0.804</td>\n",
       "      <td>-0.0119</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.20400</td>\n",
       "      <td>-0.1510</td>\n",
       "      <td>0.0552</td>\n",
       "      <td>0.105000</td>\n",
       "      <td>1.330</td>\n",
       "      <td>-0.67400</td>\n",
       "      <td>-0.840</td>\n",
       "      <td>-0.2690</td>\n",
       "      <td>0.10300</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S150</th>\n",
       "      <td>-0.72800</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>-0.0492</td>\n",
       "      <td>-0.138</td>\n",
       "      <td>-0.2200</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>0.117</td>\n",
       "      <td>-0.313</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.07820</td>\n",
       "      <td>0.1410</td>\n",
       "      <td>0.7260</td>\n",
       "      <td>0.868000</td>\n",
       "      <td>0.662</td>\n",
       "      <td>-0.57000</td>\n",
       "      <td>-0.399</td>\n",
       "      <td>-0.0959</td>\n",
       "      <td>-0.06350</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S151</th>\n",
       "      <td>-1.16000</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>0.2690</td>\n",
       "      <td>0.416</td>\n",
       "      <td>-0.2380</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>-0.140</td>\n",
       "      <td>-0.170</td>\n",
       "      <td>-0.414</td>\n",
       "      <td>0.0519</td>\n",
       "      <td>...</td>\n",
       "      <td>0.12200</td>\n",
       "      <td>-0.3800</td>\n",
       "      <td>-0.2970</td>\n",
       "      <td>0.046200</td>\n",
       "      <td>0.045</td>\n",
       "      <td>-0.31900</td>\n",
       "      <td>0.835</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>-0.00809</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S152</th>\n",
       "      <td>-0.59600</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>0.0210</td>\n",
       "      <td>0.282</td>\n",
       "      <td>-0.0356</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>-0.517</td>\n",
       "      <td>-0.815</td>\n",
       "      <td>-0.752</td>\n",
       "      <td>-0.0125</td>\n",
       "      <td>...</td>\n",
       "      <td>0.06070</td>\n",
       "      <td>-0.1260</td>\n",
       "      <td>-0.0662</td>\n",
       "      <td>-0.010300</td>\n",
       "      <td>1.050</td>\n",
       "      <td>-0.35600</td>\n",
       "      <td>0.416</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>-0.65800</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153</th>\n",
       "      <td>-0.22400</td>\n",
       "      <td>-0.04345</td>\n",
       "      <td>0.0559</td>\n",
       "      <td>-0.222</td>\n",
       "      <td>-0.5960</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>-0.263</td>\n",
       "      <td>-0.277</td>\n",
       "      <td>-0.449</td>\n",
       "      <td>-0.0564</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.13000</td>\n",
       "      <td>0.0234</td>\n",
       "      <td>0.4250</td>\n",
       "      <td>0.677000</td>\n",
       "      <td>0.587</td>\n",
       "      <td>-0.47700</td>\n",
       "      <td>-0.422</td>\n",
       "      <td>-0.0931</td>\n",
       "      <td>0.05570</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 27411 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AAGAB:NP_078942.3:K290_acetylome  ABCE1:NP_001035809.1:K431_acetylome  \\\n",
       "S001                           0.46100                             -0.22700   \n",
       "S002                           1.77000                              0.56400   \n",
       "S003                          -0.81500                              0.10500   \n",
       "S005                          -0.00334                             -0.04860   \n",
       "S006                           0.20500                              0.48900   \n",
       "...                                ...                                  ...   \n",
       "S149                          -0.75700                             -0.04345   \n",
       "S150                          -0.72800                             -0.04345   \n",
       "S151                          -1.16000                             -0.04345   \n",
       "S152                          -0.59600                             -0.04345   \n",
       "S153                          -0.22400                             -0.04345   \n",
       "\n",
       "      ACAA2:NP_006102.2:K13_acetylome  ACAA2:NP_006102.2:K137_acetylome  \\\n",
       "S001                          -1.0700                            -1.030   \n",
       "S002                          -1.0300                            -0.906   \n",
       "S003                           0.0573                             0.719   \n",
       "S005                          -0.7900                            -0.949   \n",
       "S006                           0.6510                             0.434   \n",
       "...                               ...                               ...   \n",
       "S149                           0.0604                             0.206   \n",
       "S150                          -0.0492                            -0.138   \n",
       "S151                           0.2690                             0.416   \n",
       "S152                           0.0210                             0.282   \n",
       "S153                           0.0559                            -0.222   \n",
       "\n",
       "      ACAA2:NP_006102.2:K234_acetylome  ACADM:NP_001272972.1:K312_acetylome  \\\n",
       "S001                           -1.0400                              2.41000   \n",
       "S002                           -0.6040                             -0.55700   \n",
       "S003                            0.6040                              0.65100   \n",
       "S005                           -0.3210                              0.70400   \n",
       "S006                            0.0320                             -0.45000   \n",
       "...                                ...                                  ...   \n",
       "S149                           -0.5580                              0.00757   \n",
       "S150                           -0.2200                              0.00757   \n",
       "S151                           -0.2380                              0.00757   \n",
       "S152                           -0.0356                              0.00757   \n",
       "S153                           -0.5960                              0.00757   \n",
       "\n",
       "      ACADVL:NP_001257376.1:K262_acetylome  \\\n",
       "S001                                 1.070   \n",
       "S002                                -0.329   \n",
       "S003                                 1.160   \n",
       "S005                                 0.934   \n",
       "S006                                 0.111   \n",
       "...                                    ...   \n",
       "S149                                -0.505   \n",
       "S150                                -0.116   \n",
       "S151                                -0.140   \n",
       "S152                                -0.517   \n",
       "S153                                -0.263   \n",
       "\n",
       "      ACADVL:NP_001257376.1:K299_acetylome  \\\n",
       "S001                                 1.930   \n",
       "S002                                 0.360   \n",
       "S003                                 1.100   \n",
       "S005                                 1.250   \n",
       "S006                                 0.247   \n",
       "...                                    ...   \n",
       "S149                                -0.715   \n",
       "S150                                 0.117   \n",
       "S151                                -0.170   \n",
       "S152                                -0.815   \n",
       "S153                                -0.277   \n",
       "\n",
       "      ACADVL:NP_001257376.1:K301_acetylome  ACAT1:NP_000010.1:K124_acetylome  \\\n",
       "S001                                 1.670                            0.5430   \n",
       "S002                                -0.586                            0.1670   \n",
       "S003                                 0.902                           -0.6980   \n",
       "S005                                 1.190                           -0.1600   \n",
       "S006                                -0.441                           -0.0157   \n",
       "...                                    ...                               ...   \n",
       "S149                                -0.804                           -0.0119   \n",
       "S150                                -0.313                            0.0169   \n",
       "S151                                -0.414                            0.0519   \n",
       "S152                                -0.752                           -0.0125   \n",
       "S153                                -0.449                           -0.0564   \n",
       "\n",
       "      ...  ZSWIM8_proteome  ZW10_proteome  ZWILCH_proteome  ZWINT_proteome  \\\n",
       "S001  ...         -0.08770         0.0229           0.1090       -0.000315   \n",
       "S002  ...         -0.03560         0.3630           1.0700        0.737000   \n",
       "S003  ...          0.00112         0.0105          -0.1160       -0.000315   \n",
       "S005  ...          0.07250        -0.0714           0.0933        0.156000   \n",
       "S006  ...         -0.17600        -1.2200          -0.5620        0.937000   \n",
       "...   ...              ...            ...              ...             ...   \n",
       "S149  ...         -0.20400        -0.1510           0.0552        0.105000   \n",
       "S150  ...         -0.07820         0.1410           0.7260        0.868000   \n",
       "S151  ...          0.12200        -0.3800          -0.2970        0.046200   \n",
       "S152  ...          0.06070        -0.1260          -0.0662       -0.010300   \n",
       "S153  ...         -0.13000         0.0234           0.4250        0.677000   \n",
       "\n",
       "      ZXDC_proteome  ZYG11B_proteome  ZYX_proteome  ZZEF1_proteome  \\\n",
       "S001         -0.332         -0.43300        -1.020         -0.1230   \n",
       "S002         -0.564         -0.00461        -1.130         -0.0757   \n",
       "S003          0.151         -0.07400        -0.540          0.3200   \n",
       "S005         -0.398         -0.07520        -0.797         -0.0301   \n",
       "S006         -0.646          0.20700        -1.850         -0.1760   \n",
       "...             ...              ...           ...             ...   \n",
       "S149          1.330         -0.67400        -0.840         -0.2690   \n",
       "S150          0.662         -0.57000        -0.399         -0.0959   \n",
       "S151          0.045         -0.31900         0.835          0.1690   \n",
       "S152          1.050         -0.35600         0.416          0.2730   \n",
       "S153          0.587         -0.47700        -0.422         -0.0931   \n",
       "\n",
       "      ZZZ3_proteome  sample_type  \n",
       "S001       -0.08590        tumor  \n",
       "S002       -0.47300        tumor  \n",
       "S003       -0.41900        tumor  \n",
       "S005       -0.46700        tumor  \n",
       "S006        0.05130        tumor  \n",
       "...             ...          ...  \n",
       "S149        0.10300       normal  \n",
       "S150       -0.06350       normal  \n",
       "S151       -0.00809       normal  \n",
       "S152       -0.65800       normal  \n",
       "S153        0.05570       normal  \n",
       "\n",
       "[144 rows x 27411 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_model_inputs['UCEC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "value, tumnorm_value_dif, value_type, phosph, phosph_diff, acet, acet_diff, prot, prot_dif, sample_type, cancer_type\n",
    "\n",
    "rna_exp, rna_exp_diff, somatic_mutation_bool, gene_len, in_complex_withxxxxxx?\n",
    "\n",
    "is_driver"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ll', '00']"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "re.split(r':|_', 'll_00')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AML {'Tbm', 'Tpb'}\n",
      "BRCA_prospective {'T'}\n",
      "CCRCC {'A', 'T'}\n",
      "CM {'A', 'T'}\n",
      "CRC_prospective {'T'}\n",
      "EGAD-PDAC set()\n",
      "GBM {'T'}\n",
      "HNSCC {'A', 'T'}\n",
      "LSCC {'A', 'T'}\n",
      "LUAD {'A', 'T'}\n",
      "OV_prospective {'T'}\n",
      "PDA {'A', 'T'}\n",
      "SAR {'A', 'T'}\n",
      "UCEC {'A', 'prov', 'T'}\n",
      "eocg set()\n",
      "hipsci_managed set()\n",
      "hipsci_public set()\n"
     ]
    }
   ],
   "source": [
    "for cancer, exp in cancer_type_to_expression.items():\n",
    "#     print(exp.columns)\n",
    "    print(cancer, {c.split('.')[2] for c in exp.columns if len(c.split('.'))>2})\n",
    "# {c.split('.')[2] for c in cancer_type_to_expression['UCEC'].columns}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CCRCC', 'HNSCC', 'LSCC', 'LUAD', 'UCEC']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## define cancers you want to keep\n",
    "## here i filter by whether there is normal and tumor\n",
    "to_keep = ['CCRCC', 'HNSCC', 'LSCC', 'LUAD', 'UCEC']\n",
    "to_keep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "def normalize_gene_expression_columns(metadata, sample_list, tumor_identifiers=['.T.'],\n",
    "                                      normal_identifiers=['.A.', '.N.']):\n",
    "    tumor_mask = [True if 'Tumor' in st else False for st in metadata['sample_type']]\n",
    "    nontumor_mask = [False if 'Tumor' in st else True for st in metadata['sample_type']]\n",
    "    tumor = metadata[tumor_mask]\n",
    "    non_tumor = metadata[nontumor_mask]\n",
    "    \n",
    "    new_ids = []\n",
    "    for s_id in sample_list:\n",
    "        case = s_id.split('.')[0]\n",
    "#         if '.T.' in s_id:\n",
    "        if any([i in s_id for i in tumor_identifiers]):\n",
    "            filtered = tumor[tumor['case_id']==case]\n",
    "        elif any([i in s_id for i in normal_identifiers]):\n",
    "            filtered = non_tumor[non_tumor['case_id']==case]\n",
    "        else:\n",
    "            filtered = None\n",
    "            \n",
    "        if filtered is not None and filtered.shape[0]: new_ids.append(filtered.index[0])\n",
    "        else: new_ids.append(s_id)\n",
    "\n",
    "    return new_ids\n",
    "\n",
    "def get_sample_to_mutation_status(metadata, maf_df):\n",
    "    \n",
    "    samples = sorted(set(maf_df['sample_id']))\n",
    "    prot_to_case = {c:metadata.loc[c, 'case_id'] for i, c in enumerate(metadata.index)}\n",
    "#     prot_to_case = {s1:s2 for s1, s2 in zip(new_ids, samples)}\n",
    "    case_to_prot = {v:k for k, v in prot_to_case.items()}\n",
    "    genes = sorted(set(maf_df['Hugo_Symbol']))\n",
    "#     print(case_to_prot)\n",
    "\n",
    "    sample_to_gene_to_mutation_status = {s:{} for s in prot_to_case.keys()}\n",
    "    for sample in sample_to_gene_to_mutation_status.keys():\n",
    "        maf_name = prot_to_case[sample]\n",
    "        filtered = maf_df[maf_df['sample_id']==maf_name]\n",
    "#         print(filtered.shape)\n",
    "        filtered_genes = set(filtered['Hugo_Symbol'])\n",
    "#         print(filtered.shape)\n",
    "        for gene in genes:\n",
    "#             gene_filtered = filtered[filtered['Hugo_Symbol']==gene]\n",
    "            if gene in filtered_genes:\n",
    "                sample_to_gene_to_mutation_status[sample][gene] = True\n",
    "            else:\n",
    "                sample_to_gene_to_mutation_status[sample][gene] = False\n",
    "    return sample_to_gene_to_mutation_status\n",
    "    \n",
    "            \n",
    "\n",
    "# def prepare_gene_level_aggregates(df, metadata, gene_expression_df):\n",
    "#     sample_types = df['sample_type'].to_list()\n",
    "#     dtypes = sorted(set([re.split(r'_', x)[-1] for x in df.columns if 'type' not in x]))\n",
    "# #     print(dtypes)\n",
    "#     combined = None\n",
    "#     for dtype in dtypes:\n",
    "#         filtered = df[[c for c in df.columns if dtype == c.split('_')[-1]]].copy()\n",
    "#         filtered = filtered.transpose()\n",
    "#         filtered['gene'] = [re.split(r':|_', x)[0] for x in filtered.index]\n",
    "#         filtered = filtered.fillna(np.inf).groupby('gene').mean().replace(np.inf, np.nan)\n",
    "#         filtered.columns = [f'{c}_{dtype}' for c in filtered.columns]\n",
    "#         if combined is None:\n",
    "#             combined = filtered.copy()\n",
    "#         else:\n",
    "#             combined = pd.merge(combined, filtered, how='outer', left_index=True, right_index=True)\n",
    "\n",
    "#     combined = combined.transpose()\n",
    "    \n",
    "#     combined['dtype'] = [x.split('_')[-1] for x in combined.index]\n",
    "    \n",
    "#     overall = combined.groupby('dtype').mean()\n",
    "\n",
    "#     ls = [df.loc[s.split('_')[0], 'sample_type'] for s in combined.index]\n",
    "#     combined['dtype'] = [f'{d}_{s}' for d, s in zip(combined['dtype'], ls)]\n",
    "#     grouped = combined.groupby('dtype').mean()\n",
    "#     overall = pd.concat((overall, grouped), axis=0)\n",
    "#     overall = overall.transpose()\n",
    "    \n",
    "#     for dtype in dtypes:\n",
    "#         filtered = overall[[c for c in overall.columns if (dtype == c.split('_')[0] and ('normal' in c or 'tumor' in c))]]\n",
    "#         print(filtered.columns)\n",
    "#         filtered.columns = [c.split('_')[-1] for c in filtered.columns]\n",
    "#         overall[f'{dtype}_tumor_normal_difference'] = filtered['tumor'] - filtered['normal']\n",
    "        \n",
    "# #     sample_to_pair = {}\n",
    "# #     for sample, sample_type in zip(df.index, df['sample_type']):\n",
    "# #         ## look for same case\n",
    "# #         case = metadata.loc[sample, 'case_id']\n",
    "# #         filtered = metadata[metadata['case_id']==case]\n",
    "# #         filtered = filtered[filtered['sample_type']!='Tumor'] if sample_type == 'tumor' else filtered[filtered['sample_type']=='Tumor']\n",
    "# #         if filtered.shape[0]:\n",
    "# #             sample_to_pair[sample] = filtered.index[0]\n",
    "# #     print(sample_to_pair)\n",
    "    \n",
    "#     data = []\n",
    "#     tumor_mask = [True if '.T.' in t else False for t in gene_expression_df.columns]\n",
    "#     non_tumor_mask = [True if not x else False for x in tumor_mask]\n",
    "#     if gene_expression_df is not None:\n",
    "#         expression = gene_expression_df.copy()\n",
    "#         expression.columns = normalize_gene_expression_columns(metadata, expression.columns)\n",
    "#         for gene in overall.index:\n",
    "#             if gene in expression.index:\n",
    "#                 filtered = expression.loc[gene, :]\n",
    "#                 avg = np.mean(filtered.values.flatten())\n",
    "#                 tumor_avg = np.mean([c for i, c in enumerate(filtered) if tumor_mask[i]])\n",
    "#                 normal_avg = np.mean([c for i, c in enumerate(filtered) if non_tumor_mask[i]])\n",
    "#                 data.append([avg, tumor_avg, normal_avg, tumor_avg / normal_avg])\n",
    "#             else:\n",
    "#                 data.append([np.nan, np.nan, np.nan])\n",
    "#     temp = pd.DataFrame(data=data, index=overall.index, columns=['rna_expression_average',\n",
    "#                                                                  'rna_expression_average_tumor',\n",
    "#                                                                 'rna_expression_average_normal',\n",
    "#                                                                 'rna_expression_tumor_normal_fc'])\n",
    "#     overall = pd.merge(overall, temp, left_index=True, right_index=True)\n",
    "#     return overall\n",
    "\n",
    "def prepare_site_level_features(df, metadata, gene_expression_df, maf_df):\n",
    "    dtypes = sorted(set([re.split(r'_', x)[-1] for x in df.columns if 'type' not in x]))\n",
    "#     dtypes = [t for t in dtypes if t not in ['proteome']]\n",
    "    print(dtypes)\n",
    "    print(sorted(set(metadata['sample_type'])))\n",
    "#     print(dtypes)\n",
    "    sample_to_type = {s:t for s, t in zip(df.index, df['sample_type'])}\n",
    "    sample_to_pair = {}\n",
    "    for sample, sample_type in zip(df.index, df['sample_type']):\n",
    "        ## look for same case\n",
    "#         print(metadata.index)\n",
    "#         print(gene_expression_df.columns)\n",
    "#         print(sample)\n",
    "        case = metadata.loc[sample, 'case_id']\n",
    "#         print(case)\n",
    "        filtered = metadata[metadata['case_id']==case]\n",
    "        filtered = filtered[~filtered.index.duplicated(keep='first')]\n",
    "#         print(sample, sample_type)\n",
    "#         print(list(filtered.index), list(filtered['sample_type']))\n",
    "        tumor_mask = [True if 'Tumor' in st else False for st in filtered['sample_type']]\n",
    "        nontumor_mask = [False if 'Tumor' in st else True for st in filtered['sample_type']]\n",
    "        filtered = filtered[nontumor_mask] if 'tumor' == sample_type else filtered[tumor_mask]\n",
    "#         print(list(filtered.index), list(filtered['sample_type']))\n",
    "#         print(filtered.index, filtered['sample_type'])\n",
    "\n",
    "        if filtered.shape[0]:\n",
    "            sample_to_pair[sample] = filtered.index[0]\n",
    "#     print(len(sample_to_pair), df.shape)\n",
    "    print(sample_to_pair)\n",
    "            \n",
    "    sample_to_gene_to_status = get_sample_to_mutation_status(metadata, maf_df)\n",
    "        \n",
    "    expression = gene_expression_df.copy()\n",
    "    expression.columns = normalize_gene_expression_columns(metadata, expression.columns)\n",
    "    expression = expression.loc[:,~expression.columns.duplicated()]\n",
    "    expression = expression.loc[~expression.index.duplicated(), :]\n",
    "    expressed_genes = set(expression.index)\n",
    "    expressed_columns = set(expression.columns)\n",
    "    \n",
    "    print(expressed_columns)\n",
    "    data = []\n",
    "    ids = []\n",
    "    for dtype in dtypes:\n",
    "        cs = [c for c in df.columns if dtype == c.split('_')[-1]]\n",
    "        cs += ['sample_type']\n",
    "        filtered = df[cs].copy()\n",
    "        print(dtype, filtered.shape)\n",
    "#         print(filtered.columns[-5:])\n",
    "#         print(list(expression.columns))\n",
    "        for i, (sample, sample_type) in enumerate(zip(filtered.index, filtered['sample_type'])):\n",
    "#             if i % 11 == 10:\n",
    "#                 print(i, sample)\n",
    "#                 break\n",
    "            has_expression = True if sample in expression.columns else False\n",
    "#             print(sample, has_expression, sample_to_pair.get(sample), sample_to_pair.get(sample) in expression.columns)\n",
    "#             if sample in expression.columns:\n",
    "#                 s_expression = expres\n",
    "#             print('here', len(filtered.columns))\n",
    "            paired_sample = sample_to_pair[sample] if sample in sample_to_pair else None\n",
    "            in_filtered = paired_sample in filtered.index\n",
    "            for feat in filtered.columns[:-1]:\n",
    "                s = sample.split('_')[0]\n",
    "                if dtype == 'proteome':\n",
    "                    gene = feat.split('_')[0]\n",
    "                else:\n",
    "                    gene = feat.split(':')[0]\n",
    "                ids.append(f'{s}_{dtype}_{feat}')\n",
    "                val = float(filtered.loc[sample, feat])\n",
    "#                 try:\n",
    "#                 print(sample, sample_to_pair.get(sample))\n",
    "                pair_val = float(filtered.loc[paired_sample, feat]) if in_filtered else np.nan\n",
    "#                 except KeyError as e:\n",
    "#                     print(sample, sample_to_pair[sample])\n",
    "#                     print(filtered.shape, sample_to_pair[sample] in filtered.index)\n",
    "#                     print(len(ids), i)\n",
    "#                     print(e)\n",
    "#                     raise RuntimeError()\n",
    "                    #                print(val, pair_val)\n",
    "                diff = np.nan if pd.isnull(pair_val) else val - pair_val\n",
    "                \n",
    "                ## do somatic mutation\n",
    "                has_mutation = sample_to_gene_to_status[sample].get(gene, False)\n",
    "                \n",
    "                ## do expression\n",
    "                if has_expression:\n",
    "                    \n",
    "                    exp = expression.loc[gene, sample] if gene in expressed_genes else np.nan\n",
    "                    \n",
    "                    pair_exp = expression.loc[gene, paired_sample] if paired_sample in expressed_columns and gene in expressed_genes else np.nan\n",
    "#                     if sample in sample_to_pair:\n",
    "#                         print(gene, gene in expressed_genes, exp)\n",
    "#                         print(sample, sample_to_pair[sample], sample_to_pair[sample] in expressed_columns, pair_exp)\n",
    "                    exp_diff = np.nan if pd.isnull(exp) or pd.isnull(pair_exp) else exp - pair_exp\n",
    "                else:\n",
    "                    exp, pair_exp, exp_diff = np.nan, np.nan, np.nan\n",
    "                \n",
    "                \n",
    "                \n",
    "                data.append([val, pair_val, diff, sample_type, dtype, exp, pair_exp, exp_diff, has_mutation,\n",
    "                            gene])\n",
    "    \n",
    "#     print(len(data))     \n",
    "    to_return = pd.DataFrame(data=data, index=ids,\n",
    "                             columns=['value', 'paired_value', 'paired_difference', 'sample_type', 'data_type',\n",
    "                                     'rna_expression', 'paired_rna_expression', 'difference_rna_expression',\n",
    "                                     'has_mutation', 'gene'])\n",
    "    return to_return\n",
    "    \n",
    " \n",
    "        \n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cancer = 'LSCC'\n",
    "# cancer_to_protein[cancer][metadata_key]['metadata']\n",
    "# # cancer_type_to_expression[cancer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cancer_type_to_expression[cancer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CCRCC\n",
      "['phosphoproteome', 'proteome']\n",
      "['Primary Tumor', 'Solid Tissue Normal']\n",
      "{'C3L-00004.T': 'C3L-00004.N', 'C3L-00004.N': 'C3L-00004.T', 'C3L-00010.T': 'C3L-00010.N', 'C3L-00010.N': 'C3L-00010.T', 'C3L-00011.T': 'C3L-00011.N', 'C3L-00011.N': 'C3L-00011.T', 'C3L-00026.T': 'C3L-00026.N', 'C3L-00026.N': 'C3L-00026.T', 'C3L-00079.T': 'C3L-00079.N', 'C3L-00079.N': 'C3L-00079.T', 'C3L-00088.T': 'C3L-00088.N', 'C3L-00088.N': 'C3L-00088.T', 'C3L-00096.T': 'C3L-00096.N', 'C3L-00096.N': 'C3L-00096.T', 'C3L-00097.T': 'C3L-00097.N', 'C3L-00097.N': 'C3L-00097.T', 'C3L-00103.T': 'C3L-00103.N', 'C3L-00103.N': 'C3L-00103.T', 'C3L-00183.N': 'C3L-00183.T', 'C3L-00183.T': 'C3L-00183.N', 'C3L-00360.T': 'C3L-00360.N', 'C3L-00360.N': 'C3L-00360.T', 'C3L-00369.T': 'C3L-00369.N', 'C3L-00369.N': 'C3L-00369.T', 'C3L-00416.T': 'C3L-00416.N', 'C3L-00416.N': 'C3L-00416.T', 'C3L-00418.T': 'C3L-00418.N', 'C3L-00418.N': 'C3L-00418.T', 'C3L-00447.T': 'C3L-00447.N', 'C3L-00447.N': 'C3L-00447.T', 'C3L-00448.T': 'C3L-00448.N', 'C3L-00448.N': 'C3L-00448.T', 'C3L-00561.T': 'C3L-00561.N', 'C3L-00561.N': 'C3L-00561.T', 'C3L-00581.T': 'C3L-00581.N', 'C3L-00581.N': 'C3L-00581.T', 'C3L-00583.T': 'C3L-00583.N', 'C3L-00583.N': 'C3L-00583.T', 'C3L-00606.T': 'C3L-00606.N', 'C3L-00606.N': 'C3L-00606.T', 'C3L-00607.T': 'C3L-00607.N', 'C3L-00607.N': 'C3L-00607.T', 'C3L-00791.T': 'C3L-00791.N', 'C3L-00791.N': 'C3L-00791.T', 'C3L-00814.T': 'C3L-00814.N', 'C3L-00814.N': 'C3L-00814.T', 'C3L-00902.T': 'C3L-00902.N', 'C3L-00902.N': 'C3L-00902.T', 'C3L-00907.T': 'C3L-00907.N', 'C3L-00907.N': 'C3L-00907.T', 'C3L-00908.T': 'C3L-00908.N', 'C3L-00908.N': 'C3L-00908.T', 'C3L-00910.T': 'C3L-00910.N', 'C3L-00910.N': 'C3L-00910.T', 'C3L-00917.T': 'C3L-00917.N', 'C3L-00917.N': 'C3L-00917.T', 'C3L-01281.T': 'C3L-01281.N', 'C3L-01281.N': 'C3L-01281.T', 'C3L-01286.T': 'C3L-01286.N', 'C3L-01286.N': 'C3L-01286.T', 'C3L-01287.T': 'C3L-01287.N', 'C3L-01287.N': 'C3L-01287.T', 'C3L-01302.T': 'C3L-01302.N', 'C3L-01302.N': 'C3L-01302.T', 'C3L-01313.T': 'C3L-01313.N', 'C3L-01313.N': 'C3L-01313.T', 'C3L-01603.T': 'C3L-01603.N', 'C3L-01603.N': 'C3L-01603.T', 'C3L-01607.T': 'C3L-01607.N', 'C3L-01607.N': 'C3L-01607.T', 'C3L-01836.T': 'C3L-01836.N', 'C3L-01836.N': 'C3L-01836.T', 'C3L-01861.T': 'C3L-01861.N', 'C3L-01861.N': 'C3L-01861.T', 'C3L-01882.T': 'C3L-01882.N', 'C3L-01882.N': 'C3L-01882.T', 'C3L-01885.T': 'C3L-01885.N', 'C3L-01885.N': 'C3L-01885.T', 'C3N-00148.T': 'C3N-00148.N', 'C3N-00148.N': 'C3N-00148.T', 'C3N-00149.T': 'C3N-00149.N', 'C3N-00149.N': 'C3N-00149.T', 'C3N-00150.T': 'C3N-00150.N', 'C3N-00150.N': 'C3N-00150.T', 'C3N-00168.T': 'C3N-00168.N', 'C3N-00168.N': 'C3N-00168.T', 'C3N-00177.T': 'C3N-00177.N', 'C3N-00177.N': 'C3N-00177.T', 'C3N-00194.N': 'C3N-00194.T', 'C3N-00194.T': 'C3N-00194.N', 'C3N-00242.T': 'C3N-00242.N', 'C3N-00242.N': 'C3N-00242.T', 'C3N-00244.N': 'C3N-00244.T', 'C3N-00244.T': 'C3N-00244.N', 'C3N-00246.T': 'C3N-00246.N', 'C3N-00246.N': 'C3N-00246.T', 'C3N-00310.T': 'C3N-00310.N', 'C3N-00310.N': 'C3N-00310.T', 'C3N-00312.T': 'C3N-00312.N', 'C3N-00312.N': 'C3N-00312.T', 'C3N-00314.T': 'C3N-00314.N', 'C3N-00314.N': 'C3N-00314.T', 'C3N-00317.T': 'C3N-00317.N', 'C3N-00317.N': 'C3N-00317.T', 'C3N-00320.T': 'C3N-00320.N', 'C3N-00320.N': 'C3N-00320.T', 'C3N-00390.T': 'C3N-00390.N', 'C3N-00390.N': 'C3N-00390.T', 'C3N-00435.T': 'C3N-00435.N', 'C3N-00435.N': 'C3N-00435.T', 'C3N-00437.T': 'C3N-00437.N', 'C3N-00437.N': 'C3N-00437.T', 'C3N-00491.N': 'C3N-00491.T', 'C3N-00491.T': 'C3N-00491.N', 'C3N-00492.N': 'C3N-00492.T', 'C3N-00492.T': 'C3N-00492.N', 'C3N-00494.T': 'C3N-00494.N', 'C3N-00494.N': 'C3N-00494.T', 'C3N-00495.T': 'C3N-00495.N', 'C3N-00495.N': 'C3N-00495.T', 'C3N-00573.T': 'C3N-00573.N', 'C3N-00573.N': 'C3N-00573.T', 'C3N-00577.T': 'C3N-00577.N', 'C3N-00577.N': 'C3N-00577.T', 'C3N-00646.T': 'C3N-00646.N', 'C3N-00646.N': 'C3N-00646.T', 'C3N-00733.T': 'C3N-00733.N', 'C3N-00733.N': 'C3N-00733.T', 'C3N-00831.T': 'C3N-00831.N', 'C3N-00831.N': 'C3N-00831.T', 'C3N-00834.T': 'C3N-00834.N', 'C3N-00834.N': 'C3N-00834.T', 'C3N-00852.T': 'C3N-00852.N', 'C3N-00852.N': 'C3N-00852.T', 'C3N-00953.T': 'C3N-00953.N', 'C3N-00953.N': 'C3N-00953.T', 'C3N-01175.T': 'C3N-01175.N', 'C3N-01175.N': 'C3N-01175.T', 'C3N-01176.T': 'C3N-01176.N', 'C3N-01176.N': 'C3N-01176.T', 'C3N-01178.T': 'C3N-01178.N', 'C3N-01178.N': 'C3N-01178.T', 'C3N-01179.T': 'C3N-01179.N', 'C3N-01179.N': 'C3N-01179.T', 'C3N-01200.T': 'C3N-01200.N', 'C3N-01200.N': 'C3N-01200.T', 'C3N-01214.T': 'C3N-01214.N', 'C3N-01214.N': 'C3N-01214.T', 'C3N-01220.T': 'C3N-01220.N', 'C3N-01220.N': 'C3N-01220.T', 'C3N-01261.T': 'C3N-01261.N', 'C3N-01261.N': 'C3N-01261.T', 'C3N-01361.T': 'C3N-01361.N', 'C3N-01361.N': 'C3N-01361.T', 'C3N-01522.T': 'C3N-01522.N', 'C3N-01522.N': 'C3N-01522.T', 'C3N-01524.T': 'C3N-01524.N', 'C3N-01524.N': 'C3N-01524.T', 'C3N-01646.T': 'C3N-01646.N', 'C3N-01646.N': 'C3N-01646.T', 'C3N-01648.T': 'C3N-01648.N', 'C3N-01648.N': 'C3N-01648.T', 'C3N-01649.T': 'C3N-01649.N', 'C3N-01649.N': 'C3N-01649.T', 'C3N-01651.T': 'C3N-01651.N', 'C3N-01651.N': 'C3N-01651.T', 'C3N-01808.T': 'C3N-01808.N', 'C3N-01808.N': 'C3N-01808.T'}\n",
      "{'C3L-00096.N', 'C3N-01080.RNA-Seq.T.hg38', 'C3L-01462.RNA-Seq.T.hg38', 'C3N-00314.N', 'C3L-00908.N', 'C3N-02763.RNA-Seq.T.hg38', 'C3N-00573.T', 'C3N-00168.N', 'C3L-01464.RNA-Seq.A.hg38', 'C3N-01077.RNA-Seq.A.hg38', 'C3N-00834.T', 'C3L-01288.T', 'C3L-01665.RNA-Seq.T.hg38', 'C3N-01808.T', 'C3N-01361.N', 'C3N-01088.RNA-Seq.A.hg38', 'C3L-00606.N', 'C3L-02220.RNA-Seq.T.hg38', 'C3N-00390.T', 'C3N-03019.RNA-Seq.T.hg38', 'C3N-00953.T', 'C3L-01603.N', 'C3L-02220.RNA-Seq.A.hg38', 'C3L-02556.RNA-Seq.A.hg38', 'C3N-00491.T', 'C3N-01213.T', 'C3L-02551.RNA-Seq.A.hg38', 'C3L-00917.N', 'C3N-01657.RNA-Seq.T.hg38', 'C3N-01654.RNA-Seq.T.hg38', 'C3N-01656.RNA-Seq.T.hg38', 'C3L-00583.N', 'C3L-02202.RNA-Seq.T.hg38', 'C3N-00495.N', 'C3N-02332.RNA-Seq.A.hg38', 'C3L-02842.RNA-Seq.A.hg38', 'C3N-00177.T', 'C3N-01651.N', 'C3N-00320.T', 'C3N-01200.T', 'C3N-01180.T', 'C3N-02070.RNA-Seq.T.hg38', 'C3N-01178.N', 'C3L-00010.T', 'C3N-00244.T', 'C3N-01176.T', 'C3N-01522.T', 'C3L-00765.T', 'C3N-02160.RNA-Seq.T.hg38', 'C3L-00766.T', 'C3L-00103.N', 'C3N-02338.RNA-Seq.T.hg38', 'C3L-00583.T', 'C3L-00791.T', 'C3N-02265.RNA-Seq.A.hg38', 'C3N-01088.RNA-Seq.T.hg38', 'C3L-01287.T', 'C3L-01882.N', 'C3N-01082.RNA-Seq.T.hg38', 'C3L-00096.T', 'C3N-01649.T', 'C3L-00980.RNA-Seq.A.hg38', 'C3L-00792.T', 'C3N-02263.RNA-Seq.T.hg38', 'C3L-00982.RNA-Seq.T.hg38', 'C3L-01553.T', 'C3N-01808.N', 'C3N-01176.N', 'C3N-00733.T', 'C3L-01869.RNA-Seq.T.hg38', 'C3L-01607.T', 'C3L-01929.RNA-Seq.T.hg38', 'C3N-00149.N', 'C3N-00437.T', 'C3N-00721.RNA-Seq.A.hg38', 'C3L-00011.N', 'C3L-01836.N', 'C3N-00492.T', 'C3N-00647.RNA-Seq.A.hg38', 'C3N-01081.RNA-Seq.A.hg38', 'C3L-00910.N', 'C3N-00154.T', 'C3N-00646.T', 'C3L-00097.N', 'C3L-00910.T', 'C3N-02015.RNA-Seq.T.hg38', 'C3N-00715.RNA-Seq.A.hg38', 'C3N-01361.T', 'C3L-01313.T', 'C3N-02066.RNA-Seq.T.hg38', 'C3N-01078.RNA-Seq.A.hg38', 'C3N-00246.T', 'C3N-01086.RNA-Seq.A.hg38', 'C3L-00607.N', 'C3L-02841.RNA-Seq.T.hg38', 'C3L-00969.RNA-Seq.T.hg38', 'C3L-00908.T', 'C3N-00577.T', 'C3N-02084.RNA-Seq.T.hg38', 'C3L-01560.T', 'C3L-01462.RNA-Seq.A.hg38', 'C3L-00982.RNA-Seq.A.hg38', 'C3L-01283.T', 'C3L-00800.T', 'C3L-01978.RNA-Seq.T.hg38', 'C3N-01080.RNA-Seq.A.hg38', 'C3L-01885.N', 'C3N-01536.RNA-Seq.A.hg38', 'C3N-01090.RNA-Seq.A.hg38', 'C3L-00079.T', 'C3N-00831.N', 'C3L-00814.N', 'C3L-02364.RNA-Seq.T.hg38', 'C3L-01106.RNA-Seq.A.hg38', 'C3N-01806.RNA-Seq.T.hg38', 'C3L-02856.RNA-Seq.T.hg38', 'C3N-02577.RNA-Seq.T.hg38', 'C3L-01281.T', 'C3L-00079.N', 'C3L-00917.T', 'C3L-00799.T', 'C3L-02544.RNA-Seq.A.hg38', 'C3N-00177.N', 'C3N-02332.RNA-Seq.T.hg38', 'C3L-01459.RNA-Seq.T.hg38', 'C3L-00360.N', 'C3N-01522.N', 'C3L-01681.RNA-Seq.T.hg38', 'C3L-00966.RNA-Seq.T.hg38', 'C3N-01220.T', 'C3N-02577.RNA-Seq.A.hg38', 'C3L-02211.RNA-Seq.T.hg38', 'C3L-00969.RNA-Seq.A.hg38', 'C3L-01603.T', 'C3L-00817.T', 'C3N-02431.RNA-Seq.A.hg38', 'C3L-02364.RNA-Seq.A.hg38', 'C3N-01646.N', 'C3N-02264.RNA-Seq.T.hg38', 'C3L-00088.N', 'C3L-00814.T', 'C3L-00789.RNA-Seq.T.hg38', 'C3N-01648.N', 'C3L-02551.RNA-Seq.T.hg38', 'C3L-01279.RNA-Seq.A.hg38', 'C3L-01470.RNA-Seq.T.hg38', 'C3N-02429.RNA-Seq.T.hg38', 'C3L-01676.RNA-Seq.A.hg38', 'C3N-02068.RNA-Seq.T.hg38', 'C3N-02726.RNA-Seq.T.hg38', 'C3N-02811.RNA-Seq.T.hg38', 'C3L-02352.RNA-Seq.T.hg38', 'C3L-01952.RNA-Seq.T.hg38', 'C3N-02248.RNA-Seq.T.hg38', 'C3L-00991.RNA-Seq.T.hg38', 'C3N-00733.N', 'C3L-00026.T', 'C3N-02005.RNA-Seq.T.hg38', 'C3N-00148.N', 'C3L-00970.RNA-Seq.T.hg38', 'C3L-00981.RNA-Seq.A.hg38', 'C3N-01179.N', 'C3N-01648.T', 'C3N-02063.RNA-Seq.T.hg38', 'C3N-02439.RNA-Seq.A.hg38', 'C3N-01178.T', 'C3N-00832.T', 'C3L-00448.T', 'C3L-02841.RNA-Seq.A.hg38', 'C3N-01655.RNA-Seq.T.hg38', 'C3L-00796.T', 'C3N-02248.RNA-Seq.A.hg38', 'C3L-01102.RNA-Seq.A.hg38', 'C3N-02811.RNA-Seq.A.hg38', 'C3N-01091.RNA-Seq.A.hg38', 'C3L-00966.RNA-Seq.A.hg38', 'C3N-02066.RNA-Seq.A.hg38', 'C3N-02439.RNA-Seq.T.hg38', 'C3L-00097.T', 'C3L-00790.T', 'C3L-01107.RNA-Seq.T.hg38', 'C3L-00907.T', 'C3N-00150.T', 'C3N-00313.T', 'C3L-01954.RNA-Seq.T.hg38', 'C3N-00380.T', 'C3N-01175.N', 'C3L-00970.RNA-Seq.A.hg38', 'C3N-01086.RNA-Seq.T.hg38', 'C3L-00991.RNA-Seq.A.hg38', 'C3N-03020.RNA-Seq.T.hg38', 'C3L-00981.RNA-Seq.T.hg38', 'C3N-02262.RNA-Seq.T.hg38', 'C3N-00435.N', 'C3N-01214.N', 'C3N-01175.T', 'C3N-03021.RNA-Seq.T.hg38', 'C3N-00305.T', 'C3L-00360.T', 'C3L-00165.RNA-Seq.A.hg38', 'C3L-00010.N', 'C3N-02266.RNA-Seq.T.hg38', 'C3L-01882.T', 'C3L-01352.T', 'C3N-01646.T', 'C3L-00416.N', 'C3L-02857.RNA-Seq.T.hg38', 'C3N-00573.N', 'C3L-01281.N', 'C3L-02556.RNA-Seq.T.hg38', 'C3N-00315.T', 'C3L-00183.T', 'C3N-01077.RNA-Seq.T.hg38', 'C3N-00646.N', 'C3N-01905.RNA-Seq.A.hg38', 'C3L-00815.RNA-Seq.T.hg38', 'C3L-00416.T', 'C3N-02761.RNA-Seq.T.hg38', 'C3N-01083.RNA-Seq.T.hg38', 'C3L-00610.T', 'C3L-01286.T', 'C3N-01524.T', 'C3L-00004.T', 'C3N-02945.RNA-Seq.T.hg38', 'C3N-03019.RNA-Seq.A.hg38', 'C3L-00448.N', 'C3N-00715.RNA-Seq.T.hg38', 'C3N-00312.T', 'C3N-03018.RNA-Seq.A.hg38', 'C3L-02177.RNA-Seq.T.hg38', 'C3L-02175.RNA-Seq.T.hg38', 'C3L-00447.T', 'C3L-01869.RNA-Seq.A.hg38', 'C3N-00317.T', 'C3L-00418.T', 'C3N-00834.N', 'C3N-00242.T', 'C3L-02214.RNA-Seq.T.hg38', 'C3L-02208.RNA-Seq.A.hg38', 'C3N-03018.RNA-Seq.T.hg38', 'C3N-02429.RNA-Seq.A.hg38', 'C3N-02085.RNA-Seq.T.hg38', 'C3L-01302.N', 'C3L-01963.RNA-Seq.T.hg38', 'C3N-01536.RNA-Seq.T.hg38', 'C3N-00647.RNA-Seq.T.hg38', 'C3N-01828.RNA-Seq.T.hg38', 'C3N-01085.RNA-Seq.T.hg38', 'C3L-00902.N', 'C3L-01106.RNA-Seq.T.hg38', 'C3N-00314.T', 'C3N-01524.N', 'C3L-01033.RNA-Seq.T.hg38', 'C3L-01976.RNA-Seq.T.hg38', 'C3N-00721.RNA-Seq.T.hg38', 'C3L-01287.N', 'C3N-02726.RNA-Seq.A.hg38', 'C3L-02208.RNA-Seq.T.hg38', 'C3N-01902.RNA-Seq.T.hg38', 'C3L-01557.T', 'C3L-01665.RNA-Seq.A.hg38', 'C3N-01081.RNA-Seq.T.hg38', 'C3L-00581.N', 'C3N-01091.RNA-Seq.T.hg38', 'C3N-02070.RNA-Seq.A.hg38', 'C3N-01989.RNA-Seq.T.hg38', 'C3L-00581.T', 'C3N-00149.T', 'C3N-01649.N', 'C3L-01286.N', 'C3N-01261.T', 'C3L-00088.T', 'C3L-01885.T', 'C3N-00168.T', 'C3L-02176.RNA-Seq.T.hg38', 'C3L-00968.RNA-Seq.T.hg38', 'C3N-03021.RNA-Seq.A.hg38', 'C3N-02945.RNA-Seq.A.hg38', 'C3L-00004.N', 'C3N-01085.RNA-Seq.A.hg38', 'C3N-02082.RNA-Seq.T.hg38', 'C3L-00011.T', 'C3L-02544.RNA-Seq.T.hg38', 'C3N-00150.N', 'C3N-02265.RNA-Seq.T.hg38', 'C3N-02160.RNA-Seq.A.hg38', 'C3L-01681.RNA-Seq.A.hg38', 'C3L-00359.T', 'C3N-01078.RNA-Seq.T.hg38', 'C3N-01806.RNA-Seq.A.hg38', 'C3L-01978.RNA-Seq.A.hg38', 'C3N-00494.T', 'C3N-00310.T', 'C3L-00968.RNA-Seq.A.hg38', 'C3L-01861.T', 'C3N-00577.N', 'C3L-01107.RNA-Seq.A.hg38', 'C3L-01102.RNA-Seq.T.hg38', 'C3N-01261.N', 'C3L-00980.RNA-Seq.T.hg38', 'C3L-00447.N', 'C3L-01313.N', 'C3N-01905.RNA-Seq.T.hg38', 'C3N-00242.N', 'C3N-02082.RNA-Seq.A.hg38', 'C3L-00813.T', 'C3N-01082.RNA-Seq.A.hg38', 'C3N-01220.N', 'C3L-00561.N', 'C3N-01651.T', 'C3N-02723.RNA-Seq.A.hg38', 'C3N-00435.T', 'C3L-01958.RNA-Seq.T.hg38', 'C3L-02201.RNA-Seq.T.hg38', 'C3L-00606.T', 'C3N-00194.T', 'C3L-02352.RNA-Seq.A.hg38', 'C3L-02346.RNA-Seq.A.hg38', 'C3L-00607.T', 'C3N-00852.N', 'C3N-00390.N', 'C3N-01828.RNA-Seq.A.hg38', 'C3N-02224.RNA-Seq.T.hg38', 'C3N-00494.N', 'C3N-00831.T', 'C3N-00495.T', 'C3L-00907.N', 'C3N-00194.N', 'C3N-02723.RNA-Seq.T.hg38', 'C3L-02842.RNA-Seq.T.hg38', 'C3N-01654.RNA-Seq.A.hg38', 'C3N-01989.RNA-Seq.A.hg38', 'C3L-00103.T', 'C3N-01083.RNA-Seq.A.hg38', 'C3L-02211.RNA-Seq.A.hg38', 'C3N-00491.N', 'C3L-01836.T', 'C3N-01902.RNA-Seq.A.hg38', 'C3N-01656.RNA-Seq.A.hg38', 'C3L-00369.T', 'C3N-00852.T', 'C3L-02553.RNA-Seq.T.hg38', 'C3L-01607.N', 'C3L-00812.T', 'C3L-00418.N', 'C3L-00165.RNA-Seq.T.hg38', 'C3N-01200.N', 'C3L-00026.N', 'C3L-00561.T', 'C3L-01464.RNA-Seq.T.hg38', 'C3N-00148.T', 'C3L-01861.N', 'C3L-01279.RNA-Seq.T.hg38', 'C3L-01976.RNA-Seq.A.hg38', 'C3L-01302.T', 'C3L-01465.RNA-Seq.T.hg38', 'C3N-02431.RNA-Seq.T.hg38', 'C3N-02224.RNA-Seq.A.hg38', 'C3L-01951.RNA-Seq.T.hg38', 'C3L-01676.RNA-Seq.T.hg38', 'C3N-02084.RNA-Seq.A.hg38', 'C3N-01214.T', 'C3N-00310.N', 'C3L-02346.RNA-Seq.T.hg38', 'C3L-00369.N', 'C3N-01179.T', 'C3L-00976.RNA-Seq.T.hg38', 'C3L-00976.RNA-Seq.A.hg38', 'C3N-00244.N', 'C3L-00902.T'}\n",
      "phosphoproteome (194, 15256)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "proteome (194, 8443)\n",
      "HNSCC\n",
      "['phosphoproteome', 'proteome']\n",
      "['Primary Tumor', 'Solid Tissue Normal']\n",
      "{'C3L-00997-T': 'C3L-00997-N', 'C3N-03849-T': 'C3N-03849-N', 'C3N-01858-N': 'C3N-01858-T', 'C3L-00997-N': 'C3L-00997-T', 'C3N-01858-T': 'C3N-01858-N', 'C3N-03849-N': 'C3N-03849-T', 'C3N-01859-N': 'C3N-01859-T', 'C3N-03781-T': 'C3N-03781-N', 'C3N-03933-N': 'C3N-03933-T', 'C3L-00999-N': 'C3L-00999-T', 'C3N-03781-N': 'C3N-03781-T', 'C3L-00999-T': 'C3L-00999-N', 'C3N-01859-T': 'C3N-01859-N', 'C3N-03933-T': 'C3N-03933-N', 'C3N-03015-N': 'C3N-03015-T', 'C3N-03490-T': 'C3N-03490-N', 'C3N-03012-T': 'C3N-03012-N', 'C3N-01620-N': 'C3N-01620-T', 'C3N-03012-N': 'C3N-03012-T', 'C3N-03490-N': 'C3N-03490-T', 'C3N-03015-T': 'C3N-03015-N', 'C3N-01620-T': 'C3N-01620-N', 'C3N-01758-N': 'C3N-01758-T', 'C3N-04277-T': 'C3N-04277-N', 'C3N-03045-T': 'C3N-03045-N', 'C3N-02275-T': 'C3N-02275-N-need2remove', 'C3N-02275-N-need2remove': 'C3N-02275-T', 'C3N-01758-T': 'C3N-01758-N', 'C3N-03045-N': 'C3N-03045-T', 'C3N-04277-N': 'C3N-04277-T', 'C3N-02693-T': 'C3N-02693-N', 'C3N-01337-N': 'C3N-01337-T', 'C3N-02713-N': 'C3N-02713-T', 'C3N-03619-T': 'C3N-03619-N', 'C3N-02713-T': 'C3N-02713-N', 'C3N-01337-T': 'C3N-01337-N', 'C3N-02693-N': 'C3N-02693-T', 'C3N-03619-N': 'C3N-03619-T', 'C3L-04844-T': 'C3L-04844-N', 'C3N-01338-T': 'C3N-01338-N', 'C3N-01947-N': 'C3N-01947-T', 'C3N-01645-T': 'C3N-01645-N', 'C3N-01338-N': 'C3N-01338-T', 'C3L-04844-N': 'C3L-04844-T', 'C3N-01947-T': 'C3N-01947-N', 'C3N-01645-N': 'C3N-01645-T', 'C3N-03011-N': 'C3N-03011-T', 'C3L-01237-T': 'C3L-01237-N', 'C3N-01946-N': 'C3N-01946-T', 'C3N-01756-N': 'C3N-01756-T', 'C3N-03011-T': 'C3N-03011-N', 'C3N-01756-T': 'C3N-01756-N', 'C3N-01946-T': 'C3N-01946-N', 'C3L-01237-N': 'C3L-01237-T', 'C3L-04354-N': 'C3L-04354-T', 'C3N-03612-N-need2remove': 'C3N-03612-T', 'C3L-02621-N': 'C3L-02621-T', 'C3L-04354-T': 'C3L-04354-N', 'C3N-03612-T': 'C3N-03612-N-need2remove', 'C3L-02621-T': 'C3L-02621-N', 'C3N-03837-N': 'C3N-03837-T', 'C3N-03027-T': 'C3N-03027-N', 'C3N-01943-T': 'C3N-01943-N', 'C3N-04278-N': 'C3N-04278-T', 'C3N-03027-N': 'C3N-03027-T', 'C3N-04278-T': 'C3N-04278-N', 'C3N-03837-T': 'C3N-03837-N', 'C3N-01943-N': 'C3N-01943-T', 'C3L-02617-T-duplicate': 'C3L-02617-N', 'C3N-03620-T': 'C3N-03620-N', 'C3L-00995-T': 'C3L-00995-N', 'C3N-03620-N': 'C3N-03620-T', 'C3L-02651-T': 'C3L-02651-N', 'C3L-02617-N': 'C3L-02617-T-duplicate', 'C3L-00995-N': 'C3L-00995-T', 'C3L-02651-N': 'C3L-02651-T', 'C3N-03042-T': 'C3N-03042-C', 'C3N-04279-N': 'C3N-04279-T', 'C3N-02279-N': 'C3N-02279-T', 'C3N-02714-T': 'C3N-02714-N', 'C3N-02279-T': 'C3N-02279-N', 'C3N-02714-N': 'C3N-02714-T', 'C3N-04279-T': 'C3N-04279-N', 'C3N-00857-T': 'C3N-00857-N', 'C3N-00857-N': 'C3N-00857-T', 'C3N-03841-N': 'C3N-03841-T', 'C3N-01757-T': 'C3N-01757-C', 'C3N-03841-T': 'C3N-03841-N', 'C3N-03013-T': 'C3N-03013-N', 'C3N-03013-N': 'C3N-03013-T', 'C3N-03878-T': 'C3N-03878-N', 'C3N-01340-T': 'C3N-01340-N', 'C3N-03878-N': 'C3N-03878-T', 'C3N-01754-T': 'C3N-01754-N', 'C3N-01340-N': 'C3N-01340-T', 'C3N-01754-N': 'C3N-01754-T', 'C3N-04276-T': 'C3N-04276-N', 'C3N-03928-T': 'C3N-03928-N', 'C3N-04276-N': 'C3N-04276-T', 'C3N-03888-T': 'C3N-03888-N', 'C3N-03888-N': 'C3N-03888-T', 'C3N-03928-N': 'C3N-03928-T', 'C3L-01138-T': 'C3L-01138-N-need2remove', 'C3N-03488-N': 'C3N-03488-T', 'C3N-03876-T': 'C3N-03876-N', 'C3N-03488-T': 'C3N-03488-N', 'C3N-03876-N': 'C3N-03876-T', 'C3L-01138-N-need2remove': 'C3L-01138-T', 'C3N-04275-T': 'C3N-04275-N', 'C3N-00204-T': 'C3N-00204-N', 'C3L-00994-N': 'C3L-00994-T', 'C3N-04275-N': 'C3N-04275-T', 'C3L-00994-T': 'C3L-00994-N', 'C3N-00204-N': 'C3N-00204-T', 'C3L-03378-T': 'C3L-03378-N', 'C3N-03028-N': 'C3N-03028-T', 'C3N-03008-T': 'C3N-03008-N', 'C3L-03378-N': 'C3L-03378-T', 'C3N-01944-T': 'C3N-01944-N', 'C3N-03008-N': 'C3N-03008-T', 'C3N-01944-N': 'C3N-01944-T', 'C3N-03028-T': 'C3N-03028-N', 'C3N-00519-T': 'C3N-00519-N', 'C3N-00829-N': 'C3N-00829-T', 'C3N-00295-N': 'C3N-00295-T', 'C3N-00295-T': 'C3N-00295-N', 'C3N-00519-N': 'C3N-00519-T', 'C3N-00829-T': 'C3N-00829-N', 'C3N-00306-T': 'C3N-00306-N', 'C3L-02617-T': 'C3L-02617-N', 'C3N-00306-N': 'C3N-00306-T', 'C3N-00498-T': 'C3N-00498-N', 'C3N-00307-N': 'C3N-00307-T', 'C3N-00498-N': 'C3N-00498-T', 'C3N-00307-T': 'C3N-00307-N', 'C3L-00994-N-duplicate': 'C3L-00994-T', 'C3N-03042-C': 'C3N-03042-T', 'C3L-02617-C': 'C3L-02617-T-duplicate', 'C3L-00994-C': 'C3L-00994-T', 'C3N-01757-C': 'C3N-01757-T', 'C3L-02617-N-duplicate': 'C3L-02617-T-duplicate', 'C3L-02617-T-duplicate2': 'C3L-02617-N'}\n",
      "{'C3L-00994-T', 'C3N-00204-T', 'C3N-01645-T', 'C3L-00997-N', 'C3N-00829-T', 'C3N-01946-N', 'C3N-01756-N', 'C3N-02727-T', 'C3N-00498-T', 'C3N-03490-T', 'C3L-00994-N', 'C3N-01943-N', 'C3N-01948.RNA-Seq.A.hg38', 'C3N-00306-T', 'C3N-00822-T', 'C3N-00857-T', 'C3N-01337-N', 'C3N-01340-N', 'C3N-01944-T', 'C3N-02279-T', 'C3N-00846-T', 'C3N-00299-T', 'C3N-00300.RNA-Seq.T.hg38', 'C3L-00997-T', 'C3N-00871-T', 'C3N-01859-T', 'C3N-01946-T', 'C3N-00825-T', 'C3N-01620-T', 'C3N-01943-T', 'C3L-00999-T', 'C3N-00307-T', 'C3N-01945-T', 'C3N-02333-T', 'C3N-01758-T', 'C3N-00519-T', 'C3N-01757-T', 'C3L-00977-T', 'C3L-00987-T', 'C3N-01337-T', 'C3N-02279-N', 'C3N-02275-T', 'C3N-02714-T', 'C3N-00295-T', 'C3N-01339-T', 'C3N-01643-T-need2remove', 'C3N-01755-T', 'C3N-02700-T', 'C3N-02727.RNA-Seq.A.hg38', 'C3N-02275-N-need2remove', 'C3N-00828-T', 'C3N-00297-T', 'C3N-00306-N', 'C3N-01338-T', 'C3N-01338-N', 'C3N-01340-T', 'C3N-01645-N', 'C3L-00995-N', 'C3L-00999-N', 'C3N-01752-T', 'C3N-01858-T', 'C3N-01754-T', 'C3N-00519-N', 'C3N-00498-N', 'C3N-02925.RNA-Seq.A.hg38', 'C3N-00307-N', 'C3N-01948-T', 'C3N-00204-N', 'C3N-01754-N', 'C3N-01756-T', 'C3N-01944-N', 'C3N-02716-T'}\n",
      "phosphoproteome (186, 13009)\n",
      "proteome (186, 8846)\n",
      "LSCC\n",
      "['acetylome', 'phosphoproteome', 'proteome', 'ubiquitylome']\n",
      "['Normal', 'Tumor']\n",
      "{'C3L-02665.T': 'C3L-02665.N', 'C3L-02665.N': 'C3L-02665.T', 'C3N-02575.T': 'C3N-02575.N', 'C3N-02575.N': 'C3N-02575.T', 'C3L-02963.T': 'C3L-02963.N', 'C3L-02963.N': 'C3L-02963.T', 'C3N-02285.T': 'C3N-02285.N', 'C3N-02285.N': 'C3N-02285.T', 'C3N-03875.T': 'C3N-03875.N', 'C3N-01017.T': 'C3N-01017.N', 'C3N-01017.N': 'C3N-01017.T', 'C3L-00081.T': 'C3L-00081.N', 'C3L-02649.T': 'C3L-02649.N', 'C3L-02649.N': 'C3L-02649.T', 'C3N-02252.T': 'C3N-02252.N', 'C3N-02252.N': 'C3N-02252.T', 'C3L-03963.T': 'C3L-03963.N', 'C3L-03963.N': 'C3L-03963.T', 'C3N-03072.T': 'C3N-03072.N', 'C3L-02969.T': 'C3L-02969.N', 'C3L-02969.N': 'C3L-02969.T', 'C3N-03662.T': 'C3N-03662.N', 'C3N-03662.N': 'C3N-03662.T', 'C3N-03051.T': 'C3N-03051.N', 'C3N-03051.N': 'C3N-03051.T', 'C3N-02434.T': 'C3N-02434.N', 'C3L-03961.T': 'C3L-03961.N', 'C3L-03961.N': 'C3L-03961.T', 'C3L-04391.T': 'C3L-04391.N', 'C3L-04391.N': 'C3L-04391.T', 'C3L-02619.T': 'C3L-02619.N', 'C3L-02619.N': 'C3L-02619.T', 'C3N-02494.T': 'C3N-02494.N', 'C3N-03880.T': 'C3N-03880.N', 'C3N-03092.T': 'C3N-03092.N', 'C3N-03092.N': 'C3N-03092.T', 'C3L-02669.T': 'C3L-02669.N', 'C3L-02669.N': 'C3L-02669.T', 'C3N-03877.T': 'C3N-03877.N', 'C3N-03877.N': 'C3N-03877.T', 'C3N-01846.T': 'C3N-01846.N', 'C3L-02964.T': 'C3L-02964.N', 'C3L-02964.N': 'C3L-02964.T', 'C3L-01606.T': 'C3L-01606.N', 'C3L-01606.N': 'C3L-01606.T', 'C3L-02164.T': 'C3L-02164.N', 'C3L-02164.N': 'C3L-02164.T', 'C3N-02374.T': 'C3N-02374.N', 'C3N-02374.N': 'C3N-02374.T', 'C3N-02675.T': 'C3N-02675.N', 'C3N-02523.T': 'C3N-02523.N', 'C3N-02523.N': 'C3N-02523.T', 'C3L-02130.T': 'C3L-02130.N', 'C3N-02426.T': 'C3N-02426.N', 'C3N-03486.T': 'C3N-03486.N', 'C3N-03486.N': 'C3N-03486.T', 'C3N-04127.T': 'C3N-04127.N', 'C3L-02951.T': 'C3L-02951.N', 'C3L-02951.N': 'C3L-02951.T', 'C3L-02629.T': 'C3L-02629.N', 'C3L-02629.N': 'C3L-02629.T', 'C3L-04013.T': 'C3L-04013.N', 'C3L-04013.N': 'C3L-04013.T', 'C3N-00221.T': 'C3N-00221.N', 'C3L-00568.T': 'C3L-00568.N', 'C3L-00568.N': 'C3L-00568.T', 'C3N-04155.T': 'C3N-04155.N', 'C3N-01025.T': 'C3N-01025.N', 'C3N-01025.N': 'C3N-01025.T', 'C3N-02375.T': 'C3N-02375.N', 'C3L-02660.T': 'C3L-02660.N', 'C3L-02660.N': 'C3L-02660.T', 'C3L-01884.T': 'C3L-01884.N', 'C3L-01884.N': 'C3L-01884.T', 'C3L-01285.T': 'C3L-01285.N', 'C3L-01285.N': 'C3L-01285.T', 'C3L-02627.T': 'C3L-02627.N', 'C3L-02627.N': 'C3L-02627.T', 'C3L-00993.T': 'C3L-00993.N', 'C3L-00993.N': 'C3L-00993.T', 'C3N-02425.T': 'C3N-02425.N', 'C3N-02425.N': 'C3N-02425.T', 'C3N-00555.T': 'C3N-00555.N', 'C3N-00555.N': 'C3N-00555.T', 'C3N-01893.T': 'C3N-01893.N', 'C3N-01893.N': 'C3N-01893.T', 'C3N-03093.T': 'C3N-03093.N', 'C3N-03093.N': 'C3N-03093.T', 'C3N-03848.T': 'C3N-03848.N', 'C3N-03204.T': 'C3N-03204.N', 'C3N-02289.T': 'C3N-02289.N', 'C3N-02289.N': 'C3N-02289.T', 'C3L-03962.T': 'C3L-03962.N', 'C3L-03962.N': 'C3L-03962.T', 'C3L-03272.T': 'C3L-03272.N', 'C3N-01411.T': 'C3N-01411.N', 'C3N-01411.N': 'C3N-01411.T', 'C3N-03882.T': 'C3N-03882.N', 'C3L-01000.T': 'C3L-01000.N', 'C3L-01000.N': 'C3L-01000.T', 'C3L-02358.T': 'C3L-02358.N', 'C3L-02358.N': 'C3L-02358.T', 'C3L-02168.T': 'C3L-02168.N', 'C3L-02625.T': 'C3L-02625.N', 'C3L-02625.N': 'C3L-02625.T', 'C3N-01028.T': 'C3N-01028.N', 'C3N-01028.N': 'C3N-01028.T', 'C3N-02435.T': 'C3N-02435.N', 'C3N-02435.N': 'C3N-02435.T', 'C3N-03441.T': 'C3N-03441.N', 'C3N-03441.N': 'C3N-03441.T', 'C3L-04071.T': 'C3L-04071.N', 'C3L-04071.N': 'C3L-04071.T', 'C3N-01020.T': 'C3N-01020.N', 'C3N-01020.N': 'C3N-01020.T', 'C3N-00497.T': 'C3N-00497.N', 'C3N-00497.N': 'C3N-00497.T', 'C3L-02891.T': 'C3L-02891.N', 'C3L-02891.N': 'C3L-02891.T', 'C3L-00927.T': 'C3L-00927.N', 'C3L-00927.N': 'C3L-00927.T', 'C3N-03886.T': 'C3N-03886.N', 'C3N-03886.N': 'C3N-03886.T', 'C3L-04014.T': 'C3L-04014.N', 'C3N-03076.T': 'C3N-03076.N', 'C3N-03425.T': 'C3N-03425.N', 'C3N-03425.N': 'C3N-03425.T', 'C3L-02163.T': 'C3L-02163.N', 'C3L-02163.N': 'C3L-02163.T', 'C3N-04124.T': 'C3N-04124.N', 'C3N-04124.N': 'C3N-04124.T', 'C3N-01892.T': 'C3N-01892.N', 'C3N-01892.N': 'C3N-01892.T', 'C3L-00603.T': 'C3L-00603.N', 'C3L-00603.N': 'C3L-00603.T', 'C3L-03965.T': 'C3L-03965.N', 'C3N-03851.T': 'C3N-03851.N', 'C3N-03851.N': 'C3N-03851.T'}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'C3N-03880.T', 'C3L-01606.N', 'C3L-02669.N', 'C3L-02964.T', 'C3L-00993.N', 'C3N-01017.T', 'C3L-02891.N', 'C3L-00568.T', 'C3L-01884.T', 'C3N-04124.N', 'C3L-00603.T', 'C3L-03965.N', 'C3N-04162.T', 'C3N-02575.N', 'C3L-02168.N', 'C3L-04014.T', 'C3L-02669.T', 'C3L-02168.T', 'C3N-02375.T', 'C3L-04013.N', 'C3L-00081.N', 'C3N-01411.T', 'C3L-00904.T', 'C3L-02546.N', 'C3L-02951.N', 'C3N-02494.T', 'C3L-02968.T', 'C3N-03072.T', 'C3L-00503.RNA-Seq.T.hg38', 'C3L-02619.N', 'C3L-00445.T', 'C3L-01285.T', 'C3L-01838.T', 'C3N-01020.T', 'C3L-00415.T', 'C3L-02951.T', 'C3L-01663.T', 'C3N-02252.T', 'C3L-02127.N', 'C3N-01025.N', 'C3N-00221.T', 'C3N-03882.T', 'C3L-02164.T', 'C3L-02170.T', 'C3N-03441.T', 'C3N-00497.T', 'C3L-02349.T', 'C3L-02650.N', 'C3L-02625.T', 'C3N-03424.T', 'C3N-04127.T', 'C3N-01893.T', 'C3L-02649.T', 'C3N-03093.T', 'C3N-02523.T', 'C3N-03076.T', 'C3N-01892.N', 'C3N-03886.T', 'C3L-01606.T', 'C3N-04155.T', 'C3N-03875.T', 'C3N-01017.N', 'C3N-01028.T', 'C3N-02426.N', 'C3L-00927.T', 'C3N-01018.RNA-Seq.T.hg38', 'C3N-03204.T', 'C3L-02546.T', 'C3N-03486.T', 'C3L-02627.T', 'C3L-00568.N', 'C3L-02891.T', 'C3N-03051.T', 'C3N-03882.N', 'C3N-03486.N', 'C3L-02127.T', 'C3L-02130.N', 'C3N-00309.RNA-Seq.T.hg38', 'C3L-04391.N', 'C3N-00211.T', 'C3N-02339.T', 'C3L-00415.N', 'C3L-03961.N', 'C3L-02648.N', 'C3L-02619.T', 'C3N-02425.N', 'C3L-01455.N', 'C3L-03272.N', 'C3L-02648.T', 'C3L-00923.T', 'C3N-00247.T', 'C3N-02289.N', 'C3N-01846.T', 'C3N-03076.N', 'C3L-02625.N', 'C3L-02963.N', 'C3N-02425.T', 'C3N-02675.N', 'C3N-00221.N', 'C3N-03877.N', 'C3N-00839.RNA-Seq.T.hg38', 'C3L-00445.N', 'C3L-00927.N', 'C3L-02163.T', 'C3N-02285.T', 'C3N-04127.N', 'C3L-02646.T', 'C3L-02624.RNA-Seq.A.hg38', 'C3L-01000.T', 'C3N-00555.T', 'C3L-04391.T', 'C3N-02435.T', 'C3L-02163.N', 'C3N-03848.T', 'C3L-00923.N', 'C3L-03272.T', 'C3L-03963.T', 'C3L-04014.N', 'C3N-02252.N', 'C3L-02969.N', 'C3N-03051.N', 'C3N-01194.N', 'C3N-01893.N', 'C3L-03962.T', 'C3L-03961.T', 'C3L-02552.T', 'C3N-02434.T', 'C3N-00497.N', 'C3N-02494.N', 'C3N-02284.T', 'C3N-03425.N', 'C3L-01000.N', 'C3L-02358.T', 'C3N-01020.N', 'C3N-02575.T', 'C3N-03851.N', 'C3N-04124.T', 'C3L-01455.T', 'C3L-00993.T', 'C3N-00555.N', 'C3N-01411.N', 'C3N-02288.N', 'C3L-04071.T', 'C3N-01018.RNA-Seq.A.hg38', 'C3N-02426.T', 'C3N-01892.T', 'C3N-03092.N', 'C3L-02660.N', 'C3L-01884.N', 'C3N-03880.N', 'C3L-03965.T', 'C3N-02288.T', 'C3L-01285.N', 'C3L-02665.N', 'C3N-03424.N', 'C3N-03851.T', 'C3L-02358.N', 'C3L-03962.N', 'C3N-02434.N', 'C3L-03678.T', 'C3N-02435.N', 'C3N-01028.N', 'C3N-03848.N', 'C3L-02650.T', 'C3L-02627.N', 'C3L-02964.N', 'C3N-02283.T', 'C3N-02300.RNA-Seq.A.hg38', 'C3N-02374.N', 'C3N-02285.N', 'C3L-02649.N', 'C3L-00603.N', 'C3N-03425.T', 'C3N-02523.N', 'C3N-03093.N', 'C3L-02969.T', 'C3L-02629.N', 'C3N-02289.T', 'C3N-02300.T', 'C3L-02624.RNA-Seq.T.hg38', 'C3N-03441.N', 'C3N-01194.T', 'C3L-00503.RNA-Seq.A.hg38', 'C3L-01663.N', 'C3L-02130.T', 'C3L-04071.N', 'C3L-04013.T', 'C3L-02968.N', 'C3L-02349.N', 'C3N-03662.T', 'C3L-00965.T', 'C3L-02660.T', 'C3L-02963.T', 'C3L-00904.N', 'C3L-02552.N', 'C3N-01025.T', 'C3N-02339.N', 'C3N-02374.T', 'C3N-03877.T', 'C3N-01846.N', 'C3N-03204.N', 'C3L-02164.N', 'C3L-00081.T', 'C3L-03963.N', 'C3N-03875.N', 'C3N-02675.T', 'C3N-04162.N', 'C3N-02375.N'}\n",
      "acetylome (143, 3048)\n",
      "phosphoproteome (143, 23791)\n",
      "proteome (143, 9883)\n",
      "ubiquitylome (143, 6828)\n",
      "LUAD\n",
      "['acetylome', 'phosphoproteome', 'proteome']\n",
      "['Normal', 'Tumor']\n",
      "{'C3N-01799.T': 'C3N-01799.N', 'C3N-01799.N': 'C3N-01799.T', 'C3L-01890.T': 'C3L-01890.N', 'C3L-01890.N': 'C3L-01890.T', 'C3N-00572.T': 'C3N-00572.N', 'C3N-00572.N': 'C3N-00572.T', 'C3N-02423.T': 'C3N-02423.N', 'C3N-02423.N': 'C3N-02423.T', 'C3N-02729.T': 'C3N-02729.N', 'C3N-02729.N': 'C3N-02729.T', 'C3L-00263.T': 'C3L-00263.N', 'C3L-00263.N': 'C3L-00263.T', 'C3N-01410.T': 'C3N-01410.N', 'C3N-01410.N': 'C3N-01410.T', 'C3N-00578.T': 'C3N-00578.N', 'C3N-00578.N': 'C3N-00578.T', 'C3N-02587.T': 'C3N-02587.N', 'C3N-02587.N': 'C3N-02587.T', 'C3L-00893.T': 'C3L-00893.N', 'C3L-00893.N': 'C3L-00893.T', 'C3N-01488.T': 'C3N-01488.N', 'C3N-01488.N': 'C3N-01488.T', 'C3N-01413.T': 'C3N-01413.N', 'C3N-01413.N': 'C3N-01413.T', 'C3N-01030.T': 'C3N-01030.N', 'C3N-01030.N': 'C3N-01030.T', 'C3N-02588.T': 'C3N-02588.N', 'C3N-02588.N': 'C3N-02588.T', 'C3N-00552.T': 'C3N-00552.N', 'C3N-00552.N': 'C3N-00552.T', 'C3L-01889.T': 'C3L-01889.N', 'C3L-01889.N': 'C3L-01889.T', 'C3N-00169.T': 'C3N-00169.N', 'C3N-00169.N': 'C3N-00169.T', 'C3L-00422.T': 'C3L-00422.N', 'C3L-00422.N': 'C3L-00422.T', 'C3L-00083.T': 'C3L-00083.N', 'C3L-00083.N': 'C3L-00083.T', 'C3N-00551.T': 'C3N-00551.N', 'C3N-00551.N': 'C3N-00551.T', 'C3N-02089.T': 'C3N-02089.N', 'C3N-02089.N': 'C3N-02089.T', 'C3L-01682.T': 'C3L-01682.N', 'C3L-01682.N': 'C3L-01682.T', 'C3N-01016.T': 'C3N-01016.N', 'C3N-01016.N': 'C3N-01016.T', 'C3N-00580.T': 'C3N-00580.N', 'C3N-00580.N': 'C3N-00580.T', 'C3N-02000.T': 'C3N-02000.N', 'C3N-02000.N': 'C3N-02000.T', 'C3N-01489.T': 'C3N-01489.N', 'C3N-01489.N': 'C3N-01489.T', 'C3N-00737.T': 'C3N-00737.N', 'C3N-00737.N': 'C3N-00737.T', 'C3N-01405.T': 'C3N-01405.N', 'C3N-01405.N': 'C3N-01405.T', 'C3N-01416.T': 'C3N-01416.N', 'C3N-01416.N': 'C3N-01416.T', 'C3N-02149.T': 'C3N-02149.N', 'C3N-02149.N': 'C3N-02149.T', 'C3N-02424.T': 'C3N-02424.N', 'C3N-02424.N': 'C3N-02424.T', 'C3L-00279.T': 'C3L-00279.N', 'C3L-00279.N': 'C3L-00279.T', 'C3N-02379.T': 'C3N-02379.N', 'C3L-00095.T': 'C3L-00095.N', 'C3L-00095.N': 'C3L-00095.T', 'C3N-02572.T': 'C3N-02572.N', 'C3N-02572.N': 'C3N-02572.T', 'C3N-00559.T': 'C3N-00559.N', 'C3N-00559.N': 'C3N-00559.T', 'C3L-02345.T': 'C3L-02345.N', 'C3L-02345.N': 'C3L-02345.T', 'C3L-00093.T': 'C3L-00093.N', 'C3L-00093.N': 'C3L-00093.T', 'C3L-00510.T': 'C3L-00510.N', 'C3L-00510.N': 'C3L-00510.T', 'C3L-00913.T': 'C3L-00913.N', 'C3L-00913.N': 'C3L-00913.T', 'C3L-00412.T': 'C3L-00412.N', 'C3L-00412.N': 'C3L-00412.T', 'C3N-01415.T': 'C3N-01415.N', 'C3N-01415.N': 'C3N-01415.T', 'C3N-00549.T': 'C3N-00549.N', 'C3N-00549.N': 'C3N-00549.T', 'C3N-02433.T': 'C3N-02433.N', 'C3N-02433.N': 'C3N-02433.T', 'C3N-01021.T': 'C3N-01021.N', 'C3N-01021.N': 'C3N-01021.T', 'C3N-00203.T': 'C3N-00203.N', 'C3N-00203.N': 'C3N-00203.T', 'C3L-01330.T': 'C3L-01330.N', 'C3L-01330.N': 'C3L-01330.T', 'C3N-00574.T': 'C3N-00574.N', 'C3N-00574.N': 'C3N-00574.T', 'C3N-01071.T': 'C3N-01071.N', 'C3N-01071.N': 'C3N-01071.T', 'C3N-00550.T': 'C3N-00550.N', 'C3N-00550.N': 'C3N-00550.T', 'C3N-00167.T': 'C3N-00167.N', 'C3N-00167.N': 'C3N-00167.T', 'C3L-00001.T': 'C3L-00001.N', 'C3L-00001.N': 'C3L-00001.T', 'C3L-01632.T': 'C3L-01632.N', 'C3L-01632.N': 'C3L-01632.T', 'C3N-02155.T': 'C3N-02155.N', 'C3N-02155.N': 'C3N-02155.T', 'C3N-00217.T': 'C3N-00217.N', 'C3N-00217.N': 'C3N-00217.T', 'C3N-02002.T': 'C3N-02002.N', 'C3N-02002.N': 'C3N-02002.T', 'C3N-02145.T': 'C3N-02145.N', 'C3N-02145.N': 'C3N-02145.T', 'C3N-02586.T': 'C3N-02586.N', 'C3N-02586.N': 'C3N-02586.T', 'C3L-02219.T': 'C3L-02219.N', 'C3L-02219.N': 'C3L-02219.T', 'C3N-00180.T': 'C3N-00180.N', 'C3N-00180.N': 'C3N-00180.T', 'C3N-00704.T': 'C3N-00704.N', 'C3N-00704.N': 'C3N-00704.T', 'C3N-02158.T': 'C3N-02158.N', 'C3N-02158.N': 'C3N-02158.T', 'C3N-02421.T': 'C3N-02421.N', 'C3N-02421.N': 'C3N-02421.T', 'C3N-00556.T': 'C3N-00556.N', 'C3N-00556.N': 'C3N-00556.T', 'C3N-01072.T': 'C3N-01072.N', 'C3N-01072.N': 'C3N-01072.T', 'C3L-00094.T': 'C3L-00094.N', 'C3L-00094.N': 'C3L-00094.T', 'C3L-00144.T': 'C3L-00144.N', 'C3L-00144.N': 'C3L-00144.T', 'C3N-00223.T': 'C3N-00223.N', 'C3N-00223.N': 'C3N-00223.T', 'C3N-02380.T': 'C3N-02380.N', 'C3L-01924.T': 'C3L-01924.N', 'C3L-01924.N': 'C3L-01924.T', 'C3N-00433.T': 'C3N-00433.N', 'C3N-00433.N': 'C3N-00433.T', 'C3L-01683.T': 'C3L-01683.N', 'C3L-01683.N': 'C3L-01683.T', 'C3N-00293.T': 'C3N-00293.N', 'C3N-00293.N': 'C3N-00293.T', 'C3N-02380.N': 'C3N-02380.T', 'C3L-00009.T': 'C3L-00009.N', 'C3L-00009.N': 'C3L-00009.T', 'C3N-00546.T': 'C3N-00546.N', 'C3N-00546.N': 'C3N-00546.T', 'C3N-01414.T': 'C3N-01414.N', 'C3N-01414.N': 'C3N-01414.T', 'C3N-02087.T': 'C3N-02087.N', 'C3N-02087.N': 'C3N-02087.T', 'C3L-00140.T': 'C3L-00140.N', 'C3L-00140.N': 'C3L-00140.T', 'C3N-02529.T': 'C3N-02529.N', 'C3N-02529.N': 'C3N-02529.T', 'C3L-00368.T': 'C3L-00368.N', 'C3L-00368.N': 'C3L-00368.T', 'C3N-02067.T': 'C3N-02067.N', 'C3N-02067.N': 'C3N-02067.T', 'C3L-00080.T': 'C3L-00080.N', 'C3L-00080.N': 'C3L-00080.T', 'C3N-00547.T': 'C3N-00547.N', 'C3N-00547.N': 'C3N-00547.T', 'C3N-01023.T': 'C3N-01023.N', 'C3N-01023.N': 'C3N-01023.T', 'C3N-01024.T': 'C3N-01024.N', 'C3N-01024.N': 'C3N-01024.T', 'C3L-00973.T': 'C3L-00973.N', 'C3L-00973.N': 'C3L-00973.T', 'C3L-02348.T': 'C3L-02348.N', 'C3L-02348.N': 'C3L-02348.T', 'C3L-02508.T': 'C3L-02508.N', 'C3L-02508.N': 'C3L-02508.T', 'C3N-00560.T': 'C3N-00560.N', 'C3N-00560.N': 'C3N-00560.T', 'C3L-00604.T': 'C3L-00604.N', 'C3L-00604.N': 'C3L-00604.T', 'C3N-00199.T': 'C3N-00199.N', 'C3N-00199.N': 'C3N-00199.T', 'C3N-00579.T': 'C3N-00579.N', 'C3N-00579.N': 'C3N-00579.T', 'C3N-02582.T': 'C3N-02582.N', 'C3N-02582.N': 'C3N-02582.T', 'C3L-02350.T': 'C3L-02350.N', 'C3L-02350.N': 'C3L-02350.T', 'C3N-00738.T': 'C3N-00738.N', 'C3N-00738.N': 'C3N-00738.T', 'C3N-00959.T': 'C3N-00959.N', 'C3N-00959.N': 'C3N-00959.T', 'C3N-02003.T': 'C3N-02003.N', 'C3N-02003.N': 'C3N-02003.T', 'C3N-02379.N': 'C3N-02379.T', 'C3N-00175.T': 'C3N-00175.N', 'C3N-00175.N': 'C3N-00175.T', 'C3N-01823.T': 'C3N-01823.N', 'C3N-01823.N': 'C3N-01823.T', 'C3L-02549.T': 'C3L-02549.N', 'C3L-02549.N': 'C3L-02549.T', 'C3L-02365.T': 'C3L-02365.N', 'C3L-02365.N': 'C3L-02365.T'}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'C3L-00510.T', 'C3L-00368.N', 'C3N-03212.RNA-Seq.A.hg38', 'C3N-02242.RNA-Seq.A.hg38', 'C3N-01417.RNA-Seq.A.hg38', 'C3N-03233.RNA-Seq.A.hg38', 'C3N-01019.RNA-Seq.A.hg38', 'C3L-02169.RNA-Seq.A.hg38', 'C3N-00959.N', 'C3N-00169.T', 'C3L-00263.T', 'C3L-00001.N', 'C3N-00180.T', 'C3N-00550.N', 'C3N-02234.RNA-Seq.A.hg38', 'C3N-02149.N', 'C3N-02587.N', 'C3N-02240.RNA-Seq.A.hg38', 'C3N-00552.T', 'C3N-01024.N', 'C3N-00574.N', 'C3N-02002.N', 'C3N-00547.N', 'C3N-01419.RNA-Seq.T.hg38', 'C3N-02153.RNA-Seq.A.hg38', 'C3N-02672.RNA-Seq.T.hg38', 'C3N-02150.RNA-Seq.T.hg38', 'C3N-00572.T', 'C3N-01416.T', 'C3N-02281.RNA-Seq.T.hg38', 'C3N-02758.RNA-Seq.T.hg38', 'C3N-03205.RNA-Seq.T.hg38', 'C3L-00893.N', 'C3N-02234.RNA-Seq.T.hg38', 'C3N-00579.N', 'C3N-02144.RNA-Seq.T.hg38', '11LU013.T', 'C3L-00144.T', 'C3N-00559.N', 'C3N-01021.N', 'C3N-01409.RNA-Seq.A.hg38', 'C3N-02729.T', 'C3L-01683.T', 'C3N-01016.N', 'C3N-01022.RNA-Seq.T.hg38', 'C3N-02380.N', 'C3N-00578.N', 'C3L-01924.N', 'C3N-00551.T', 'C3L-00412.N', 'C3N-00199.T', 'C3N-02000.T', 'C3N-02433.T', 'C3L-02345.N', 'C3N-01410.T', 'C3N-00546.T', 'C3N-02282.RNA-Seq.A.hg38', 'C3N-01489.T', 'C3L-00009.T', 'C3N-02715.RNA-Seq.T.hg38', 'C3L-01889.N', 'C3L-01632.T', 'C3N-00579.T', 'C3L-00080.T', 'C3L-02549.N', 'C3L-01330.N', 'C3N-02718.RNA-Seq.T.hg38', 'C3N-02728.RNA-Seq.A.hg38', 'C3L-00368.T', 'C3N-02379.N', 'C3N-00704.T', 'C3N-02146.RNA-Seq.A.hg38', 'C3L-02601.RNA-Seq.T.hg38', 'C3N-00738.N', 'C3N-02087.T', 'C3N-02433.N', 'C3L-00913.N', 'C3N-02282.RNA-Seq.T.hg38', 'C3L-02350.N', 'C3L-02661.RNA-Seq.T.hg38', 'C3N-01021.T', 'C3N-03420.RNA-Seq.T.hg38', 'C3N-00550.T', 'C3N-02721.RNA-Seq.A.hg38', 'C3N-02230.RNA-Seq.T.hg38', '11LU035.T', 'C3L-00083.T', 'C3N-01413.N', 'C3N-02142.RNA-Seq.T.hg38', 'C3N-02718.RNA-Seq.A.hg38', 'C3N-02729.N', 'C3N-02588.T', 'C3N-00180.N', 'C3N-02926.RNA-Seq.A.hg38', 'C3L-02560.RNA-Seq.T.hg38', 'C3N-02764.RNA-Seq.A.hg38', 'C3N-01415.N', 'C3L-00144.N', 'C3N-02089.N', 'C3N-00738.T', 'C3N-02582.N', 'C3N-01413.T', 'C3N-02588.N', 'C3L-02508.T', 'C3N-00433.T', 'C3N-02929.RNA-Seq.T.hg38', 'C3N-02090.RNA-Seq.A.hg38', 'C3N-01408.RNA-Seq.A.hg38', 'C3N-02088.RNA-Seq.T.hg38', 'C3N-00293.N', 'C3L-02508.N', 'C3L-00009.N', 'C3L-01890.N', 'C3N-01022.RNA-Seq.A.hg38', 'C3N-00175.T', 'C3N-02003.T', 'C3L-03262.RNA-Seq.T.hg38', 'C3N-02141.RNA-Seq.A.hg38', 'C3L-02549.T', 'C3N-02142.RNA-Seq.A.hg38', 'C3N-00549.N', 'C3N-02529.N', 'C3L-01632.N', 'C3L-02560.RNA-Seq.A.hg38', 'C3N-00549.T', 'C3L-00263.N', 'C3N-01023.T', 'C3N-02087.N', 'C3N-01019.RNA-Seq.T.hg38', 'C3N-01489.N', 'C3N-02572.T', '11LU016.T', 'C3L-00279.T', 'C3N-01071.T', 'C3L-02219.N', 'C3N-01030.N', 'C3L-02834.RNA-Seq.T.hg38', 'C3N-01799.N', 'C3N-00547.T', 'C3N-02230.RNA-Seq.A.hg38', 'C3N-02421.N', 'C3N-02587.T', 'C3L-02957.RNA-Seq.A.hg38', 'C3L-00412.T', 'C3L-00140.T', 'C3N-02586.N', 'C3L-00446.RNA-Seq.T.hg38', 'C3N-00572.N', 'C3N-02728.RNA-Seq.T.hg38', 'C3N-02764.RNA-Seq.T.hg38', 'C3N-02379.T', 'C3N-01842.T', 'C3N-02145.N', 'C3N-02240.RNA-Seq.T.hg38', 'C3L-02661.RNA-Seq.A.hg38', 'C3L-02834.RNA-Seq.A.hg38', 'C3N-00433.N', 'C3N-02000.N', 'C3N-00203.N', 'C3N-03420.RNA-Seq.A.hg38', 'C3N-02089.T', 'C3N-00169.N', 'C3N-02146.RNA-Seq.T.hg38', 'C3N-01405.N', 'C3N-03222.RNA-Seq.T.hg38', 'C3N-01415.T', 'C3N-02973.RNA-Seq.A.hg38', 'C3L-01682.T', 'C3N-02237.RNA-Seq.A.hg38', 'C3N-02926.RNA-Seq.T.hg38', 'C3N-02158.T', 'C3L-02348.T', 'C3N-01409.RNA-Seq.T.hg38', 'C3N-01408.RNA-Seq.T.hg38', 'C3L-00422.T', 'C3N-02929.RNA-Seq.A.hg38', 'C3L-00973.N', 'C3N-02424.N', 'C3N-02582.T', 'C3N-00546.N', 'C3N-00580.T', 'C3N-01074.T', 'C3L-02348.N', 'C3N-02003.N', 'C3N-00545.RNA-Seq.T.hg38', 'C3N-03212.RNA-Seq.T.hg38', 'C3L-01924.T', 'C3L-00913.T', 'C3N-02672.RNA-Seq.A.hg38', 'C3L-02616.RNA-Seq.T.hg38', 'C3L-02957.RNA-Seq.T.hg38', 'C3N-00560.T', 'C3L-02601.RNA-Seq.A.hg38', 'C3N-02290.RNA-Seq.T.hg38', 'C3N-00217.N', 'C3N-02090.RNA-Seq.T.hg38', 'C3L-02365.N', 'C3N-01419.RNA-Seq.A.hg38', 'C3N-02141.RNA-Seq.T.hg38', 'C3N-00560.N', 'C3N-02242.RNA-Seq.T.hg38', 'C3N-01030.T', 'C3N-01799.T', 'C3L-00093.T', 'C3N-03238.RNA-Seq.A.hg38', 'C3L-00444.RNA-Seq.A.hg38', 'C3L-02365.T', 'C3N-02923.RNA-Seq.T.hg38', 'C3N-02757.RNA-Seq.T.hg38', 'C3N-02757.RNA-Seq.A.hg38', 'C3N-00203.T', 'C3L-00140.N', 'C3N-01023.N', 'C3N-01823.N', 'C3L-00973.T', 'C3N-00223.N', 'C3N-02380.T', 'C3N-01072.N', 'C3N-01823.T', 'C3L-02165.RNA-Seq.A.hg38', 'C3N-02586.T', 'C3N-00175.N', 'C3N-00580.N', 'C3N-00704.N', 'C3L-00604.N', 'C3N-02145.T', 'C3L-01889.T', 'C3N-02923.RNA-Seq.A.hg38', 'C3N-01417.RNA-Seq.T.hg38', 'C3L-00604.T', 'C3N-02067.T', 'C3N-02721.RNA-Seq.T.hg38', 'C3N-00294.T', 'C3N-00959.T', '11LU022.T', 'C3L-00080.N', 'C3L-00094.T', 'C3L-01682.N', 'C3N-02144.RNA-Seq.A.hg38', 'C3N-00556.N', 'C3N-02149.T', 'C3N-02067.N', 'C3N-02290.RNA-Seq.A.hg38', 'C3N-02423.N', 'C3L-00093.N', 'C3L-00444.RNA-Seq.T.hg38', 'C3N-01414.N', 'C3N-00559.T', 'C3L-01890.T', 'C3N-00199.N', 'C3N-02758.RNA-Seq.A.hg38', 'C3N-02237.RNA-Seq.T.hg38', 'C3N-00551.N', 'C3L-02559.RNA-Seq.T.hg38', 'C3L-00446.RNA-Seq.A.hg38', 'C3N-00552.N', 'C3N-00167.T', 'C3L-00279.N', 'C3N-00293.T', 'C3L-00095.T', 'C3N-00223.T', 'C3L-02129.RNA-Seq.T.hg38', 'C3L-01683.N', 'C3N-01024.T', 'C3N-01405.T', 'C3N-00545.RNA-Seq.A.hg38', 'C3N-00556.T', 'C3N-03233.RNA-Seq.T.hg38', 'C3N-01488.N', 'C3N-00578.T', 'C3N-02002.T', 'C3N-01416.N', 'C3N-03238.RNA-Seq.T.hg38', 'C3N-01410.N', 'C3L-00422.N', 'C3N-03205.RNA-Seq.A.hg38', 'C3L-00001.T', 'C3N-03222.RNA-Seq.A.hg38', 'C3L-02345.T', 'C3N-02424.T', 'C3L-01330.T', 'C3L-01862.T', 'C3N-01414.T', 'C3L-00510.N', 'C3N-02150.RNA-Seq.A.hg38', 'C3N-02423.T', 'C3L-02559.RNA-Seq.A.hg38', 'C3N-02088.RNA-Seq.A.hg38', 'C3N-02155.T', 'C3L-02219.T', 'C3N-02155.N', 'C3N-02281.RNA-Seq.A.hg38', 'C3N-00737.T', 'C3N-00737.N', 'C3N-02529.T', 'C3N-01071.N', 'C3L-02165.RNA-Seq.T.hg38', 'C3N-01016.T', 'C3L-03262.RNA-Seq.A.hg38', 'C3N-00167.N', 'C3L-00893.T', 'C3N-00217.T', 'C3N-02158.N', 'C3N-02973.RNA-Seq.T.hg38', 'C3N-02715.RNA-Seq.A.hg38', 'C3N-02421.T', 'C3N-01488.T', 'C3L-00095.N', 'C3L-02350.T', 'C3N-02572.N', 'C3L-02169.RNA-Seq.T.hg38', 'C3N-00574.T', 'C3L-00083.N', 'C3N-01072.T', 'C3L-00094.N', 'C3N-02153.RNA-Seq.T.hg38', 'C3L-02616.RNA-Seq.A.hg38', 'C3N-02422.T'}\n",
      "acetylome (211, 2750)\n",
      "phosphoproteome (211, 22345)\n",
      "proteome (211, 8988)\n",
      "UCEC\n",
      "['acetylome', 'phosphoproteome', 'proteome']\n",
      "['Adjacent_normal', 'Enriched_normal', 'Myometrium_normal', 'Tumor']\n",
      "{'S001': 'S105', 'S017': 'S106', 'S021': 'S130', 'S022': 'S107', 'S023': 'S108', 'S024': 'S131', 'S026': 'S109', 'S027': 'S132', 'S028': 'S133', 'S034': 'S111', 'S038': 'S112', 'S041': 'S113', 'S042': 'S114', 'S045': 'S115', 'S046': 'S116', 'S048': 'S117', 'S049': 'S118', 'S051': 'S119', 'S053': 'S120', 'S054': 'S121', 'S055': 'S122', 'S057': 'S123', 'S060': 'S134', 'S067': 'S124', 'S075': 'S125', 'S079': 'S126', 'S086': 'S127', 'S087': 'S128', 'S091': 'S135', 'S096': 'S129', 'S105': 'S001', 'S106': 'S017', 'S107': 'S022', 'S108': 'S023', 'S109': 'S026', 'S111': 'S034', 'S112': 'S038', 'S113': 'S041', 'S114': 'S042', 'S115': 'S045', 'S116': 'S046', 'S117': 'S048', 'S118': 'S049', 'S119': 'S051', 'S120': 'S053', 'S121': 'S054', 'S122': 'S055', 'S123': 'S057', 'S124': 'S067', 'S125': 'S075', 'S126': 'S079', 'S127': 'S086', 'S128': 'S087', 'S129': 'S096', 'S130': 'S021', 'S131': 'S024', 'S132': 'S027', 'S133': 'S028', 'S134': 'S060', 'S135': 'S091'}\n",
      "{'C3L-02468.RNA-Seq.A.hg38', 'S101', 'C3N-01847.RNA-Seq.T.hg38', 'C3L-00586.RNA-Seq.prov.CPT0023630009', 'C3N-00860.RNA-Seq.T.hg38', 'C3N-01847.RNA-Seq.A.hg38', 'C3L-02216.RNA-Seq.T.hg38', 'C3L-00356.RNA-Seq.T.hg38', 'C3N-01876.RNA-Seq.A.hg38', 'S071', 'S044', 'S026', 'C3L-00932.RNA-Seq.prov.CPT0027120006', 'C3L-01633.RNA-Seq.T.hg38', 'S021', 'C3L-02468.RNA-Seq.T.hg38', 'S076', 'S080', 'C3L-01749.RNA-Seq.T.hg38', 'S095', 'C3L-02121.RNA-Seq.T.hg38', 'S008', 'S050', 'C3L-02347.RNA-Seq.A.hg38', 'S016', 'C3L-02384.RNA-Seq.T.hg38', 'S060', 'C3L-00361.RNA-Seq.prov.CPT0002430005', 'S041', 'C3L-01277.RNA-Seq.T.hg38', 'C3L-01732.RNA-Seq.T.hg38', 'S028', 'C3L-01639.RNA-Seq.A.hg38', 'C3N-01876.RNA-Seq.T.hg38', 'C3N-02012.RNA-Seq.T.hg38', 'C3N-01342.RNA-Seq.T.hg38', 'C3L-00006.RNA-Seq.prov.CPT0001470006', 'C3L-02119.RNA-Seq.T.hg38', 'C3L-00935.RNA-Seq.T.hg38', 'S029', 'C3L-02408.RNA-Seq.A.hg38', 'S081', 'S005', 'C3L-01833.RNA-Seq.T.hg38', 'C3L-00157.RNA-Seq.T.hg38', 'C3L-02409.RNA-Seq.A.hg38', 'C3L-02555.RNA-Seq.T.hg38', 'C3L-01253.RNA-Seq.T.hg38', 'C3L-02412.RNA-Seq.A.hg38', 'C3N-00859.RNA-Seq.T.hg38', 'C3L-00086.RNA-Seq.T.hg38', 'S100', 'C3N-01765.RNA-Seq.A.hg38', 'C3N-01761.RNA-Seq.T.hg38', 'C3N-01875.RNA-Seq.A.hg38', 'C3L-02408.RNA-Seq.T.hg38', 'S036', 'C3L-01664.RNA-Seq.T.hg38', 'S096', 'C3N-00753.RNA-Seq.T.hg38', 'S099', 'S024', 'S087', 'C3L-00361.RNA-Seq.prov.CPT0002450006', 'S007', 'C3L-01284.RNA-Seq.T.hg38', 'C3L-02216.RNA-Seq.A.hg38', 'C3N-00520.RNA-Seq.A.hg38', 'S129', 'C3L-01277.RNA-Seq.A.hg38', 'C3N-01525.RNA-Seq.T.hg38', 'S085', 'S086', 'C3L-01304.RNA-Seq.prov.CPT0064240006', 'C3L-01631.RNA-Seq.T.hg38', 'C3L-00932.RNA-Seq.prov.CPT0027130006', 'C3L-01307.RNA-Seq.prov.CPT0064290010', 'C3L-00935.RNA-Seq.prov.CPT0027230006', 'S069', 'S064', 'C3N-01848.RNA-Seq.A.hg38', 'S102', 'C3L-01967.RNA-Seq.A.hg38', 'S093', 'S037', 'S065', 'C3L-00601.RNA-Seq.prov.CPT0007660006', 'C3N-01765.RNA-Seq.T.hg38', 'C3N-02436.RNA-Seq.T.hg38', 'S030', 'C3L-01639.RNA-Seq.T.hg38', 'C3L-02121.RNA-Seq.A.hg38', 'C3L-02353.RNA-Seq.A.hg38', 'S046', 'C3N-00872.RNA-Seq.A.hg38', 'S014', 'C3L-01604.RNA-Seq.T.hg38', 'S063', 'S073', 'C3L-01739.RNA-Seq.T.hg38', 'C3L-01840.RNA-Seq.T.hg38', 'C3N-01525.RNA-Seq.A.hg38', 'S058', 'S002', 'S056', 'S040', 'C3L-02353.RNA-Seq.T.hg38', 'S098', 'S025', 'C3N-01830.RNA-Seq.T.hg38', 'C3L-01355.RNA-Seq.T.hg38', 'C3N-01171.RNA-Seq.T.hg38', 'S006', 'C3L-01967.RNA-Seq.T.hg38', 'C3N-00155.RNA-Seq.T.hg38', 'C3L-00930.RNA-Seq.T.hg38', 'S084', 'C3N-01877.RNA-Seq.T.hg38', 'S128', 'S067', 'C3L-02347.RNA-Seq.T.hg38', 'S045', 'S066', 'S049', 'S061', 'S070', 'C3L-02411.RNA-Seq.T.hg38', 'C3N-01763.RNA-Seq.T.hg38', 'S033', 'C3N-01873.RNA-Seq.T.hg38', 'C3N-02253.RNA-Seq.A.hg38', 'S124', 'S032', 'C3L-01599.RNA-Seq.T.hg38', 'C3L-01729.RNA-Seq.T.hg38', 'C3L-01274.RNA-Seq.T.hg38', 'C3N-01878.RNA-Seq.T.hg38', 'C3L-00601.RNA-Seq.prov.CPT0007600006', 'C3L-02354.RNA-Seq.T.hg38', 'S088', 'S127', 'C3N-01830.RNA-Seq.A.hg38', 'C3L-00084.RNA-Seq.T.hg38', 'S038', 'S122', 'C3L-00006.RNA-Seq.prov.CPT0001460164', 'C3N-01526.RNA-Seq.A.hg38', 'S068', 'C3N-01875.RNA-Seq.T.hg38', 'C3L-01064.RNA-Seq.T.hg38', 'S020', 'S077', 'S091', 'C3N-01529.RNA-Seq.T.hg38', 'C3L-01277.RNA-Seq.prov.CPT0093170006', 'C3N-00520.RNA-Seq.T.hg38', 'S083', 'C3N-01001.RNA-Seq.T.hg38', 'C3L-00935.RNA-Seq.A.hg38', 'C3N-02436.RNA-Seq.A.hg38', 'C3N-01172.RNA-Seq.T.hg38', 'C3N-00859.RNA-Seq.A.hg38', 'C3L-01247.RNA-Seq.T.hg38', 'S125', 'S059', 'S042', 'C3L-01282.RNA-Seq.prov.CPT0077650009', 'S010', 'C3L-02602.RNA-Seq.A.hg38', 'S003', 'S078', 'C3L-00586.RNA-Seq.prov.CPT0023640006', 'C3L-01913.RNA-Seq.T.hg38', 'S009', 'C3L-02412.RNA-Seq.T.hg38', 'S012', 'S057', 'C3N-01003.RNA-Seq.A.hg38', 'S094', 'C3N-01344.RNA-Seq.A.hg38', 'S031', 'C3L-01282.RNA-Seq.prov.CPT0077660006', 'S072', 'S055', 'C3L-01304.RNA-Seq.prov.CPT0064210009', 'C3N-01526.RNA-Seq.T.hg38', 'S092', 'S027', 'C3L-02357.RNA-Seq.T.hg38', 'C3N-01871.RNA-Seq.T.hg38', 'C3N-02244.RNA-Seq.T.hg38', 'C3N-00872.RNA-Seq.T.hg38', 'S039', 'C3L-01864.RNA-Seq.T.hg38', 'C3N-01843.RNA-Seq.T.hg38', 'S062', 'C3L-00935.RNA-Seq.prov.CPT0027220013', 'C3L-00938.RNA-Seq.T.hg38', 'C3L-02545.RNA-Seq.T.hg38', 'C3N-01848.RNA-Seq.T.hg38', 'C3N-01761.RNA-Seq.A.hg38', 'S018', 'C3L-01277.RNA-Seq.prov.CPT0093180006', 'C3N-01874.RNA-Seq.T.hg38', 'S011', 'C3N-01007.RNA-Seq.T.hg38', 'S019', 'C3N-02249.RNA-Seq.T.hg38', 'S048', 'C3N-00732.RNA-Seq.T.hg38', 'S103', 'C3L-02602.RNA-Seq.T.hg38', 'C3N-00755.RNA-Seq.T.hg38', 'S075', 'C3N-02253.RNA-Seq.T.hg38', 'C3N-01008.RNA-Seq.T.hg38', 'S074', 'C3N-01344.RNA-Seq.T.hg38', 'C3L-02557.RNA-Seq.T.hg38', 'C3L-00943.RNA-Seq.T.hg38', 'S079', 'C3L-02557.RNA-Seq.A.hg38', 'C3L-02409.RNA-Seq.T.hg38', 'C3N-01341.RNA-Seq.T.hg38', 'S090', 'S082', 'C3L-01307.RNA-Seq.prov.CPT0064320006', 'C3L-02395.RNA-Seq.T.hg38', 'C3N-02249.RNA-Seq.A.hg38', 'S097'}\n",
      "acetylome (144, 2184)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phosphoproteome (144, 15983)\n",
      "proteome (144, 9246)\n"
     ]
    }
   ],
   "source": [
    "cancer_to_aggs = {}\n",
    "for cancer, d in cancer_to_model_inputs.items():\n",
    "    if cancer in to_keep:\n",
    "        print(cancer)\n",
    "        metadata_key = 'acetylome' if 'acetylome' in cancer_to_protein[cancer] else 'phosphoproteome'\n",
    "        site_level_agg = prepare_site_level_features(d,\n",
    "                               cancer_to_protein[cancer][metadata_key]['metadata'],\n",
    "                                cancer_type_to_expression[cancer],\n",
    "                                cancer_to_maf[cancer])\n",
    "        \n",
    "#         gene_level_agg = prepare_gene_level_aggregates(d,\n",
    "#                               cancer_to_protein[cancer][metadata_key]['metadata'],\n",
    "#                              cancer_type_to_expression[cancer])\n",
    "        cancer_to_aggs[cancer] = {\n",
    "            'site_level': site_level_agg,\n",
    "#             'gene_level': gene_level_agg\n",
    "        }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cancer_to_protein['CCRCC']['phosphoproteome']['metadata'].loc[['CPT0001550001', 'CPT0001540009']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "case_id               C3N-03875\n",
       "aliquot_id        CPT0237430003\n",
       "tmt_experiment                2\n",
       "tmt_channel                131N\n",
       "sample_type              Normal\n",
       "Name: C3N-03875.N, dtype: object"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_protein['LSCC']['phosphoproteome']['metadata'].loc['C3N-03875.N']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cancer_to_protein['LSCC']['proteome']['processed_data'][['C3N.03875.N']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value</th>\n",
       "      <th>paired_value</th>\n",
       "      <th>paired_difference</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_type</th>\n",
       "      <th>rna_expression</th>\n",
       "      <th>paired_rna_expression</th>\n",
       "      <th>difference_rna_expression</th>\n",
       "      <th>has_mutation</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acetylome</th>\n",
       "      <td>-0.0997</td>\n",
       "      <td>-1.7808</td>\n",
       "      <td>1.6811</td>\n",
       "      <td>tumor</td>\n",
       "      <td>acetylome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acetylome</th>\n",
       "      <td>0.1159</td>\n",
       "      <td>1.8238</td>\n",
       "      <td>-1.7079</td>\n",
       "      <td>tumor</td>\n",
       "      <td>acetylome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>AASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210_acetylome</th>\n",
       "      <td>0.2270</td>\n",
       "      <td>-1.2052</td>\n",
       "      <td>1.4322</td>\n",
       "      <td>tumor</td>\n",
       "      <td>acetylome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ABCE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415_acetylome</th>\n",
       "      <td>1.1944</td>\n",
       "      <td>-0.5094</td>\n",
       "      <td>1.7038</td>\n",
       "      <td>tumor</td>\n",
       "      <td>acetylome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ABCE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431_acetylome</th>\n",
       "      <td>3.1825</td>\n",
       "      <td>-0.3380</td>\n",
       "      <td>3.5205</td>\n",
       "      <td>tumor</td>\n",
       "      <td>acetylome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ABCE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K328_ubiquitylome</th>\n",
       "      <td>-1.6167</td>\n",
       "      <td>-1.8588</td>\n",
       "      <td>0.2421</td>\n",
       "      <td>normal</td>\n",
       "      <td>ubiquitylome</td>\n",
       "      <td>0.639734</td>\n",
       "      <td>1.565725</td>\n",
       "      <td>-0.925991</td>\n",
       "      <td>False</td>\n",
       "      <td>ZSWIM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K276_ubiquitylome</th>\n",
       "      <td>0.6940</td>\n",
       "      <td>0.3584</td>\n",
       "      <td>0.3356</td>\n",
       "      <td>normal</td>\n",
       "      <td>ubiquitylome</td>\n",
       "      <td>6.038561</td>\n",
       "      <td>8.687788</td>\n",
       "      <td>-2.649227</td>\n",
       "      <td>False</td>\n",
       "      <td>ZUP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111_ubiquitylome</th>\n",
       "      <td>0.2040</td>\n",
       "      <td>4.1551</td>\n",
       "      <td>-3.9511</td>\n",
       "      <td>normal</td>\n",
       "      <td>ubiquitylome</td>\n",
       "      <td>5.910446</td>\n",
       "      <td>74.706409</td>\n",
       "      <td>-68.795962</td>\n",
       "      <td>False</td>\n",
       "      <td>ZWINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K2490_ubiquitylome</th>\n",
       "      <td>0.0106</td>\n",
       "      <td>0.0323</td>\n",
       "      <td>-0.0217</td>\n",
       "      <td>normal</td>\n",
       "      <td>ubiquitylome</td>\n",
       "      <td>11.741871</td>\n",
       "      <td>12.478456</td>\n",
       "      <td>-0.736585</td>\n",
       "      <td>False</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K2501_ubiquitylome</th>\n",
       "      <td>0.1164</td>\n",
       "      <td>0.0106</td>\n",
       "      <td>0.1058</td>\n",
       "      <td>normal</td>\n",
       "      <td>ubiquitylome</td>\n",
       "      <td>11.741871</td>\n",
       "      <td>12.478456</td>\n",
       "      <td>-0.736585</td>\n",
       "      <td>False</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6227078 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     value  paired_value  \\\n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet... -0.0997       -1.7808   \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...  0.1159        1.8238   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...  0.2270       -1.2052   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...  1.1944       -0.5094   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...  3.1825       -0.3380   \n",
       "...                                                    ...           ...   \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32... -1.6167       -1.8588   \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...  0.6940        0.3584   \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...  0.2040        4.1551   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...  0.0106        0.0323   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...  0.1164        0.0106   \n",
       "\n",
       "                                                    paired_difference  \\\n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet...             1.6811   \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...            -1.7079   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...             1.4322   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...             1.7038   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...             3.5205   \n",
       "...                                                               ...   \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32...             0.2421   \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...             0.3356   \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...            -3.9511   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...            -0.0217   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...             0.1058   \n",
       "\n",
       "                                                   sample_type     data_type  \\\n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet...       tumor     acetylome   \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...       tumor     acetylome   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...       tumor     acetylome   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...       tumor     acetylome   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...       tumor     acetylome   \n",
       "...                                                        ...           ...   \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32...      normal  ubiquitylome   \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...      normal  ubiquitylome   \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...      normal  ubiquitylome   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...      normal  ubiquitylome   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...      normal  ubiquitylome   \n",
       "\n",
       "                                                    rna_expression  \\\n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet...             NaN   \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...             NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...             NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...             NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...             NaN   \n",
       "...                                                            ...   \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32...        0.639734   \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...        6.038561   \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...        5.910446   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...       11.741871   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...       11.741871   \n",
       "\n",
       "                                                    paired_rna_expression  \\\n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet...                    NaN   \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...                    NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...                    NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...                    NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...                    NaN   \n",
       "...                                                                   ...   \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32...               1.565725   \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...               8.687788   \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...              74.706409   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...              12.478456   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...              12.478456   \n",
       "\n",
       "                                                    difference_rna_expression  \\\n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet...                        NaN   \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...                        NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...                        NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...                        NaN   \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...                        NaN   \n",
       "...                                                                       ...   \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32...                  -0.925991   \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...                  -2.649227   \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...                 -68.795962   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...                  -0.736585   \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...                  -0.736585   \n",
       "\n",
       "                                                    has_mutation    gene  \n",
       "C3L-02665.T_acetylome_A2M:NP_000005.2:K664_acet...         False     A2M  \n",
       "C3L-02665.T_acetylome_AASS:NP_005754.2:K93_acet...         False    AASS  \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K210...         False   ABCE1  \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K415...         False   ABCE1  \n",
       "C3L-02665.T_acetylome_ABCE1:NP_001035809.1:K431...         False   ABCE1  \n",
       "...                                                          ...     ...  \n",
       "C3N-03851.N_ubiquitylome_ZSWIM9:NP_955373.3:K32...         False  ZSWIM9  \n",
       "C3N-03851.N_ubiquitylome_ZUP1:NP_001348118.1:K2...         False    ZUP1  \n",
       "C3N-03851.N_ubiquitylome_ZWINT:NP_008988.2:K111...         False   ZWINT  \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K249...         False   ZZEF1  \n",
       "C3N-03851.N_ubiquitylome_ZZEF1:NP_055928.3:K250...         False   ZZEF1  \n",
       "\n",
       "[6227078 rows x 10 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_to_aggs['LSCC']['site_level']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1175742"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.count_nonzero(pd.isnull(cancer_to_aggs['LSCC']['site_level']['paired_value']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "def combine_gene_site_level(gene_level_agg, site_level_agg):\n",
    "    combined = pd.merge(site_level_agg, gene_level_agg, how='left', left_on='gene', right_index=True)\n",
    "    return combined\n",
    "\n",
    "\n",
    "def combine_cancer_types(cancer_to_aggs):\n",
    "    df = None\n",
    "    for cancer, d in cancer_to_aggs.items():\n",
    "        data = d['site_level']\n",
    "        data['cancer_type'] = [cancer] * data.shape[0]\n",
    "        if df is None:\n",
    "            df = data\n",
    "        else:\n",
    "            df = pd.concat((df, data), axis=0)\n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value</th>\n",
       "      <th>paired_value</th>\n",
       "      <th>paired_difference</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_type</th>\n",
       "      <th>rna_expression</th>\n",
       "      <th>paired_rna_expression</th>\n",
       "      <th>difference_rna_expression</th>\n",
       "      <th>has_mutation</th>\n",
       "      <th>gene</th>\n",
       "      <th>cancer_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1:T491_phosphoproteome</th>\n",
       "      <td>0.265550</td>\n",
       "      <td>-0.000111</td>\n",
       "      <td>0.265661</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>15.287382</td>\n",
       "      <td>8.086648</td>\n",
       "      <td>7.200734</td>\n",
       "      <td>False</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1:S462_phosphoproteome</th>\n",
       "      <td>0.288098</td>\n",
       "      <td>0.087886</td>\n",
       "      <td>0.200212</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>8.623087</td>\n",
       "      <td>10.856750</td>\n",
       "      <td>-2.233663</td>\n",
       "      <td>False</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1:Y452_phosphoproteome</th>\n",
       "      <td>0.540065</td>\n",
       "      <td>-0.229424</td>\n",
       "      <td>0.769489</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>8.623087</td>\n",
       "      <td>10.856750</td>\n",
       "      <td>-2.233663</td>\n",
       "      <td>False</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S12_phosphoproteome</th>\n",
       "      <td>0.785612</td>\n",
       "      <td>-0.366038</td>\n",
       "      <td>1.151650</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>AAED1</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAGAB:NP_001258814.1:S201;S202_phosphoproteome</th>\n",
       "      <td>0.307438</td>\n",
       "      <td>-0.163721</td>\n",
       "      <td>0.471159</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>10.663889</td>\n",
       "      <td>15.487737</td>\n",
       "      <td>-4.823848</td>\n",
       "      <td>False</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZXDC_proteome</th>\n",
       "      <td>0.587000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>UCEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZYG11B_proteome</th>\n",
       "      <td>-0.477000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>UCEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZYX_proteome</th>\n",
       "      <td>-0.422000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>UCEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZZEF1_proteome</th>\n",
       "      <td>-0.093100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>UCEC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZZZ3_proteome</th>\n",
       "      <td>0.055700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>UCEC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26026874 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       value  paired_value  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...  0.265550     -0.000111   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  0.288098      0.087886   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  0.540065     -0.229424   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...  0.785612     -0.366038   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  0.307438     -0.163721   \n",
       "...                                                      ...           ...   \n",
       "S153_proteome_ZXDC_proteome                         0.587000           NaN   \n",
       "S153_proteome_ZYG11B_proteome                      -0.477000           NaN   \n",
       "S153_proteome_ZYX_proteome                         -0.422000           NaN   \n",
       "S153_proteome_ZZEF1_proteome                       -0.093100           NaN   \n",
       "S153_proteome_ZZZ3_proteome                         0.055700           NaN   \n",
       "\n",
       "                                                    paired_difference  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...           0.265661   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...           0.200212   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...           0.769489   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...           1.151650   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....           0.471159   \n",
       "...                                                               ...   \n",
       "S153_proteome_ZXDC_proteome                                       NaN   \n",
       "S153_proteome_ZYG11B_proteome                                     NaN   \n",
       "S153_proteome_ZYX_proteome                                        NaN   \n",
       "S153_proteome_ZZEF1_proteome                                      NaN   \n",
       "S153_proteome_ZZZ3_proteome                                       NaN   \n",
       "\n",
       "                                                   sample_type  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       tumor   \n",
       "...                                                        ...   \n",
       "S153_proteome_ZXDC_proteome                             normal   \n",
       "S153_proteome_ZYG11B_proteome                           normal   \n",
       "S153_proteome_ZYX_proteome                              normal   \n",
       "S153_proteome_ZZEF1_proteome                            normal   \n",
       "S153_proteome_ZZZ3_proteome                             normal   \n",
       "\n",
       "                                                          data_type  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  phosphoproteome   \n",
       "...                                                             ...   \n",
       "S153_proteome_ZXDC_proteome                                proteome   \n",
       "S153_proteome_ZYG11B_proteome                              proteome   \n",
       "S153_proteome_ZYX_proteome                                 proteome   \n",
       "S153_proteome_ZZEF1_proteome                               proteome   \n",
       "S153_proteome_ZZZ3_proteome                                proteome   \n",
       "\n",
       "                                                    rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       15.287382   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...        8.623087   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...        8.623087   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...             NaN   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       10.663889   \n",
       "...                                                            ...   \n",
       "S153_proteome_ZXDC_proteome                                    NaN   \n",
       "S153_proteome_ZYG11B_proteome                                  NaN   \n",
       "S153_proteome_ZYX_proteome                                     NaN   \n",
       "S153_proteome_ZZEF1_proteome                                   NaN   \n",
       "S153_proteome_ZZZ3_proteome                                    NaN   \n",
       "\n",
       "                                                    paired_rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...               8.086648   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...              10.856750   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...              10.856750   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...                    NaN   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....              15.487737   \n",
       "...                                                                   ...   \n",
       "S153_proteome_ZXDC_proteome                                           NaN   \n",
       "S153_proteome_ZYG11B_proteome                                         NaN   \n",
       "S153_proteome_ZYX_proteome                                            NaN   \n",
       "S153_proteome_ZZEF1_proteome                                          NaN   \n",
       "S153_proteome_ZZZ3_proteome                                           NaN   \n",
       "\n",
       "                                                    difference_rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...                   7.200734   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...                  -2.233663   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...                  -2.233663   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...                        NaN   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....                  -4.823848   \n",
       "...                                                                       ...   \n",
       "S153_proteome_ZXDC_proteome                                               NaN   \n",
       "S153_proteome_ZYG11B_proteome                                             NaN   \n",
       "S153_proteome_ZYX_proteome                                                NaN   \n",
       "S153_proteome_ZZEF1_proteome                                              NaN   \n",
       "S153_proteome_ZZZ3_proteome                                               NaN   \n",
       "\n",
       "                                                    has_mutation    gene  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...         False    A1CF   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...         False    AAAS   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...         False    AAAS   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...         False   AAED1   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....         False   AAGAB   \n",
       "...                                                          ...     ...   \n",
       "S153_proteome_ZXDC_proteome                                False    ZXDC   \n",
       "S153_proteome_ZYG11B_proteome                              False  ZYG11B   \n",
       "S153_proteome_ZYX_proteome                                 False     ZYX   \n",
       "S153_proteome_ZZEF1_proteome                               False   ZZEF1   \n",
       "S153_proteome_ZZZ3_proteome                                False    ZZZ3   \n",
       "\n",
       "                                                   cancer_type  \n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       CCRCC  \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       CCRCC  \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       CCRCC  \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...       CCRCC  \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       CCRCC  \n",
       "...                                                        ...  \n",
       "S153_proteome_ZXDC_proteome                               UCEC  \n",
       "S153_proteome_ZYG11B_proteome                             UCEC  \n",
       "S153_proteome_ZYX_proteome                                UCEC  \n",
       "S153_proteome_ZZEF1_proteome                              UCEC  \n",
       "S153_proteome_ZZZ3_proteome                               UCEC  \n",
       "\n",
       "[26026874 rows x 11 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combined = combine_gene_site_level(gene_level_agg, site_level_agg)\n",
    "combined = combine_cancer_types(cancer_to_aggs)\n",
    "combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_drivers(combined):\n",
    "    is_driver, is_expanded = [], []\n",
    "    expanded_drivers = set(drivers['Gene'])\n",
    "    for gene, cancer in zip(combined['gene'], combined['cancer_type']):\n",
    "        is_driver.append(gene in cancer_to_driver_genes[cancer].index)\n",
    "        is_expanded.append(gene in expanded_drivers)\n",
    "    combined['driver'] = is_driver\n",
    "    combined['expanded_driver'] = is_expanded\n",
    "    return combined\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value</th>\n",
       "      <th>paired_value</th>\n",
       "      <th>paired_difference</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_type</th>\n",
       "      <th>rna_expression</th>\n",
       "      <th>paired_rna_expression</th>\n",
       "      <th>difference_rna_expression</th>\n",
       "      <th>has_mutation</th>\n",
       "      <th>gene</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>driver</th>\n",
       "      <th>expanded_driver</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1:T491_phosphoproteome</th>\n",
       "      <td>0.265550</td>\n",
       "      <td>-0.000111</td>\n",
       "      <td>0.265661</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>15.287382</td>\n",
       "      <td>8.086648</td>\n",
       "      <td>7.200734</td>\n",
       "      <td>False</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1:S462_phosphoproteome</th>\n",
       "      <td>0.288098</td>\n",
       "      <td>0.087886</td>\n",
       "      <td>0.200212</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>8.623087</td>\n",
       "      <td>10.856750</td>\n",
       "      <td>-2.233663</td>\n",
       "      <td>False</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1:Y452_phosphoproteome</th>\n",
       "      <td>0.540065</td>\n",
       "      <td>-0.229424</td>\n",
       "      <td>0.769489</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>8.623087</td>\n",
       "      <td>10.856750</td>\n",
       "      <td>-2.233663</td>\n",
       "      <td>False</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S12_phosphoproteome</th>\n",
       "      <td>0.785612</td>\n",
       "      <td>-0.366038</td>\n",
       "      <td>1.151650</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>AAED1</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAGAB:NP_001258814.1:S201;S202_phosphoproteome</th>\n",
       "      <td>0.307438</td>\n",
       "      <td>-0.163721</td>\n",
       "      <td>0.471159</td>\n",
       "      <td>tumor</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>10.663889</td>\n",
       "      <td>15.487737</td>\n",
       "      <td>-4.823848</td>\n",
       "      <td>False</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZXDC_proteome</th>\n",
       "      <td>0.587000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZYG11B_proteome</th>\n",
       "      <td>-0.477000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZYX_proteome</th>\n",
       "      <td>-0.422000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZZEF1_proteome</th>\n",
       "      <td>-0.093100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153_proteome_ZZZ3_proteome</th>\n",
       "      <td>0.055700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>normal</td>\n",
       "      <td>proteome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26026874 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       value  paired_value  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...  0.265550     -0.000111   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  0.288098      0.087886   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  0.540065     -0.229424   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...  0.785612     -0.366038   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  0.307438     -0.163721   \n",
       "...                                                      ...           ...   \n",
       "S153_proteome_ZXDC_proteome                         0.587000           NaN   \n",
       "S153_proteome_ZYG11B_proteome                      -0.477000           NaN   \n",
       "S153_proteome_ZYX_proteome                         -0.422000           NaN   \n",
       "S153_proteome_ZZEF1_proteome                       -0.093100           NaN   \n",
       "S153_proteome_ZZZ3_proteome                         0.055700           NaN   \n",
       "\n",
       "                                                    paired_difference  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...           0.265661   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...           0.200212   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...           0.769489   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...           1.151650   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....           0.471159   \n",
       "...                                                               ...   \n",
       "S153_proteome_ZXDC_proteome                                       NaN   \n",
       "S153_proteome_ZYG11B_proteome                                     NaN   \n",
       "S153_proteome_ZYX_proteome                                        NaN   \n",
       "S153_proteome_ZZEF1_proteome                                      NaN   \n",
       "S153_proteome_ZZZ3_proteome                                       NaN   \n",
       "\n",
       "                                                   sample_type  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...       tumor   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       tumor   \n",
       "...                                                        ...   \n",
       "S153_proteome_ZXDC_proteome                             normal   \n",
       "S153_proteome_ZYG11B_proteome                           normal   \n",
       "S153_proteome_ZYX_proteome                              normal   \n",
       "S153_proteome_ZZEF1_proteome                            normal   \n",
       "S153_proteome_ZZZ3_proteome                             normal   \n",
       "\n",
       "                                                          data_type  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  phosphoproteome   \n",
       "...                                                             ...   \n",
       "S153_proteome_ZXDC_proteome                                proteome   \n",
       "S153_proteome_ZYG11B_proteome                              proteome   \n",
       "S153_proteome_ZYX_proteome                                 proteome   \n",
       "S153_proteome_ZZEF1_proteome                               proteome   \n",
       "S153_proteome_ZZZ3_proteome                                proteome   \n",
       "\n",
       "                                                    rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       15.287382   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...        8.623087   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...        8.623087   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...             NaN   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       10.663889   \n",
       "...                                                            ...   \n",
       "S153_proteome_ZXDC_proteome                                    NaN   \n",
       "S153_proteome_ZYG11B_proteome                                  NaN   \n",
       "S153_proteome_ZYX_proteome                                     NaN   \n",
       "S153_proteome_ZZEF1_proteome                                   NaN   \n",
       "S153_proteome_ZZZ3_proteome                                    NaN   \n",
       "\n",
       "                                                    paired_rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...               8.086648   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...              10.856750   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...              10.856750   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...                    NaN   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....              15.487737   \n",
       "...                                                                   ...   \n",
       "S153_proteome_ZXDC_proteome                                           NaN   \n",
       "S153_proteome_ZYG11B_proteome                                         NaN   \n",
       "S153_proteome_ZYX_proteome                                            NaN   \n",
       "S153_proteome_ZZEF1_proteome                                          NaN   \n",
       "S153_proteome_ZZZ3_proteome                                           NaN   \n",
       "\n",
       "                                                    difference_rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...                   7.200734   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...                  -2.233663   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...                  -2.233663   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...                        NaN   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....                  -4.823848   \n",
       "...                                                                       ...   \n",
       "S153_proteome_ZXDC_proteome                                               NaN   \n",
       "S153_proteome_ZYG11B_proteome                                             NaN   \n",
       "S153_proteome_ZYX_proteome                                                NaN   \n",
       "S153_proteome_ZZEF1_proteome                                              NaN   \n",
       "S153_proteome_ZZZ3_proteome                                               NaN   \n",
       "\n",
       "                                                    has_mutation    gene  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...         False    A1CF   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...         False    AAAS   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...         False    AAAS   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...         False   AAED1   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....         False   AAGAB   \n",
       "...                                                          ...     ...   \n",
       "S153_proteome_ZXDC_proteome                                False    ZXDC   \n",
       "S153_proteome_ZYG11B_proteome                              False  ZYG11B   \n",
       "S153_proteome_ZYX_proteome                                 False     ZYX   \n",
       "S153_proteome_ZZEF1_proteome                               False   ZZEF1   \n",
       "S153_proteome_ZZZ3_proteome                                False    ZZZ3   \n",
       "\n",
       "                                                   cancer_type  driver  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       CCRCC   False   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       CCRCC   False   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       CCRCC   False   \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...       CCRCC   False   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       CCRCC   False   \n",
       "...                                                        ...     ...   \n",
       "S153_proteome_ZXDC_proteome                               UCEC   False   \n",
       "S153_proteome_ZYG11B_proteome                             UCEC   False   \n",
       "S153_proteome_ZYX_proteome                                UCEC   False   \n",
       "S153_proteome_ZZEF1_proteome                              UCEC   False   \n",
       "S153_proteome_ZZZ3_proteome                               UCEC   False   \n",
       "\n",
       "                                                    expanded_driver  \n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...            False  \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...            False  \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...            False  \n",
       "C3L-00004.T_phosphoproteome_AAED1:NP_714542.1:S...            False  \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....            False  \n",
       "...                                                             ...  \n",
       "S153_proteome_ZXDC_proteome                                   False  \n",
       "S153_proteome_ZYG11B_proteome                                 False  \n",
       "S153_proteome_ZYX_proteome                                    False  \n",
       "S153_proteome_ZZEF1_proteome                                  False  \n",
       "S153_proteome_ZZZ3_proteome                                   False  \n",
       "\n",
       "[26026874 rows x 13 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## add drivers\n",
    "# combined['driver'] = [0 if g not in ucec_genes.index else 1 for g in combined['gene']]\n",
    "# combined['expanded_driver'] = [0 if g not in drivers['Gene'] else 1 for g in combined['gene']]\n",
    "# combined\n",
    "combined = add_drivers(combined)\n",
    "combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "for c in combined.columns:\n",
    "    for dtype in sorted(set(combined['data_type'])):\n",
    "        f = combined[combined['data_type']==dtype]\n",
    "        print(f.shape, dtype, c, np.count_nonzero(\n",
    "            pd.isnull(f[c])) / f.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value</th>\n",
       "      <th>paired_value</th>\n",
       "      <th>paired_difference</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>gene</th>\n",
       "      <th>data_type</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>has_mutation</th>\n",
       "      <th>rna_expression</th>\n",
       "      <th>paired_rna_expression</th>\n",
       "      <th>difference_rna_expression</th>\n",
       "      <th>driver</th>\n",
       "      <th>expanded_driver</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1:T491_phosphoproteome</th>\n",
       "      <td>0.265550</td>\n",
       "      <td>-0.000111</td>\n",
       "      <td>0.265661</td>\n",
       "      <td>tumor</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>15.287382</td>\n",
       "      <td>8.086648</td>\n",
       "      <td>7.200734</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1:S462_phosphoproteome</th>\n",
       "      <td>0.288098</td>\n",
       "      <td>0.087886</td>\n",
       "      <td>0.200212</td>\n",
       "      <td>tumor</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>8.623087</td>\n",
       "      <td>10.856750</td>\n",
       "      <td>-2.233663</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1:Y452_phosphoproteome</th>\n",
       "      <td>0.540065</td>\n",
       "      <td>-0.229424</td>\n",
       "      <td>0.769489</td>\n",
       "      <td>tumor</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>8.623087</td>\n",
       "      <td>10.856750</td>\n",
       "      <td>-2.233663</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAGAB:NP_001258814.1:S201;S202_phosphoproteome</th>\n",
       "      <td>0.307438</td>\n",
       "      <td>-0.163721</td>\n",
       "      <td>0.471159</td>\n",
       "      <td>tumor</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>10.663889</td>\n",
       "      <td>15.487737</td>\n",
       "      <td>-4.823848</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00004.T_phosphoproteome_AAGAB:NP_001258814.1:S202_phosphoproteome</th>\n",
       "      <td>-0.294636</td>\n",
       "      <td>0.282808</td>\n",
       "      <td>-0.577444</td>\n",
       "      <td>tumor</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>phosphoproteome</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>False</td>\n",
       "      <td>10.663889</td>\n",
       "      <td>15.487737</td>\n",
       "      <td>-4.823848</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S096_proteome_ZXDC_proteome</th>\n",
       "      <td>0.637000</td>\n",
       "      <td>0.420000</td>\n",
       "      <td>0.217000</td>\n",
       "      <td>tumor</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>proteome</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>17.059799</td>\n",
       "      <td>9.967964</td>\n",
       "      <td>7.091835</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S096_proteome_ZYG11B_proteome</th>\n",
       "      <td>0.537000</td>\n",
       "      <td>0.341000</td>\n",
       "      <td>0.196000</td>\n",
       "      <td>tumor</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>proteome</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>11.874810</td>\n",
       "      <td>15.654434</td>\n",
       "      <td>-3.779625</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S096_proteome_ZYX_proteome</th>\n",
       "      <td>-0.338000</td>\n",
       "      <td>1.730000</td>\n",
       "      <td>-2.068000</td>\n",
       "      <td>tumor</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>proteome</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>19.686493</td>\n",
       "      <td>57.144577</td>\n",
       "      <td>-37.458084</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S096_proteome_ZZEF1_proteome</th>\n",
       "      <td>-0.214000</td>\n",
       "      <td>-0.151000</td>\n",
       "      <td>-0.063000</td>\n",
       "      <td>tumor</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>proteome</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>7.169003</td>\n",
       "      <td>10.606031</td>\n",
       "      <td>-3.437028</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S096_proteome_ZZZ3_proteome</th>\n",
       "      <td>-0.071700</td>\n",
       "      <td>0.224000</td>\n",
       "      <td>-0.295700</td>\n",
       "      <td>tumor</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>proteome</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>False</td>\n",
       "      <td>17.452804</td>\n",
       "      <td>14.850757</td>\n",
       "      <td>2.602047</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7624952 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       value  paired_value  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...  0.265550     -0.000111   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  0.288098      0.087886   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  0.540065     -0.229424   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  0.307438     -0.163721   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814.... -0.294636      0.282808   \n",
       "...                                                      ...           ...   \n",
       "S096_proteome_ZXDC_proteome                         0.637000      0.420000   \n",
       "S096_proteome_ZYG11B_proteome                       0.537000      0.341000   \n",
       "S096_proteome_ZYX_proteome                         -0.338000      1.730000   \n",
       "S096_proteome_ZZEF1_proteome                       -0.214000     -0.151000   \n",
       "S096_proteome_ZZZ3_proteome                        -0.071700      0.224000   \n",
       "\n",
       "                                                    paired_difference  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...           0.265661   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...           0.200212   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...           0.769489   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....           0.471159   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....          -0.577444   \n",
       "...                                                               ...   \n",
       "S096_proteome_ZXDC_proteome                                  0.217000   \n",
       "S096_proteome_ZYG11B_proteome                                0.196000   \n",
       "S096_proteome_ZYX_proteome                                  -2.068000   \n",
       "S096_proteome_ZZEF1_proteome                                -0.063000   \n",
       "S096_proteome_ZZZ3_proteome                                 -0.295700   \n",
       "\n",
       "                                                   sample_type    gene  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       tumor    A1CF   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       tumor    AAAS   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       tumor    AAAS   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       tumor   AAGAB   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       tumor   AAGAB   \n",
       "...                                                        ...     ...   \n",
       "S096_proteome_ZXDC_proteome                              tumor    ZXDC   \n",
       "S096_proteome_ZYG11B_proteome                            tumor  ZYG11B   \n",
       "S096_proteome_ZYX_proteome                               tumor     ZYX   \n",
       "S096_proteome_ZZEF1_proteome                             tumor   ZZEF1   \n",
       "S096_proteome_ZZZ3_proteome                              tumor    ZZZ3   \n",
       "\n",
       "                                                          data_type  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  phosphoproteome   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....  phosphoproteome   \n",
       "...                                                             ...   \n",
       "S096_proteome_ZXDC_proteome                                proteome   \n",
       "S096_proteome_ZYG11B_proteome                              proteome   \n",
       "S096_proteome_ZYX_proteome                                 proteome   \n",
       "S096_proteome_ZZEF1_proteome                               proteome   \n",
       "S096_proteome_ZZZ3_proteome                                proteome   \n",
       "\n",
       "                                                   cancer_type  has_mutation  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       CCRCC         False   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       CCRCC         False   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...       CCRCC         False   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       CCRCC         False   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       CCRCC         False   \n",
       "...                                                        ...           ...   \n",
       "S096_proteome_ZXDC_proteome                               UCEC         False   \n",
       "S096_proteome_ZYG11B_proteome                             UCEC         False   \n",
       "S096_proteome_ZYX_proteome                                UCEC         False   \n",
       "S096_proteome_ZZEF1_proteome                              UCEC         False   \n",
       "S096_proteome_ZZZ3_proteome                               UCEC         False   \n",
       "\n",
       "                                                    rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...       15.287382   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...        8.623087   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...        8.623087   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       10.663889   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....       10.663889   \n",
       "...                                                            ...   \n",
       "S096_proteome_ZXDC_proteome                              17.059799   \n",
       "S096_proteome_ZYG11B_proteome                            11.874810   \n",
       "S096_proteome_ZYX_proteome                               19.686493   \n",
       "S096_proteome_ZZEF1_proteome                              7.169003   \n",
       "S096_proteome_ZZZ3_proteome                              17.452804   \n",
       "\n",
       "                                                    paired_rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...               8.086648   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...              10.856750   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...              10.856750   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....              15.487737   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....              15.487737   \n",
       "...                                                                   ...   \n",
       "S096_proteome_ZXDC_proteome                                      9.967964   \n",
       "S096_proteome_ZYG11B_proteome                                   15.654434   \n",
       "S096_proteome_ZYX_proteome                                      57.144577   \n",
       "S096_proteome_ZZEF1_proteome                                    10.606031   \n",
       "S096_proteome_ZZZ3_proteome                                     14.850757   \n",
       "\n",
       "                                                    difference_rna_expression  \\\n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...                   7.200734   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...                  -2.233663   \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...                  -2.233663   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....                  -4.823848   \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....                  -4.823848   \n",
       "...                                                                       ...   \n",
       "S096_proteome_ZXDC_proteome                                          7.091835   \n",
       "S096_proteome_ZYG11B_proteome                                       -3.779625   \n",
       "S096_proteome_ZYX_proteome                                         -37.458084   \n",
       "S096_proteome_ZZEF1_proteome                                        -3.437028   \n",
       "S096_proteome_ZZZ3_proteome                                          2.602047   \n",
       "\n",
       "                                                    driver  expanded_driver  \n",
       "C3L-00004.T_phosphoproteome_A1CF:NP_001185747.1...   False            False  \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...   False            False  \n",
       "C3L-00004.T_phosphoproteome_AAAS:NP_001166937.1...   False            False  \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....   False            False  \n",
       "C3L-00004.T_phosphoproteome_AAGAB:NP_001258814....   False            False  \n",
       "...                                                    ...              ...  \n",
       "S096_proteome_ZXDC_proteome                          False            False  \n",
       "S096_proteome_ZYG11B_proteome                        False            False  \n",
       "S096_proteome_ZYX_proteome                           False            False  \n",
       "S096_proteome_ZZEF1_proteome                         False            False  \n",
       "S096_proteome_ZZZ3_proteome                          False            False  \n",
       "\n",
       "[7624952 rows x 13 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## filter out samples with no paired value and only include tumor to avoid biased training\n",
    "filtered_combined = combined[~pd.isnull(combined['paired_value'])]\n",
    "filtered_combined = filtered_combined[filtered_combined['sample_type']=='tumor']\n",
    "filtered_combined = filtered_combined[filtered_combined['data_type']!='ubiquitylome']\n",
    "filtered_combined = filtered_combined[~pd.isnull(filtered_combined['paired_rna_expression'])]\n",
    "\n",
    "to_keep = ['value', 'paired_value', 'paired_difference', 'sample_type', 'gene', 'data_type', 'cancer_type',\n",
    "           'has_mutation', 'rna_expression', 'paired_rna_expression', 'difference_rna_expression',\n",
    "           'driver', 'expanded_driver']\n",
    "\n",
    "filtered_combined = filtered_combined[to_keep]\n",
    "filtered_combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "def collapse_by_dtype_and_gene(df):\n",
    "    dtypes = sorted(set(df['data_type']))\n",
    "    \n",
    "\n",
    "    categorical = ['cancer_type']\n",
    "    \n",
    "    \n",
    "    df['collapse'] = [f'{dtype}_{gene}_{c}_{sample.split(\"_\")[0]}' for dtype, gene, sample, c in zip(\n",
    "                                                                               df['data_type'],\n",
    "                                                                               df['gene'],\n",
    "                                                                               df.index, df['cancer_type'])]\n",
    "    collapse_to_cancer_type = {'_'.join(c.split('_')[1:]):cancer\n",
    "                               for c, cancer in zip(df['collapse'], df['cancer_type'])}\n",
    "    df = df.groupby('collapse').mean()\n",
    "    shared_cols = ['rna_expression', 'paired_rna_expression', 'difference_rna_expression',\n",
    "                   'has_mutation', 'driver', 'expanded_driver']\n",
    "#     shared = df[shared_cols]\n",
    "#     shared.index = ['_'.join(x.split('_')[1:]) for x in shared.index]\n",
    "#     shared \n",
    "\n",
    "        \n",
    "#     return df\n",
    "#     print(Counter(df.index).most_common()[:5])\n",
    "    dfs = []\n",
    "    for dtype in dtypes:\n",
    "        mask = [True if dtype in x else False for x in df.index]\n",
    "        filtered = df[mask].copy()\n",
    "        filtered.index = ['_'.join(x.split('_')[1:]) for x in filtered.index]\n",
    "        filtered.columns = [f'{dtype}_{c}' if c not in shared_cols else c for c in filtered.columns]\n",
    "        dfs.append(filtered)\n",
    "    \n",
    "    combined = None\n",
    "    for i, df in enumerate(dfs):\n",
    "#         print(dtypes[i], Counter(df.index).most_common()[:5])\n",
    "        df = df[~df.index.duplicated(keep='first')]\n",
    "        if combined is None:\n",
    "            combined = df\n",
    "        else:\n",
    "#             print(df.index[:5], combined.index[:5])\n",
    "#             print(len(set(df.index).intersection(set(combined.index))))\n",
    "            combined = pd.merge(combined, df, right_index=True, left_index=True, how='inner')\n",
    "#     print(Counter(combined.index).most_common()[:5])\n",
    "#     return combined\n",
    "    combined = combined[[c for c in combined.columns if c[-2:] not in ['_x', '_y']]]\n",
    "    \n",
    "    combined['cancer_type'] = [collapse_to_cancer_type[c] for c in combined.index]\n",
    "    \n",
    "#     return combined\n",
    "    \n",
    "    cat_columns = []\n",
    "    cat_data = None\n",
    "    for c in categorical:\n",
    "        ls = combined[c]\n",
    "        value_map = {c:i for i, c in enumerate(sorted(set(ls)))}\n",
    "        print(value_map)\n",
    "        r_value_map = {v:k for k, v in value_map.items()}\n",
    "        zeros = np.zeros((combined.shape[0], len(value_map)))\n",
    "        for i, val in enumerate(ls):\n",
    "            zeros[i, value_map[val]] = 1\n",
    "        \n",
    "        if cat_data is None:\n",
    "            cat_data = zeros\n",
    "        else:\n",
    "            cat_data = np.concatenate((cat_data, zeros), axis=1)\n",
    "        cat_columns += sorted(set(ls))\n",
    "    cat_df = pd.DataFrame(data=cat_data, columns=cat_columns, index=combined.index)\n",
    "#     print(Counter(combined['LSCC']).most_common())\n",
    "#     combined = combined[[c for c in combined.columns if c != 'cancer_type']]\n",
    "    \n",
    "    \n",
    "    combined = pd.merge(left=combined, right=cat_df, right_index=True, left_index=True)\n",
    "    combined = combined[[c for c in combined.columns if c != 'cancer_type']]\n",
    "\n",
    "#     return combined\n",
    "    print(combined.columns)\n",
    "    ## filter down shared columns\n",
    "#     cols = [c for c in combined.columns if c[:-2] not in shared_cols]\n",
    "#     print(cols)\n",
    "#     cols += [c for c in combined.columns if c[:-2] in shared_cols if c[-1]=='x']\n",
    "#     print(cols)\n",
    "#     combined = combined[cols]\n",
    "#     combined.columns = [c if c[-1]!='x' else c[:-2] for c in combined.columns]\n",
    "\n",
    "    return combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'LSCC': 0, 'LUAD': 1, 'UCEC': 2}\n",
      "Index(['acetylome_value', 'acetylome_paired_value',\n",
      "       'acetylome_paired_difference', 'phosphoproteome_value',\n",
      "       'phosphoproteome_paired_value', 'phosphoproteome_paired_difference',\n",
      "       'proteome_value', 'proteome_paired_value', 'proteome_paired_difference',\n",
      "       'has_mutation', 'rna_expression', 'paired_rna_expression',\n",
      "       'difference_rna_expression', 'driver', 'expanded_driver', 'LSCC',\n",
      "       'LUAD', 'UCEC'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>acetylome_value</th>\n",
       "      <th>acetylome_paired_value</th>\n",
       "      <th>acetylome_paired_difference</th>\n",
       "      <th>phosphoproteome_value</th>\n",
       "      <th>phosphoproteome_paired_value</th>\n",
       "      <th>phosphoproteome_paired_difference</th>\n",
       "      <th>proteome_value</th>\n",
       "      <th>proteome_paired_value</th>\n",
       "      <th>proteome_paired_difference</th>\n",
       "      <th>has_mutation</th>\n",
       "      <th>rna_expression</th>\n",
       "      <th>paired_rna_expression</th>\n",
       "      <th>difference_rna_expression</th>\n",
       "      <th>driver</th>\n",
       "      <th>expanded_driver</th>\n",
       "      <th>LSCC</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>UCEC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCE1_LUAD_C3L-00001.T</th>\n",
       "      <td>0.246850</td>\n",
       "      <td>-0.915100</td>\n",
       "      <td>1.16195</td>\n",
       "      <td>-0.719400</td>\n",
       "      <td>-1.255500</td>\n",
       "      <td>0.536100</td>\n",
       "      <td>-0.719400</td>\n",
       "      <td>-1.255500</td>\n",
       "      <td>0.536100</td>\n",
       "      <td>False</td>\n",
       "      <td>13.128813</td>\n",
       "      <td>11.484098</td>\n",
       "      <td>1.644715</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCE1_LUAD_C3L-00009.T</th>\n",
       "      <td>-0.354350</td>\n",
       "      <td>-0.333900</td>\n",
       "      <td>-0.02045</td>\n",
       "      <td>0.213700</td>\n",
       "      <td>-3.734400</td>\n",
       "      <td>3.948100</td>\n",
       "      <td>0.213700</td>\n",
       "      <td>-3.734400</td>\n",
       "      <td>3.948100</td>\n",
       "      <td>False</td>\n",
       "      <td>21.503211</td>\n",
       "      <td>10.644732</td>\n",
       "      <td>10.858479</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCE1_LUAD_C3L-00080.T</th>\n",
       "      <td>0.347850</td>\n",
       "      <td>-0.093750</td>\n",
       "      <td>0.44160</td>\n",
       "      <td>-0.791400</td>\n",
       "      <td>-2.024100</td>\n",
       "      <td>1.232700</td>\n",
       "      <td>-0.791400</td>\n",
       "      <td>-2.024100</td>\n",
       "      <td>1.232700</td>\n",
       "      <td>False</td>\n",
       "      <td>13.288162</td>\n",
       "      <td>11.205924</td>\n",
       "      <td>2.082239</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCE1_LUAD_C3L-00083.T</th>\n",
       "      <td>0.221800</td>\n",
       "      <td>0.118350</td>\n",
       "      <td>0.10345</td>\n",
       "      <td>2.379900</td>\n",
       "      <td>1.897900</td>\n",
       "      <td>0.482000</td>\n",
       "      <td>2.379900</td>\n",
       "      <td>1.897900</td>\n",
       "      <td>0.482000</td>\n",
       "      <td>False</td>\n",
       "      <td>7.451822</td>\n",
       "      <td>10.310898</td>\n",
       "      <td>-2.859075</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCE1_LUAD_C3L-00093.T</th>\n",
       "      <td>0.557200</td>\n",
       "      <td>0.734500</td>\n",
       "      <td>-0.17730</td>\n",
       "      <td>-0.312550</td>\n",
       "      <td>-0.312550</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.312550</td>\n",
       "      <td>-0.312550</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>18.150735</td>\n",
       "      <td>14.897932</td>\n",
       "      <td>3.252803</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_UCEC_S067</th>\n",
       "      <td>0.003386</td>\n",
       "      <td>0.003386</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.512067</td>\n",
       "      <td>0.950653</td>\n",
       "      <td>-0.438587</td>\n",
       "      <td>0.512067</td>\n",
       "      <td>0.950653</td>\n",
       "      <td>-0.438587</td>\n",
       "      <td>False</td>\n",
       "      <td>20.352733</td>\n",
       "      <td>32.856939</td>\n",
       "      <td>-12.504206</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_UCEC_S075</th>\n",
       "      <td>0.019500</td>\n",
       "      <td>0.741000</td>\n",
       "      <td>-0.72150</td>\n",
       "      <td>-0.048744</td>\n",
       "      <td>0.950256</td>\n",
       "      <td>-0.999000</td>\n",
       "      <td>-0.048744</td>\n",
       "      <td>0.950256</td>\n",
       "      <td>-0.999000</td>\n",
       "      <td>False</td>\n",
       "      <td>22.233861</td>\n",
       "      <td>26.952750</td>\n",
       "      <td>-4.718888</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_UCEC_S086</th>\n",
       "      <td>-0.521500</td>\n",
       "      <td>-0.190000</td>\n",
       "      <td>-0.33150</td>\n",
       "      <td>-0.269828</td>\n",
       "      <td>0.012692</td>\n",
       "      <td>-0.282520</td>\n",
       "      <td>-0.269828</td>\n",
       "      <td>0.012692</td>\n",
       "      <td>-0.282520</td>\n",
       "      <td>False</td>\n",
       "      <td>21.290153</td>\n",
       "      <td>18.449954</td>\n",
       "      <td>2.840199</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_UCEC_S087</th>\n",
       "      <td>-0.040500</td>\n",
       "      <td>0.219000</td>\n",
       "      <td>-0.25950</td>\n",
       "      <td>-0.511296</td>\n",
       "      <td>0.331604</td>\n",
       "      <td>-0.842900</td>\n",
       "      <td>-0.511296</td>\n",
       "      <td>0.331604</td>\n",
       "      <td>-0.842900</td>\n",
       "      <td>False</td>\n",
       "      <td>17.309874</td>\n",
       "      <td>36.406614</td>\n",
       "      <td>-19.096740</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_UCEC_S096</th>\n",
       "      <td>-0.514000</td>\n",
       "      <td>0.052000</td>\n",
       "      <td>-0.56600</td>\n",
       "      <td>-0.088976</td>\n",
       "      <td>0.989714</td>\n",
       "      <td>-1.078690</td>\n",
       "      <td>-0.088976</td>\n",
       "      <td>0.989714</td>\n",
       "      <td>-1.078690</td>\n",
       "      <td>False</td>\n",
       "      <td>19.686493</td>\n",
       "      <td>57.144577</td>\n",
       "      <td>-37.458084</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>103362 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        acetylome_value  acetylome_paired_value  \\\n",
       "ABCE1_LUAD_C3L-00001.T         0.246850               -0.915100   \n",
       "ABCE1_LUAD_C3L-00009.T        -0.354350               -0.333900   \n",
       "ABCE1_LUAD_C3L-00080.T         0.347850               -0.093750   \n",
       "ABCE1_LUAD_C3L-00083.T         0.221800                0.118350   \n",
       "ABCE1_LUAD_C3L-00093.T         0.557200                0.734500   \n",
       "...                                 ...                     ...   \n",
       "ZYX_UCEC_S067                  0.003386                0.003386   \n",
       "ZYX_UCEC_S075                  0.019500                0.741000   \n",
       "ZYX_UCEC_S086                 -0.521500               -0.190000   \n",
       "ZYX_UCEC_S087                 -0.040500                0.219000   \n",
       "ZYX_UCEC_S096                 -0.514000                0.052000   \n",
       "\n",
       "                        acetylome_paired_difference  phosphoproteome_value  \\\n",
       "ABCE1_LUAD_C3L-00001.T                      1.16195              -0.719400   \n",
       "ABCE1_LUAD_C3L-00009.T                     -0.02045               0.213700   \n",
       "ABCE1_LUAD_C3L-00080.T                      0.44160              -0.791400   \n",
       "ABCE1_LUAD_C3L-00083.T                      0.10345               2.379900   \n",
       "ABCE1_LUAD_C3L-00093.T                     -0.17730              -0.312550   \n",
       "...                                             ...                    ...   \n",
       "ZYX_UCEC_S067                               0.00000               0.512067   \n",
       "ZYX_UCEC_S075                              -0.72150              -0.048744   \n",
       "ZYX_UCEC_S086                              -0.33150              -0.269828   \n",
       "ZYX_UCEC_S087                              -0.25950              -0.511296   \n",
       "ZYX_UCEC_S096                              -0.56600              -0.088976   \n",
       "\n",
       "                        phosphoproteome_paired_value  \\\n",
       "ABCE1_LUAD_C3L-00001.T                     -1.255500   \n",
       "ABCE1_LUAD_C3L-00009.T                     -3.734400   \n",
       "ABCE1_LUAD_C3L-00080.T                     -2.024100   \n",
       "ABCE1_LUAD_C3L-00083.T                      1.897900   \n",
       "ABCE1_LUAD_C3L-00093.T                     -0.312550   \n",
       "...                                              ...   \n",
       "ZYX_UCEC_S067                               0.950653   \n",
       "ZYX_UCEC_S075                               0.950256   \n",
       "ZYX_UCEC_S086                               0.012692   \n",
       "ZYX_UCEC_S087                               0.331604   \n",
       "ZYX_UCEC_S096                               0.989714   \n",
       "\n",
       "                        phosphoproteome_paired_difference  proteome_value  \\\n",
       "ABCE1_LUAD_C3L-00001.T                           0.536100       -0.719400   \n",
       "ABCE1_LUAD_C3L-00009.T                           3.948100        0.213700   \n",
       "ABCE1_LUAD_C3L-00080.T                           1.232700       -0.791400   \n",
       "ABCE1_LUAD_C3L-00083.T                           0.482000        2.379900   \n",
       "ABCE1_LUAD_C3L-00093.T                           0.000000       -0.312550   \n",
       "...                                                   ...             ...   \n",
       "ZYX_UCEC_S067                                   -0.438587        0.512067   \n",
       "ZYX_UCEC_S075                                   -0.999000       -0.048744   \n",
       "ZYX_UCEC_S086                                   -0.282520       -0.269828   \n",
       "ZYX_UCEC_S087                                   -0.842900       -0.511296   \n",
       "ZYX_UCEC_S096                                   -1.078690       -0.088976   \n",
       "\n",
       "                        proteome_paired_value  proteome_paired_difference  \\\n",
       "ABCE1_LUAD_C3L-00001.T              -1.255500                    0.536100   \n",
       "ABCE1_LUAD_C3L-00009.T              -3.734400                    3.948100   \n",
       "ABCE1_LUAD_C3L-00080.T              -2.024100                    1.232700   \n",
       "ABCE1_LUAD_C3L-00083.T               1.897900                    0.482000   \n",
       "ABCE1_LUAD_C3L-00093.T              -0.312550                    0.000000   \n",
       "...                                       ...                         ...   \n",
       "ZYX_UCEC_S067                        0.950653                   -0.438587   \n",
       "ZYX_UCEC_S075                        0.950256                   -0.999000   \n",
       "ZYX_UCEC_S086                        0.012692                   -0.282520   \n",
       "ZYX_UCEC_S087                        0.331604                   -0.842900   \n",
       "ZYX_UCEC_S096                        0.989714                   -1.078690   \n",
       "\n",
       "                        has_mutation  rna_expression  paired_rna_expression  \\\n",
       "ABCE1_LUAD_C3L-00001.T         False       13.128813              11.484098   \n",
       "ABCE1_LUAD_C3L-00009.T         False       21.503211              10.644732   \n",
       "ABCE1_LUAD_C3L-00080.T         False       13.288162              11.205924   \n",
       "ABCE1_LUAD_C3L-00083.T         False        7.451822              10.310898   \n",
       "ABCE1_LUAD_C3L-00093.T         False       18.150735              14.897932   \n",
       "...                              ...             ...                    ...   \n",
       "ZYX_UCEC_S067                  False       20.352733              32.856939   \n",
       "ZYX_UCEC_S075                  False       22.233861              26.952750   \n",
       "ZYX_UCEC_S086                  False       21.290153              18.449954   \n",
       "ZYX_UCEC_S087                  False       17.309874              36.406614   \n",
       "ZYX_UCEC_S096                  False       19.686493              57.144577   \n",
       "\n",
       "                        difference_rna_expression  driver  expanded_driver  \\\n",
       "ABCE1_LUAD_C3L-00001.T                   1.644715   False            False   \n",
       "ABCE1_LUAD_C3L-00009.T                  10.858479   False            False   \n",
       "ABCE1_LUAD_C3L-00080.T                   2.082239   False            False   \n",
       "ABCE1_LUAD_C3L-00083.T                  -2.859075   False            False   \n",
       "ABCE1_LUAD_C3L-00093.T                   3.252803   False            False   \n",
       "...                                           ...     ...              ...   \n",
       "ZYX_UCEC_S067                          -12.504206   False            False   \n",
       "ZYX_UCEC_S075                           -4.718888   False            False   \n",
       "ZYX_UCEC_S086                            2.840199   False            False   \n",
       "ZYX_UCEC_S087                          -19.096740   False            False   \n",
       "ZYX_UCEC_S096                          -37.458084   False            False   \n",
       "\n",
       "                        LSCC  LUAD  UCEC  \n",
       "ABCE1_LUAD_C3L-00001.T   0.0   1.0   0.0  \n",
       "ABCE1_LUAD_C3L-00009.T   0.0   1.0   0.0  \n",
       "ABCE1_LUAD_C3L-00080.T   0.0   1.0   0.0  \n",
       "ABCE1_LUAD_C3L-00083.T   0.0   1.0   0.0  \n",
       "ABCE1_LUAD_C3L-00093.T   0.0   1.0   0.0  \n",
       "...                      ...   ...   ...  \n",
       "ZYX_UCEC_S067            0.0   0.0   1.0  \n",
       "ZYX_UCEC_S075            0.0   0.0   1.0  \n",
       "ZYX_UCEC_S086            0.0   0.0   1.0  \n",
       "ZYX_UCEC_S087            0.0   0.0   1.0  \n",
       "ZYX_UCEC_S096            0.0   0.0   1.0  \n",
       "\n",
       "[103362 rows x 18 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fc = collapse_by_dtype_and_gene(filtered_combined)\n",
    "fc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## remove na over x percent and impute the rest\n",
    "# remove = ['paired_value', 'paired_difference', 'paired_rna_expression', 'difference_rna_expression',\n",
    "#          ]\n",
    "# filtered_combined = combined[[c for c in combined.columns if c not in remove]]\n",
    "# filtered_combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from sklearn.impute import SimpleImputer\n",
    "# categorical = ['sample_type', 'data_type', 'has_mutation', 'gene']\n",
    "# cont = [c for c in filtered_combined.columns if c not in categorical]\n",
    "# filtered_combined = filtered_combined.replace(np.inf, np.nan)\n",
    "# imputer = SimpleImputer(strategy='median')\n",
    "# X = imputer.fit_transform(filtered_combined[cont])\n",
    "\n",
    "# imputed_df = pd.DataFrame(data=X, index=filtered_combined.index, columns=cont)\n",
    "# new = pd.concat((filtered_combined[categorical], imputed_df), axis=1)\n",
    "# new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "fc.to_csv('/data/results/all_test_v1.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cancer_to_protein['CPTAC3_UCEC_discovery']['proteome']['metadata'].sort_values('case_id').to_csv(\n",
    "#         '/data/results/ucec_metadata.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## differentials"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fps = sorted(os_helpers.listfiles('/data/differentials/', regex='.tsv.gz'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fps = sorted(os_helpers.listfiles('/data/differentials/tumor_normal_enrichment/', regex='.tsv.gz'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_to_site_differentials = {}\n",
    "for fp in fps:\n",
    "    \n",
    "    if 'tumor_normal_diff' in fp:\n",
    "        cancer = fp.split('/')[-1].split('.')[0]\n",
    "        if cancer not in cancer_to_site_differentials:\n",
    "            cancer_to_site_differentials[cancer] = {}\n",
    "        df = pd.read_csv(fp, sep='\\t', index_col='site_id')\n",
    "        df = df[['tumor_normal_median_log2fc', 't_fdr']]\n",
    "        if '/acetyl/' in fp: cancer_to_site_differentials[cancer]['acetyl'] = df\n",
    "        if '/phospho/' in fp: cancer_to_site_differentials[cancer]['phospho'] = df\n",
    "        if '/ubiquityl/' in fp: cancer_to_site_differentials[cancer]['ubiquityl'] = df\n",
    "\n",
    "cancer_to_combined_site_diffs = {}\n",
    "for cancer, diffs in cancer_to_site_differentials.items():\n",
    "    combined = None\n",
    "    for k, df in diffs.items():\n",
    "        df.index = [f'{x}_{k}' for x in df.index]\n",
    "        df.index.name = 'site_id'\n",
    "        df['datatype'] = [k] * df.shape[0]\n",
    "        if combined is None:\n",
    "            combined = df\n",
    "        else:\n",
    "            combined = pd.concat((combined, df), axis=0)\n",
    "    cancer_to_combined_site_diffs[cancer] = combined\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_to_combined_site_diffs['CPTAC3_UCEC_discovery']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_to_combined_site_diffs['CPTAC3_UCEC_discovery'].to_csv('/data/results/ucec_differentials.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/data/differentials/tumor_normal_enrichment/acetyl/CPTAC3_UCEC_discovery.fisher_exact.v2.1.tsv.gz',\n",
    "                sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv('/data/differentials/tumor_normal_diff/acetyl/CPTAC3_UCEC_discovery.fisher_exact.v2.1.tsv.gz',\n",
    "#                 sep='\\t')\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv('/data/differentials/tumor_normal_diff/phospho/CPTAC3_UCEC_discovery.fisher_exact.v2.1.tsv.gz',\n",
    "#                 sep='\\t')\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# len(set(df['site_id']).intersection(set(cancer_to_protein['CPTAC3_UCEC_discovery']['acetylome']['processed_data'].index)))\n",
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv('/data/differentials/tumor_normal_diff/phospho/CPTAC3_GBM_discovery.fisher_exact.v2.1.tsv.gz',\n",
    "#                 sep='\\t')\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv('/data/differentials/tumor_normal_enrichment/phospho/CPTAC3_GBM_discovery.fisher_exact.v2.1.tsv.gz',\n",
    "#                 sep='\\t')\n",
    "# df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## corum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/data/uniprot_to_hugo_map.tsv', sep='\\t', header=None)\n",
    "uniprot_to_hugo = {k:v for k, v in zip(df[0], df[1])}\n",
    "uniprot_to_hugo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/data/corum.tsv', sep='\\t')\n",
    "complexes = [[uniprot_to_hugo[p] for p in entry.split(';') if p in uniprot_to_hugo]\n",
    "             for entry in df['subunits(UniProt IDs)']]\n",
    "complexes = [c for c in complexes if len(c)>1]\n",
    "complexes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "json.dump(complexes, open('/data/results/corum_complexes.json', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/data/reactome.tsv', sep='\\t', header=None)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
